{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Variants in DNAJC13 and their Association with Parkinson's Disease Across Different Ancestral Backgrounds\n",
    "\n",
    "* **Project**: DNAJC13 Gene Assesment\n",
    "* **Last updated**: September 2024\n",
    "* **Version**: Python 3.9\n",
    "* **Data**: GP2 Release 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "This notebook is focused on analysing association between DNAJC13 variants in Ashkenazi Jewish (AJ) ancestry and Parkinson's disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_15583/3943855956.py:26: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## AMP-PD v2.5\n",
    "## Explicitly define release v2.5 path \n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_GATK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "print('AMP-PD v2.5')\n",
    "print(f'Path to AMP-PD v2.5 Clinical Data @ `AMP_CLINICAL_RELEASE_PATH`: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v2.5 WGS Data @ `AMP_WGS_RELEASE_PLINK_PATH`: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')\n",
    "\n",
    "## GP2 v7.0\n",
    "## Explicitly define release v7.0 path \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = 'AJ'\n",
    "WORK_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "WORKSPACE_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "\n",
    "! mkdir {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/plink2_linux_amd_avx2_20240704.zip\n",
    "unzip -o plink2_linux_amd_avx2_20240704.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR: Download resources for annotation\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rvtests is already installed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## chr 3\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_IMPUTED_GENO_PATH}/{ancestry}/chr3_{ancestry}_release7.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## master key\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_CLINICAL_RELEASE_PATH}/master_key_release7_final.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/{ancestry}_release7.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## PCs\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RAW_GENO_PATH}/{ancestry}/{ancestry}_release7.eigenvec {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# refFlat\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {WORKSPACE_BUCKET}/notebooks/misc/refFlat_HG38_all_chr.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_release7.eigenvec  chr3_AJ_release7.pgen  master_key_release7_final.csv\r\n",
      "AJ_release7.related   chr3_AJ_release7.psam  refFlat_HG38_all_chr.txt\r\n",
      "chr3_AJ_release7.log  chr3_AJ_release7.pvar\r\n"
     ]
    }
   ],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Covariate File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(f'{WORK_DIR}/master_key_release7_final.csv')\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'race_for_qc', 'label']]\n",
    "key.rename(columns = {'GP2sampleID': 'IID',\n",
    "                      'baseline_GP2_phenotype':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'race_for_qc':'RACE',\n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = key[key['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals \n",
    "related_df = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.related')\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID1'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(ancestry_key.shape)\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (0/1)\n",
    "    # PD = 1; control = 0\n",
    "pheno_mapping = {\"PD\": 1, \"Control\": 0}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "1       1585\n",
       "0        771\n",
       "<NA>     240\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get the PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.eigenvec', sep='\\t')\n",
    "\n",
    "# Split the single column into multiple columns (No longer needed with release 7)\n",
    "#pcs_split = pcs['#FID,IID,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10'].str.split(',', expand=True)\n",
    "\n",
    "selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "\n",
    "\n",
    "#pcs = pd.DataFrame(data=pcs_split.iloc[:, 1:7].values, columns=selected_columns)\n",
    "pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "\n",
    "# Drop the first row (since it's now the column names)\n",
    "pcs = pcs.drop(0)\n",
    "\n",
    "# Reset the index to remove any potential issues\n",
    "pcs = pcs.reset_index(drop=True)\n",
    "\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "print(pcs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert sex to binary (0/1)\n",
    "    # Female = 1; male = 0\n",
    "sex_mapping = {\"Female\": 1, \"Male\": 0}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "0       1598\n",
       "1        987\n",
       "<NA>      11\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX\n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
       "       'AAO', 'RACE', 'label', 'PHENO'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Make covariate file\n",
    "df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['FID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = df[['FID','IID', 'SEX', 'PHENO', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO  SEX\n",
       "0      0      394\n",
       "       1      358\n",
       "1      0      986\n",
       "       1      492\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_covariate_file.txt  chr3_AJ_release7.log   master_key_release7_final.csv\r\n",
      "AJ_release7.eigenvec   chr3_AJ_release7.pgen  refFlat_HG38_all_chr.txt\r\n",
      "AJ_release7.related    chr3_AJ_release7.psam\r\n",
      "AJ.samplestoKeep       chr3_AJ_release7.pvar\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}//')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation using ANNOVAR\n",
    "\n",
    "* *DNAJC13* from NCBI gene\n",
    "* hg38 (chr3:132417502-132539032)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --export vcf id-paste=iid\n",
      "  --from-bp 132417502\n",
      "  --mac 2\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13\n",
      "  --pfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Tue Jul  9 23:23:07 2024\n",
      "7450 MiB RAM detected, ~5306 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2577 samples (987 females, 1590 males; 2577 founders) loaded from\n",
      "202407_BURDEN_AJ_R7/chr3_AJ_release7.psam.\n",
      "1681997 variants loaded from 202407_BURDEN_AJ_R7/chr3_AJ_release7.pvar.\n",
      "1 binary phenotype loaded (1246 cases, 393 controls).\n",
      "Calculating allele frequencies... done.\n",
      "163 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "690 variants remaining after main filters.\n",
      "--export vcf to 202407_BURDEN_AJ_R7/AJ_DNAJC13.vcf ... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "End time: Tue Jul  9 23:23:07 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 202407_BURDEN_AJ_R7/AJ_DNAJC13.vcf.gz > 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.avinput>\n",
      "NOTICE: Finished reading 709 lines from VCF file\n",
      "NOTICE: A total of 690 locus in VCF file passed QC threshold, representing 628 SNPs (442 transitions and 186 transversions) and 62 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1618356 SNP genotypes (1139034 transitions and 479322 transversions) and 159774 indels/substitutions for 2577 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.refGene -exonsort -nofirstcodondel 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.refGene.variant_function, 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 690 unique variants in 690 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 690 unique variants in 690 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 6\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_DNAJC13.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo2580</th>\n",
       "      <th>Otherinfo2581</th>\n",
       "      <th>Otherinfo2582</th>\n",
       "      <th>Otherinfo2583</th>\n",
       "      <th>Otherinfo2584</th>\n",
       "      <th>Otherinfo2585</th>\n",
       "      <th>Otherinfo2586</th>\n",
       "      <th>Otherinfo2587</th>\n",
       "      <th>Otherinfo2588</th>\n",
       "      <th>Otherinfo2589</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132417864</td>\n",
       "      <td>132417864</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132418296</td>\n",
       "      <td>132418296</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132418415</td>\n",
       "      <td>132418415</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132418488</td>\n",
       "      <td>132418488</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132418506</td>\n",
       "      <td>132418506</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>3</td>\n",
       "      <td>132538679</td>\n",
       "      <td>132538679</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*397A&gt;T;NM_015268:c.*397A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>3</td>\n",
       "      <td>132538787</td>\n",
       "      <td>132538787</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*505T&gt;G;NM_015268:c.*505T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>3</td>\n",
       "      <td>132538890</td>\n",
       "      <td>132538890</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*608G&gt;A;NM_015268:c.*608G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>3</td>\n",
       "      <td>132538903</td>\n",
       "      <td>132538903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*621G&gt;A;NM_015268:c.*621G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>3</td>\n",
       "      <td>132539011</td>\n",
       "      <td>132539011</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*729A&gt;G;NM_015268:c.*729A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>690 rows Ã— 2600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      3  132417864  132417864   G   C     intronic      DNAJC13   \n",
       "1      3  132418296  132418296   C   T     intronic      DNAJC13   \n",
       "2      3  132418415  132418415   A   G     intronic      DNAJC13   \n",
       "3      3  132418488  132418488   G   T     intronic      DNAJC13   \n",
       "4      3  132418506  132418506   T   G     intronic      DNAJC13   \n",
       "..   ...        ...        ...  ..  ..          ...          ...   \n",
       "685    3  132538679  132538679   A   T         UTR3      DNAJC13   \n",
       "686    3  132538787  132538787   T   G         UTR3      DNAJC13   \n",
       "687    3  132538890  132538890   G   A         UTR3      DNAJC13   \n",
       "688    3  132538903  132538903   G   A         UTR3      DNAJC13   \n",
       "689    3  132539011  132539011   A   G         UTR3      DNAJC13   \n",
       "\n",
       "                             GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                             .                  .   \n",
       "1                                             .                  .   \n",
       "2                                             .                  .   \n",
       "3                                             .                  .   \n",
       "4                                             .                  .   \n",
       "..                                          ...                ...   \n",
       "685  NM_001329126:c.*397A>T;NM_015268:c.*397A>T                  .   \n",
       "686  NM_001329126:c.*505T>G;NM_015268:c.*505T>G                  .   \n",
       "687  NM_001329126:c.*608G>A;NM_015268:c.*608G>A                  .   \n",
       "688  NM_001329126:c.*621G>A;NM_015268:c.*621G>A                  .   \n",
       "689  NM_001329126:c.*729A>G;NM_015268:c.*729A>G                  .   \n",
       "\n",
       "    AAChange.refGene  ... Otherinfo2580  Otherinfo2581 Otherinfo2582  \\\n",
       "0                  .  ...           0|0            0|0           0|0   \n",
       "1                  .  ...           0|0            0|0           0|0   \n",
       "2                  .  ...           1|0            0|1           0|0   \n",
       "3                  .  ...           0|0            0|0           0|0   \n",
       "4                  .  ...           0|0            0|0           0|0   \n",
       "..               ...  ...           ...            ...           ...   \n",
       "685                .  ...           0|0            0|0           0|0   \n",
       "686                .  ...           0|0            0|0           0|0   \n",
       "687                .  ...           0|0            0|0           0|0   \n",
       "688                .  ...           1|0            0|0           0|0   \n",
       "689                .  ...           0|0            0|0           0|0   \n",
       "\n",
       "    Otherinfo2583  Otherinfo2584  Otherinfo2585 Otherinfo2586 Otherinfo2587  \\\n",
       "0             0|0            0|0            0|0           0|0           0|0   \n",
       "1             0|0            0|0            0|0           0|0           0|0   \n",
       "2             0|0            1|0            0|0           1|1           0|0   \n",
       "3             0|0            0|0            0|0           0|0           0|0   \n",
       "4             0|0            0|0            0|0           0|0           0|0   \n",
       "..            ...            ...            ...           ...           ...   \n",
       "685           0|0            0|0            0|0           0|0           0|0   \n",
       "686           0|0            0|0            0|0           0|0           0|0   \n",
       "687           0|0            0|0            0|0           0|0           0|0   \n",
       "688           1|1            0|1            1|0           0|0           0|0   \n",
       "689           0|0            0|0            0|0           0|0           0|0   \n",
       "\n",
       "    Otherinfo2588 Otherinfo2589  \n",
       "0             0|0           0|0  \n",
       "1             0|0           0|0  \n",
       "2             0|0           0|0  \n",
       "3             0|0           0|0  \n",
       "4             0|1           0|0  \n",
       "..            ...           ...  \n",
       "685           0|0           0|0  \n",
       "686           0|0           0|0  \n",
       "687           0|0           0|0  \n",
       "688           0|0           0|0  \n",
       "689           0|0           0|0  \n",
       "\n",
       "[690 rows x 2600 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter intronic\n",
    "intronic = gene[(gene['Func.refGene'] == 'intronic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR3\n",
    "utr3 = gene[(gene['Func.refGene'] == 'UTR3')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR5\n",
    "utr5 = gene[(gene['Func.refGene'] == 'UTR5')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter exonic and synonymous variants\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo2580</th>\n",
       "      <th>Otherinfo2581</th>\n",
       "      <th>Otherinfo2582</th>\n",
       "      <th>Otherinfo2583</th>\n",
       "      <th>Otherinfo2584</th>\n",
       "      <th>Otherinfo2585</th>\n",
       "      <th>Otherinfo2586</th>\n",
       "      <th>Otherinfo2587</th>\n",
       "      <th>Otherinfo2588</th>\n",
       "      <th>Otherinfo2589</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>132434618</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon2:c.A68G:p.K23R,DNAJC...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>132447465</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon4:c.G289A:p.A97T,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>3</td>\n",
       "      <td>132492497</td>\n",
       "      <td>132492497</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon33:c.A3707G:p.Y1236C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>3</td>\n",
       "      <td>132499215</td>\n",
       "      <td>132499215</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.T4246A:p.S1416T,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>3</td>\n",
       "      <td>132525718</td>\n",
       "      <td>132525718</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon52:c.G6169T:p.A2057S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15 rows Ã— 2600 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "82     3  132434618  132434618   A   G       exonic      DNAJC13   \n",
       "156    3  132447465  132447465   G   A       exonic      DNAJC13   \n",
       "283    3  132475009  132475009   C   A       exonic      DNAJC13   \n",
       "389    3  132492497  132492497   A   G       exonic      DNAJC13   \n",
       "405    3  132494190  132494190   A   G       exonic      DNAJC13   \n",
       "435    3  132499215  132499215   T   A       exonic      DNAJC13   \n",
       "436    3  132499231  132499231   C   T       exonic      DNAJC13   \n",
       "440    3  132499777  132499777   G   A       exonic      DNAJC13   \n",
       "441    3  132499779  132499779   G   T       exonic      DNAJC13   \n",
       "449    3  132502295  132502295   C   T       exonic      DNAJC13   \n",
       "450    3  132502299  132502299   G   A       exonic      DNAJC13   \n",
       "476    3  132507256  132507256   A   G       exonic      DNAJC13   \n",
       "602    3  132525718  132525718   G   T       exonic      DNAJC13   \n",
       "615    3  132528316  132528316   T   G       exonic      DNAJC13   \n",
       "681    3  132538225  132538225   G   A       exonic      DNAJC13   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "82                   .  nonsynonymous SNV   \n",
       "156                  .  nonsynonymous SNV   \n",
       "283                  .  nonsynonymous SNV   \n",
       "389                  .  nonsynonymous SNV   \n",
       "405                  .  nonsynonymous SNV   \n",
       "435                  .  nonsynonymous SNV   \n",
       "436                  .  nonsynonymous SNV   \n",
       "440                  .  nonsynonymous SNV   \n",
       "441                  .  nonsynonymous SNV   \n",
       "449                  .  nonsynonymous SNV   \n",
       "450                  .  nonsynonymous SNV   \n",
       "476                  .  nonsynonymous SNV   \n",
       "602                  .  nonsynonymous SNV   \n",
       "615                  .  nonsynonymous SNV   \n",
       "681                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo2580  \\\n",
       "82   DNAJC13:NM_001329126:exon2:c.A68G:p.K23R,DNAJC...  ...           0|0   \n",
       "156  DNAJC13:NM_001329126:exon4:c.G289A:p.A97T,DNAJ...  ...           0|0   \n",
       "283  DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...  ...           0|0   \n",
       "389  DNAJC13:NM_015268:exon33:c.A3707G:p.Y1236C,DNA...  ...           0|0   \n",
       "405  DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...  ...           0|0   \n",
       "435  DNAJC13:NM_015268:exon37:c.T4246A:p.S1416T,DNA...  ...           0|0   \n",
       "436  DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...  ...           0|0   \n",
       "440  DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...  ...           0|0   \n",
       "441  DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...  ...           1|1   \n",
       "449  DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...  ...           0|0   \n",
       "450  DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...  ...           0|0   \n",
       "476  DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...  ...           0|0   \n",
       "602  DNAJC13:NM_015268:exon52:c.G6169T:p.A2057S,DNA...  ...           0|0   \n",
       "615  DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...  ...           0|0   \n",
       "681  DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...  ...           0|0   \n",
       "\n",
       "     Otherinfo2581 Otherinfo2582 Otherinfo2583  Otherinfo2584  Otherinfo2585  \\\n",
       "82             0|0           0|0           0|0            0|0            0|0   \n",
       "156            0|0           0|0           0|0            0|0            0|0   \n",
       "283            0|0           0|0           0|0            0|0            0|0   \n",
       "389            0|0           0|0           0|0            0|0            0|0   \n",
       "405            0|0           0|0           0|0            0|0            0|0   \n",
       "435            0|0           0|0           0|0            0|0            0|0   \n",
       "436            0|0           0|0           0|0            0|0            0|0   \n",
       "440            0|0           0|0           0|0            0|0            0|0   \n",
       "441            0|1           1|0           1|1            1|1            1|1   \n",
       "449            0|0           0|0           0|0            0|0            0|0   \n",
       "450            0|0           0|0           0|0            0|0            0|0   \n",
       "476            0|1           0|0           0|0            1|0            0|1   \n",
       "602            0|0           0|0           0|0            0|0            0|0   \n",
       "615            0|0           0|0           0|0            0|0            0|0   \n",
       "681            0|0           0|0           0|0            0|0            0|0   \n",
       "\n",
       "    Otherinfo2586 Otherinfo2587 Otherinfo2588 Otherinfo2589  \n",
       "82            0|0           0|0           0|0           0|0  \n",
       "156           0|0           0|0           0|0           0|0  \n",
       "283           0|0           0|0           0|0           0|0  \n",
       "389           0|0           0|0           0|0           0|0  \n",
       "405           0|0           0|0           0|0           0|0  \n",
       "435           0|0           0|0           0|0           0|0  \n",
       "436           1|0           0|0           0|0           0|0  \n",
       "440           0|0           0|0           0|0           0|0  \n",
       "441           1|1           1|1           0|0           0|0  \n",
       "449           0|0           0|0           0|0           0|0  \n",
       "450           0|0           0|0           0|0           0|0  \n",
       "476           0|1           0|0           0|0           0|0  \n",
       "602           0|0           0|0           0|0           0|0  \n",
       "615           0|0           0|0           0|0           0|0  \n",
       "681           0|0           0|0           0|0           0|0  \n",
       "\n",
       "[15 rows x 2600 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "totalVariants = len(gene.axes[0])\n",
    "totalIntronic = len(intronic.axes[0])\n",
    "totalUTR3 = len(utr3.axes[0])\n",
    "totalUTR5 = len(utr5.axes[0])\n",
    "totalExonicSyn = len(coding_synonymous.axes[0])\n",
    "totalExonicNonSyn = len(coding_nonsynonymous.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Variants:  690\n",
      "Intronic: 660\n",
      "UTR3: 8\n",
      "UTR5: 0\n",
      "Exonic Syn: 7\n",
      "Exonic NonSyn: 15\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Variants: \", totalVariants)\n",
    "print(\"Intronic:\", totalIntronic)\n",
    "print(\"UTR3:\", totalUTR3)\n",
    "print(\"UTR5:\", totalUTR5)\n",
    "print(\"Exonic Syn:\", totalExonicSyn)\n",
    "print(\"Exonic NonSyn:\", totalExonicNonSyn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>132434618</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>132447465</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>3</td>\n",
       "      <td>132492497</td>\n",
       "      <td>132492497</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>3</td>\n",
       "      <td>132499215</td>\n",
       "      <td>132499215</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>3</td>\n",
       "      <td>132525718</td>\n",
       "      <td>132525718</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "82     3  132434618  132434618      DNAJC13\n",
       "156    3  132447465  132447465      DNAJC13\n",
       "283    3  132475009  132475009      DNAJC13\n",
       "389    3  132492497  132492497      DNAJC13\n",
       "405    3  132494190  132494190      DNAJC13\n",
       "435    3  132499215  132499215      DNAJC13\n",
       "436    3  132499231  132499231      DNAJC13\n",
       "440    3  132499777  132499777      DNAJC13\n",
       "441    3  132499779  132499779      DNAJC13\n",
       "449    3  132502295  132502295      DNAJC13\n",
       "450    3  132502299  132502299      DNAJC13\n",
       "476    3  132507256  132507256      DNAJC13\n",
       "602    3  132525718  132525718      DNAJC13\n",
       "615    3  132528316  132528316      DNAJC13\n",
       "681    3  132538225  132538225      DNAJC13"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_covariate_file.txt\t\t\t     AJ_release7.related\r\n",
      "AJ_DNAJC13.all_coding_nonsyn.variantstoKeep  AJ.samplestoKeep\r\n",
      "AJ_DNAJC13.annovar.avinput\t\t     chr3_AJ_release7.log\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.txt\t     chr3_AJ_release7.pgen\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.vcf\t     chr3_AJ_release7.psam\r\n",
      "AJ_DNAJC13.log\t\t\t\t     chr3_AJ_release7.pvar\r\n",
      "AJ_DNAJC13.vcf.gz\t\t\t     master_key_release7_final.csv\r\n",
      "AJ_DNAJC13.vcf.gz.tbi\t\t\t     refFlat_HG38_all_chr.txt\r\n",
      "AJ_release7.eigenvec\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep {WORKSPACE_BUCKET}//{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/chr3_AJ_release7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --pfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "\n",
      "Start time: Tue Jul  9 23:23:40 2024\n",
      "7450 MiB RAM detected, ~5282 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2577 samples (987 females, 1590 males; 2577 founders) loaded from\n",
      "202407_BURDEN_AJ_R7/chr3_AJ_release7.psam.\n",
      "1681997 variants loaded from 202407_BURDEN_AJ_R7/chr3_AJ_release7.pvar.\n",
      "1 binary phenotype loaded (1246 cases, 393 controls).\n",
      "Writing 202407_BURDEN_AJ_R7/chr3_AJ_release7.fam ... done.\n",
      "Writing 202407_BURDEN_AJ_R7/chr3_AJ_release7.bim ... done.\n",
      "Writing 202407_BURDEN_AJ_R7/chr3_AJ_release7.bed ... 1115192327313538424650545862667074778185899397done.\n",
      "End time: Tue Jul  9 23:23:45 2024\n"
     ]
    }
   ],
   "source": [
    "# Convert the files from Plink 2.0 to Plink 1.9 format\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}_release7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding\n",
      "  --recode vcf-iid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "15 variants and 2576 people pass filters and QC.\n",
      "Among remaining phenotypes, 1245 are cases and 393 are controls.  (938\n",
      "phenotypes are missing.)\n",
      "--recode vcf-iid to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "    # later do based on class and frequency\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--recode vcf-iid \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\r\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\r\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\r\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\r\n",
      "\r\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\r\n",
      "\r\n",
      "Available Options\r\n",
      "      Basic Input/Output: --inVcf [202407_BURDEN_AJ_R7/AJ_DNAJC13.coding.vcf.gz]\r\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\r\n",
      "                          --out [202407_BURDEN_AJ_R7/AJ_DNAJC13.burden.coding]\r\n",
      "                          --outputRaw\r\n",
      "       Specify Covariate: --covar [202407_BURDEN_AJ_R7/AJ_covariate_file.txt]\r\n",
      "                          --covar-name [SEX,PC1,PC2,PC3,PC4], --sex\r\n",
      "       Specify Phenotype: --pheno [202407_BURDEN_AJ_R7/AJ_covariate_file.txt]\r\n",
      "                          --inverseNormal, --useResidualAsPhenotype\r\n",
      "                          --mpheno [4], --pheno-name [], --qtl\r\n",
      "                          --multiplePheno []\r\n",
      "        Specify Genotype: --dosage [], --multipleAllele\r\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\r\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\r\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\r\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\r\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\r\n",
      "                          --annoType []\r\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\r\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skat\r\n",
      "                         skato], --meta []\r\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\r\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\r\n",
      "                          --boltPlinkNoCheck\r\n",
      "          Grouping Unit :\r\n",
      "                          --geneFile [202407_BURDEN_AJ_R7/refFlat_HG38_all_chr.txt]\r\n",
      "                          --gene [DNAJC13], --setList [], --setFile [], --set []\r\n",
      "        Frequency Cutoff: --freqUpper [], --freqLower []\r\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\r\n",
      "    Conditional Analysis: --condition []\r\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\r\n",
      "                          --outputID, --help\r\n",
      "\r\n",
      "Effective Options\r\n",
      "    --inVcf 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding.vcf.gz\r\n",
      "    --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.burden.coding\r\n",
      "    --covar 202407_BURDEN_AJ_R7/AJ_covariate_file.txt\r\n",
      "    --covar-name SEX,PC1,PC2,PC3,PC4\r\n",
      "    --pheno 202407_BURDEN_AJ_R7/AJ_covariate_file.txt\r\n",
      "    --mpheno 4\r\n",
      "    --kernel skat,skato\r\n",
      "    --geneFile 202407_BURDEN_AJ_R7/refFlat_HG38_all_chr.txt\r\n",
      "    --gene DNAJC13\r\n",
      "    --noweb\r\n",
      "    --hide-covar\r\n",
      "\r\n",
      "[INFO]\tProgram version: 20190205\r\n",
      "[INFO]\tAnalysis started at: Tue Jul  9 23:23:51 2024\r\n",
      "[INFO]\tLoaded [ 2576 ] samples from genotype files\r\n",
      "[WARN]\tSkip line 13 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 14 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 18 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 20 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 25 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 26 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 27 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 28 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 31 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 35 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 65 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 103 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 390 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 399 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 450 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 471 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 485 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 606 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 683 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 684 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 685 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 686 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 687 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 688 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 689 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 690 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 691 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 692 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 693 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 694 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 695 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 696 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 697 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 710 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 739 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 819 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 834 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 838 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 843 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 863 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 875 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 885 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 886 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 891 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 892 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 893 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 895 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 911 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 912 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 913 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 915 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 916 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 939 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 946 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 952 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 979 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1003 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1090 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1383 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1452 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1461 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1522 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1523 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1552 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1563 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1564 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1572 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1602 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1606 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1607 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1613 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1617 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1701 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1719 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1730 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1732 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1741 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1751 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1756 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1757 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1761 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1768 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1787 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1789 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1806 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1808 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1809 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1813 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1814 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1824 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1828 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1831 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1854 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1858 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1864 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1865 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1875 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1878 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1890 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1897 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1901 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1907 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1908 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1914 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1919 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1926 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1935 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1946 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1958 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1976 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1982 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1983 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1992 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 1997 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2013 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2016 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2020 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2027 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2029 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2030 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2031 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2034 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2038 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2043 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2046 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2047 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2052 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2064 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2069 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2073 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2075 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2076 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2084 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2091 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2092 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2094 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2108 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2111 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2113 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2114 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2115 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2119 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2120 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2125 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2129 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2131 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2141 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2142 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2145 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2148 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2149 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2152 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2160 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2162 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2164 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2167 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2179 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2182 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2186 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2190 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2197 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2201 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2202 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2208 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2211 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2221 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2227 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2228 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2234 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2240 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2245 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2249 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2254 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2256 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2259 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2273 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2283 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2286 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2289 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2290 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2294 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2295 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2296 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2297 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2298 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2299 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2300 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2301 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2302 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2303 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2304 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2305 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2306 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2307 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2308 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2309 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2310 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2311 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2312 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2313 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2314 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2315 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2316 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2317 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2318 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2319 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2320 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2321 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2322 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2323 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2324 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2325 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2326 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2327 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2328 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2329 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2330 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2331 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2332 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2333 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2334 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2335 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2336 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2337 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2338 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2339 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2340 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2341 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2342 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2343 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2344 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2345 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2346 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2347 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2348 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2349 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2350 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2351 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2352 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2353 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2354 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2355 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2356 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2357 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2358 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2359 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2360 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2361 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2362 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2363 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2364 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2365 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2366 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2367 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2368 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2369 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2370 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2371 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2372 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2373 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2374 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2375 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2376 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2377 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2378 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2379 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2380 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2381 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2382 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2383 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2384 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2385 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2386 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2387 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2388 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2389 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2390 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2391 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2392 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2393 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2394 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2395 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2396 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2397 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2398 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2399 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2400 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2401 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2402 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2403 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2404 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2405 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2406 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2407 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2408 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2409 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2410 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2411 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2412 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2413 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2414 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2415 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2416 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2417 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2418 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2419 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2420 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2421 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2422 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2423 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2424 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2425 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2426 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2427 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2428 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2429 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2430 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2431 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2432 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2433 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2434 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2435 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2436 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2437 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2438 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2439 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2440 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2441 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2442 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2443 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2444 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2445 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2446 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2447 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2448 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2449 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2450 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2451 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2470 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2523 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2524 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2525 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2526 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2527 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2528 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 2529 (short of columns) in phenotype file [ 202407_BURDEN_AJ_R7/AJ_covariate_file.txt ]\r\n",
      "[INFO]\tLoaded [ 2230 ] sample phenotypes\r\n",
      "[WARN]\tTotal [ 346 ] samples are dropped from VCF file due to missing phenotype.\r\n",
      "[WARN]\tDrop sample [ BBDP_000024_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000027_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000103_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000168_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000474_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000478_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000548_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000567_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000651_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ BBDP_000688_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tSkip outputting additional [ 336 ] samples due to missing phenotype.\r\n",
      "[INFO]\tBegin to read covariate file\r\n",
      "[WARN]\tTotal [ 346 ] samples are skipped from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000024_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000027_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000103_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000168_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000474_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000478_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000548_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000567_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000651_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BBDP_000688_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BIOFIND_000080_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip outputting additional [ 336 ] samples from covariate file with missing phenotypes\r\n",
      "[INFO]\tLoaded 0 male, 3 female and 2227 sex-unknown samples from 202407_BURDEN_AJ_R7/AJ_covariate_file.txt\n",
      "[INFO]\tAnalysis begins with [ 2230 ] samples...\n",
      "[INFO]\tSKAT test significance will be evaluated using 10000 permutations at alpha = 0.05 weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 15 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Tue Jul  9 23:23:52 2024\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "## RVtests with covariates \n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.burden.coding \\\n",
    "--kernel skat,skato \\\n",
    "--inVcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--mpheno 4 \\\n",
    "--gene DNAJC13 \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38_all_chr.txt \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4 #[WARN]Covariate [ PC5 ] has strong correlation [ r^2 = 1 ] with the response!\n",
    "\n",
    "\n",
    "# --out : Name of output \n",
    "# --burden cmc --kernel skato: tests to run \n",
    "# --inVcf : VCF file \n",
    "# --gene: gene name (if only looking at one or a few)\n",
    "# --geneFile refFlat.txt\n",
    "# --pheno :  covar file\n",
    "# --mpheno : # column that has phenotype information\n",
    "# --pheno-name : column name with phenotype in file\n",
    "# --covar : covar file\n",
    "# --freqUpper : optional, MAF cut-off\n",
    "# --covar-name : covariates, listed by column name, separated by commas (no spaces between commas)\n",
    "## 0=controls; 1=cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t2230\t15\t15\t19.9558\t0.5422\t10000\t2499\t19.9558\t1000\t0\t0.40016\r\n"
     ]
    }
   ],
   "source": [
    "## look at results \n",
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t2230\t15\t15\t41454.5\t0\t0.742893\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_covariate_file.txt\t\t\t     AJ_DNAJC13.vcf.gz.tbi\r\n",
      "AJ_DNAJC13.all_coding_nonsyn.variantstoKeep  AJ_release7.eigenvec\r\n",
      "AJ_DNAJC13.annovar.avinput\t\t     AJ_release7.related\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.txt\t     AJ.samplestoKeep\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.vcf\t     chr3_AJ_release7.bed\r\n",
      "AJ_DNAJC13.burden.coding.log\t\t     chr3_AJ_release7.bim\r\n",
      "AJ_DNAJC13.burden.coding.Skat.assoc\t     chr3_AJ_release7.fam\r\n",
      "AJ_DNAJC13.burden.coding.SkatO.assoc\t     chr3_AJ_release7.log\r\n",
      "AJ_DNAJC13.coding.log\t\t\t     chr3_AJ_release7.pgen\r\n",
      "AJ_DNAJC13.coding.vcf.gz\t\t     chr3_AJ_release7.psam\r\n",
      "AJ_DNAJC13.coding.vcf.gz.tbi\t\t     chr3_AJ_release7.pvar\r\n",
      "AJ_DNAJC13.log\t\t\t\t     master_key_release7_final.csv\r\n",
      "AJ_DNAJC13.vcf.gz\t\t\t     refFlat_HG38_all_chr.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.*.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --ci 0.95\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "15 variants and 2576 people pass filters and QC.\n",
      "Among remaining phenotypes, 1245 are cases and 393 are controls.  (938\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR           SE          L95          U95 \r\n",
      "   3   chr3:132434618:A:G  132434618    G 0.0008032        0    A       0.6317       0.4267           NA           NA           NA           NA \r\n",
      "   3   chr3:132447465:G:A  132447465    A 0.003213        0    G        2.531       0.1116           NA           NA           NA           NA \r\n",
      "   3   chr3:132475009:C:A  132475009    A 0.001205        0    C       0.9479       0.3303           NA           NA           NA           NA \r\n",
      "   3   chr3:132492497:A:G  132492497    G 0.0008032        0    A       0.6317       0.4267           NA           NA           NA           NA \r\n",
      "   3   chr3:132494190:A:G  132494190    G 0.0008032        0    A       0.6317       0.4267           NA           NA           NA           NA \r\n",
      "   3   chr3:132499215:T:A  132499215    A 0.0004026        0    T       0.3157       0.5742           NA           NA           NA           NA \r\n",
      "   3   chr3:132499231:C:T  132499231    T  0.01647  0.01908    C       0.2437       0.6215       0.8605       0.3046       0.4737        1.563 \r\n",
      "   3   chr3:132499777:G:A  132499777    A 0.007229 0.001272    G        3.676      0.05519        5.716        1.028       0.7619        42.89 \r\n",
      "   3   chr3:132499779:G:T  132499779    T   0.4823    0.486    G      0.03232       0.8573       0.9854      0.08186       0.8393        1.157 \r\n",
      "   3   chr3:132502295:C:T  132502295    T   0.0261  0.02672    C     0.008786       0.9253       0.9764       0.2544       0.5931        1.608 \r\n",
      "   3   chr3:132502299:G:A  132502299    A 0.005622 0.005089    G      0.03111         0.86        1.105       0.5684       0.3628        3.368 \r\n",
      "   3   chr3:132507256:A:G  132507256    G  0.08795    0.112    A        4.054      0.04406       0.7649       0.1334       0.5889       0.9935 \r\n",
      "   3   chr3:132525718:G:T  132525718    T 0.002414 0.001272    G       0.3643       0.5461        1.899        1.081       0.2283         15.8 \r\n",
      "   3   chr3:132528316:T:G  132528316    G 0.003213  0.01272    T        9.887     0.001664       0.2501       0.4761      0.09837        0.636 \r\n",
      "   3   chr3:132538225:G:A  132538225    A        0        0    G           NA           NA           NA           NA           NA           NA \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.111600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.330300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.574200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "      <td>C</td>\n",
       "      <td>0.621500</td>\n",
       "      <td>0.8605</td>\n",
       "      <td>0.47370</td>\n",
       "      <td>1.5630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>G</td>\n",
       "      <td>0.055190</td>\n",
       "      <td>5.7160</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>42.8900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.857300</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.83930</td>\n",
       "      <td>1.1570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "      <td>C</td>\n",
       "      <td>0.925300</td>\n",
       "      <td>0.9764</td>\n",
       "      <td>0.59310</td>\n",
       "      <td>1.6080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "      <td>G</td>\n",
       "      <td>0.860000</td>\n",
       "      <td>1.1050</td>\n",
       "      <td>0.36280</td>\n",
       "      <td>3.3680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.044060</td>\n",
       "      <td>0.7649</td>\n",
       "      <td>0.58890</td>\n",
       "      <td>0.9935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>G</td>\n",
       "      <td>0.546100</td>\n",
       "      <td>1.8990</td>\n",
       "      <td>0.22830</td>\n",
       "      <td>15.8000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001664</td>\n",
       "      <td>0.2501</td>\n",
       "      <td>0.09837</td>\n",
       "      <td>0.6360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1       F_A       F_U A2         P      OR      L95  \\\n",
       "0   chr3:132434618:A:G  G  0.000803  0.000000  A  0.426700     NaN      NaN   \n",
       "1   chr3:132447465:G:A  A  0.003213  0.000000  G  0.111600     NaN      NaN   \n",
       "2   chr3:132475009:C:A  A  0.001205  0.000000  C  0.330300     NaN      NaN   \n",
       "3   chr3:132492497:A:G  G  0.000803  0.000000  A  0.426700     NaN      NaN   \n",
       "4   chr3:132494190:A:G  G  0.000803  0.000000  A  0.426700     NaN      NaN   \n",
       "5   chr3:132499215:T:A  A  0.000403  0.000000  T  0.574200     NaN      NaN   \n",
       "6   chr3:132499231:C:T  T  0.016470  0.019080  C  0.621500  0.8605  0.47370   \n",
       "7   chr3:132499777:G:A  A  0.007229  0.001272  G  0.055190  5.7160  0.76190   \n",
       "8   chr3:132499779:G:T  T  0.482300  0.486000  G  0.857300  0.9854  0.83930   \n",
       "9   chr3:132502295:C:T  T  0.026100  0.026720  C  0.925300  0.9764  0.59310   \n",
       "10  chr3:132502299:G:A  A  0.005622  0.005089  G  0.860000  1.1050  0.36280   \n",
       "11  chr3:132507256:A:G  G  0.087950  0.112000  A  0.044060  0.7649  0.58890   \n",
       "12  chr3:132525718:G:T  T  0.002414  0.001272  G  0.546100  1.8990  0.22830   \n",
       "13  chr3:132528316:T:G  G  0.003213  0.012720  T  0.001664  0.2501  0.09837   \n",
       "14  chr3:132538225:G:A  A  0.000000  0.000000  G       NaN     NaN      NaN   \n",
       "\n",
       "        U95  \n",
       "0       NaN  \n",
       "1       NaN  \n",
       "2       NaN  \n",
       "3       NaN  \n",
       "4       NaN  \n",
       "5       NaN  \n",
       "6    1.5630  \n",
       "7   42.8900  \n",
       "8    1.1570  \n",
       "9    1.6080  \n",
       "10   3.3680  \n",
       "11   0.9935  \n",
       "12  15.8000  \n",
       "13   0.6360  \n",
       "14      NaN  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc', delim_whitespace=True, usecols=['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR', 'L95', 'U95'])\n",
    "assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "15 variants and 2576 people pass filters and QC.\n",
      "Among remaining phenotypes, 1245 are cases and 393 are controls.  (938\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.38533.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (14 variants) written to\n",
      "202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "## repeat association study applying multiple testing corrections for the raw p-values (like Bonferroni post-hoc)\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP      UNADJ         GC       BONF       HOLM   SIDAK_SS   SIDAK_SD     FDR_BH     FDR_BY\r\n",
      "   3   chr3:132528316:T:G   0.001664    0.00755     0.0233     0.0233    0.02305    0.02305     0.0233    0.07575 \r\n",
      "   3   chr3:132507256:A:G    0.04406    0.08713     0.6168     0.5728     0.4679     0.4433     0.2576     0.8375 \r\n",
      "   3   chr3:132499777:G:A    0.05519     0.1033     0.7727     0.6623     0.5484      0.494     0.2576     0.8375 \r\n",
      "   3   chr3:132447465:G:A     0.1116     0.1764          1          1     0.8092     0.7279     0.3906          1 \r\n",
      "   3   chr3:132475009:C:A     0.3303     0.4081          1          1     0.9963     0.9818     0.7468          1 \r\n",
      "   3   chr3:132434618:A:G     0.4267     0.4995          1          1     0.9996     0.9933     0.7468          1 \r\n",
      "   3   chr3:132492497:A:G     0.4267     0.4995          1          1     0.9996     0.9933     0.7468          1 \r\n",
      "   3   chr3:132494190:A:G     0.4267     0.4995          1          1     0.9996     0.9933     0.7468          1 \r\n",
      "   3   chr3:132525718:G:T     0.5461     0.6081          1          1          1     0.9933      0.791          1 \r\n",
      "   3   chr3:132499215:T:A     0.5742     0.6331          1          1          1     0.9933      0.791          1 \r\n",
      "   3   chr3:132499231:C:T     0.6215     0.6749          1          1          1     0.9933      0.791          1 \r\n",
      "   3   chr3:132499779:G:T     0.8573     0.8786          1          1          1     0.9971     0.9253          1 \r\n",
      "   3   chr3:132502299:G:A       0.86     0.8809          1          1          1     0.9971     0.9253          1 \r\n",
      "   3   chr3:132502295:C:T     0.9253     0.9365          1          1          1     0.9971     0.9253          1 \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>0.0233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.6168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.7727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP    BONF\n",
       "0   chr3:132528316:T:G  0.0233\n",
       "1   chr3:132507256:A:G  0.6168\n",
       "2   chr3:132499777:G:A  0.7727\n",
       "3   chr3:132447465:G:A  1.0000\n",
       "4   chr3:132475009:C:A  1.0000\n",
       "5   chr3:132434618:A:G  1.0000\n",
       "6   chr3:132492497:A:G  1.0000\n",
       "7   chr3:132494190:A:G  1.0000\n",
       "8   chr3:132525718:G:T  1.0000\n",
       "9   chr3:132499215:T:A  1.0000\n",
       "10  chr3:132499231:C:T  1.0000\n",
       "11  chr3:132499779:G:T  1.0000\n",
       "12  chr3:132502299:G:A  1.0000\n",
       "13  chr3:132502295:C:T  1.0000"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_adjusted = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted', delim_whitespace=True, usecols=['SNP', 'BONF'])\n",
    "assoc_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.frq .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2          MAF  NCHROBS\r\n",
      "   3   chr3:132434618:A:G    G    A     0.001553     5152\r\n",
      "   3   chr3:132447465:G:A    A    G     0.002911     5152\r\n",
      "   3   chr3:132475009:C:A    A    C    0.0009705     5152\r\n",
      "   3   chr3:132492497:A:G    G    A    0.0005828     5148\r\n",
      "   3   chr3:132494190:A:G    G    A    0.0003882     5152\r\n",
      "   3   chr3:132499215:T:A    A    T    0.0003894     5136\r\n",
      "   3   chr3:132499231:C:T    T    C      0.01533     5152\r\n",
      "   3   chr3:132499777:G:A    A    G     0.006599     5152\r\n",
      "   3   chr3:132499779:G:T    T    G       0.4917     5152\r\n",
      "   3   chr3:132502295:C:T    T    C      0.02717     5152\r\n",
      "   3   chr3:132502299:G:A    A    G     0.005241     5152\r\n",
      "   3   chr3:132507256:A:G    G    A      0.09569     5152\r\n",
      "   3   chr3:132525718:G:T    T    G     0.001943     5146\r\n",
      "   3   chr3:132528316:T:G    G    T     0.004076     5152\r\n",
      "   3   chr3:132538225:G:A    A    G    0.0003882     5152\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>0.001553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>0.002911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.000971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>0.000583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000388</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF\n",
       "0  chr3:132434618:A:G  0.001553\n",
       "1  chr3:132447465:G:A  0.002911\n",
       "2  chr3:132475009:C:A  0.000971\n",
       "3  chr3:132492497:A:G  0.000583\n",
       "4  chr3:132494190:A:G  0.000388"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq', delim_whitespace=True, usecols=['SNP', 'MAF'])\n",
    "freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq case-control\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.frq.cc .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq case-control \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2        MAF_A        MAF_U  NCHROBS_A  NCHROBS_U\r\n",
      "   3   chr3:132434618:A:G    G    A    0.0008032            0       2490        786\r\n",
      "   3   chr3:132447465:G:A    A    G     0.003213            0       2490        786\r\n",
      "   3   chr3:132475009:C:A    A    C     0.001205            0       2490        786\r\n",
      "   3   chr3:132492497:A:G    G    A    0.0008032            0       2490        786\r\n",
      "   3   chr3:132494190:A:G    G    A    0.0008032            0       2490        786\r\n",
      "   3   chr3:132499215:T:A    A    T    0.0004026            0       2484        784\r\n",
      "   3   chr3:132499231:C:T    T    C      0.01647      0.01908       2490        786\r\n",
      "   3   chr3:132499777:G:A    A    G     0.007229     0.001272       2490        786\r\n",
      "   3   chr3:132499779:G:T    T    G       0.4823        0.486       2490        786\r\n",
      "   3   chr3:132502295:C:T    T    C       0.0261      0.02672       2490        786\r\n",
      "   3   chr3:132502299:G:A    A    G     0.005622     0.005089       2490        786\r\n",
      "   3   chr3:132507256:A:G    G    A      0.08795        0.112       2490        786\r\n",
      "   3   chr3:132525718:G:T    T    G     0.002414     0.001272       2486        786\r\n",
      "   3   chr3:132528316:T:G    G    T     0.003213      0.01272       2490        786\r\n",
      "   3   chr3:132538225:G:A    A    G            0            0       2490        786\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP     MAF_A  MAF_U\n",
       "0  chr3:132434618:A:G  0.000803    0.0\n",
       "1  chr3:132447465:G:A  0.003213    0.0\n",
       "2  chr3:132475009:C:A  0.001205    0.0\n",
       "3  chr3:132492497:A:G  0.000803    0.0\n",
       "4  chr3:132494190:A:G  0.000803    0.0"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq_cc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc', delim_whitespace=True, usecols=['SNP', 'MAF_A', 'MAF_U'])\n",
    "freq_cc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>0.001553</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>0.002911</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.000971</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>0.000583</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "      <td>0.000389</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.015330</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.006599</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.491700</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.027170</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "      <td>0.005241</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.095690</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "      <td>0.001943</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>0.004076</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP       MAF     MAF_A     MAF_U\n",
       "0   chr3:132434618:A:G  0.001553  0.000803  0.000000\n",
       "1   chr3:132447465:G:A  0.002911  0.003213  0.000000\n",
       "2   chr3:132475009:C:A  0.000971  0.001205  0.000000\n",
       "3   chr3:132492497:A:G  0.000583  0.000803  0.000000\n",
       "4   chr3:132494190:A:G  0.000388  0.000803  0.000000\n",
       "5   chr3:132499215:T:A  0.000389  0.000403  0.000000\n",
       "6   chr3:132499231:C:T  0.015330  0.016470  0.019080\n",
       "7   chr3:132499777:G:A  0.006599  0.007229  0.001272\n",
       "8   chr3:132499779:G:T  0.491700  0.482300  0.486000\n",
       "9   chr3:132502295:C:T  0.027170  0.026100  0.026720\n",
       "10  chr3:132502299:G:A  0.005241  0.005622  0.005089\n",
       "11  chr3:132507256:A:G  0.095690  0.087950  0.112000\n",
       "12  chr3:132525718:G:T  0.001943  0.002414  0.001272\n",
       "13  chr3:132528316:T:G  0.004076  0.003213  0.012720\n",
       "14  chr3:132538225:G:A  0.000388  0.000000  0.000000"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_freq = pd.merge(freq, freq_cc, on=\"SNP\", how=\"left\")\n",
    "all_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>0.001553</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>0.002911</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.111600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.000971</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.330300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>0.000583</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "      <td>0.000389</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.574200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.015330</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "      <td>T</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "      <td>C</td>\n",
       "      <td>0.621500</td>\n",
       "      <td>0.8605</td>\n",
       "      <td>0.47370</td>\n",
       "      <td>1.5630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.006599</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>A</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>G</td>\n",
       "      <td>0.055190</td>\n",
       "      <td>5.7160</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>42.8900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.491700</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.857300</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.83930</td>\n",
       "      <td>1.1570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.027170</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "      <td>T</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "      <td>C</td>\n",
       "      <td>0.925300</td>\n",
       "      <td>0.9764</td>\n",
       "      <td>0.59310</td>\n",
       "      <td>1.6080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "      <td>0.005241</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "      <td>A</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "      <td>G</td>\n",
       "      <td>0.860000</td>\n",
       "      <td>1.1050</td>\n",
       "      <td>0.36280</td>\n",
       "      <td>3.3680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.095690</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.044060</td>\n",
       "      <td>0.7649</td>\n",
       "      <td>0.58890</td>\n",
       "      <td>0.9935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "      <td>0.001943</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>T</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>G</td>\n",
       "      <td>0.546100</td>\n",
       "      <td>1.8990</td>\n",
       "      <td>0.22830</td>\n",
       "      <td>15.8000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>0.004076</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001664</td>\n",
       "      <td>0.2501</td>\n",
       "      <td>0.09837</td>\n",
       "      <td>0.6360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP       MAF     MAF_A     MAF_U A1       F_A       F_U  \\\n",
       "0   chr3:132434618:A:G  0.001553  0.000803  0.000000  G  0.000803  0.000000   \n",
       "1   chr3:132447465:G:A  0.002911  0.003213  0.000000  A  0.003213  0.000000   \n",
       "2   chr3:132475009:C:A  0.000971  0.001205  0.000000  A  0.001205  0.000000   \n",
       "3   chr3:132492497:A:G  0.000583  0.000803  0.000000  G  0.000803  0.000000   \n",
       "4   chr3:132494190:A:G  0.000388  0.000803  0.000000  G  0.000803  0.000000   \n",
       "5   chr3:132499215:T:A  0.000389  0.000403  0.000000  A  0.000403  0.000000   \n",
       "6   chr3:132499231:C:T  0.015330  0.016470  0.019080  T  0.016470  0.019080   \n",
       "7   chr3:132499777:G:A  0.006599  0.007229  0.001272  A  0.007229  0.001272   \n",
       "8   chr3:132499779:G:T  0.491700  0.482300  0.486000  T  0.482300  0.486000   \n",
       "9   chr3:132502295:C:T  0.027170  0.026100  0.026720  T  0.026100  0.026720   \n",
       "10  chr3:132502299:G:A  0.005241  0.005622  0.005089  A  0.005622  0.005089   \n",
       "11  chr3:132507256:A:G  0.095690  0.087950  0.112000  G  0.087950  0.112000   \n",
       "12  chr3:132525718:G:T  0.001943  0.002414  0.001272  T  0.002414  0.001272   \n",
       "13  chr3:132528316:T:G  0.004076  0.003213  0.012720  G  0.003213  0.012720   \n",
       "14  chr3:132538225:G:A  0.000388  0.000000  0.000000  A  0.000000  0.000000   \n",
       "\n",
       "   A2         P      OR      L95      U95  \n",
       "0   A  0.426700     NaN      NaN      NaN  \n",
       "1   G  0.111600     NaN      NaN      NaN  \n",
       "2   C  0.330300     NaN      NaN      NaN  \n",
       "3   A  0.426700     NaN      NaN      NaN  \n",
       "4   A  0.426700     NaN      NaN      NaN  \n",
       "5   T  0.574200     NaN      NaN      NaN  \n",
       "6   C  0.621500  0.8605  0.47370   1.5630  \n",
       "7   G  0.055190  5.7160  0.76190  42.8900  \n",
       "8   G  0.857300  0.9854  0.83930   1.1570  \n",
       "9   C  0.925300  0.9764  0.59310   1.6080  \n",
       "10  G  0.860000  1.1050  0.36280   3.3680  \n",
       "11  A  0.044060  0.7649  0.58890   0.9935  \n",
       "12  G  0.546100  1.8990  0.22830  15.8000  \n",
       "13  T  0.001664  0.2501  0.09837   0.6360  \n",
       "14  G       NaN     NaN      NaN      NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_and_freq = pd.merge(all_freq, assoc, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>0.001553</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>0.4267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>0.002911</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.000971</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>0.3303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>0.000583</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>0.4267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>0.4267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF     MAF_A  MAF_U A1       F_A  F_U A2       P  \\\n",
       "0  chr3:132434618:A:G  0.001553  0.000803    0.0  G  0.000803  0.0  A  0.4267   \n",
       "1  chr3:132447465:G:A  0.002911  0.003213    0.0  A  0.003213  0.0  G  0.1116   \n",
       "2  chr3:132475009:C:A  0.000971  0.001205    0.0  A  0.001205  0.0  C  0.3303   \n",
       "3  chr3:132492497:A:G  0.000583  0.000803    0.0  G  0.000803  0.0  A  0.4267   \n",
       "4  chr3:132494190:A:G  0.000388  0.000803    0.0  G  0.000803  0.0  A  0.4267   \n",
       "\n",
       "   OR  L95  U95  BONF  \n",
       "0 NaN  NaN  NaN   1.0  \n",
       "1 NaN  NaN  NaN   1.0  \n",
       "2 NaN  NaN  NaN   1.0  \n",
       "3 NaN  NaN  NaN   1.0  \n",
       "4 NaN  NaN  NaN   1.0  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the results\n",
    "assoc_and_freq = pd.merge(assoc_and_freq, assoc_adjusted, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_AJ_R7/chr3_AJ_release7\n",
      "  --extract range 202407_BURDEN_AJ_R7/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_AJ_R7/AJ.samplestoKeep\n",
      "  --out 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn\n",
      "  --recode A\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1681997 variants loaded from .bim file.\n",
      "2577 people (1590 males, 987 females) loaded from .fam.\n",
      "1639 phenotype values loaded from .fam.\n",
      "--extract range: 1681982 variants excluded.\n",
      "--extract range: 15 variants remaining.\n",
      "--keep: 2576 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2576 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999664.\n",
      "15 variants and 2576 people pass filters and QC.\n",
      "Among remaining phenotypes, 1245 are cases and 393 are controls.  (938\n",
      "phenotypes are missing.)\n",
      "--recode A to 202407_BURDEN_AJ_R7/AJ_DNAJC13.coding_nonsyn.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in AJ for DNAJC13: 15\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr3:132434618:A:G_G',\n",
       " 'chr3:132447465:G:A_A',\n",
       " 'chr3:132475009:C:A_A',\n",
       " 'chr3:132492497:A:G_G',\n",
       " 'chr3:132494190:A:G_G',\n",
       " 'chr3:132499215:T:A_A',\n",
       " 'chr3:132499231:C:T_T',\n",
       " 'chr3:132499777:G:A_A',\n",
       " 'chr3:132499779:G:T_T',\n",
       " 'chr3:132502295:C:T_T',\n",
       " 'chr3:132502299:G:A_A',\n",
       " 'chr3:132507256:A:G_G',\n",
       " 'chr3:132525718:G:T_T',\n",
       " 'chr3:132528316:T:G_G',\n",
       " 'chr3:132538225:G:A_A']"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for DNAJC13: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132499215:T:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499231:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132499777:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499779:G:T_T</td>\n",
       "      <td>290</td>\n",
       "      <td>621</td>\n",
       "      <td>1245</td>\n",
       "      <td>96</td>\n",
       "      <td>190</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132502295:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132502299:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G_G</td>\n",
       "      <td>9</td>\n",
       "      <td>201</td>\n",
       "      <td>1245</td>\n",
       "      <td>4</td>\n",
       "      <td>80</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132525718:G:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Variant  Hom Cases  Het Cases  Total Cases  Hom Controls  \\\n",
       "0   chr3:132434618:A:G_G          0          2         1245             0   \n",
       "1   chr3:132447465:G:A_A          0          8         1245             0   \n",
       "2   chr3:132475009:C:A_A          0          3         1245             0   \n",
       "3   chr3:132492497:A:G_G          0          2         1245             0   \n",
       "4   chr3:132494190:A:G_G          0          2         1245             0   \n",
       "5   chr3:132499215:T:A_A          0          1         1245             0   \n",
       "6   chr3:132499231:C:T_T          0         41         1245             0   \n",
       "7   chr3:132499777:G:A_A          0         18         1245             0   \n",
       "8   chr3:132499779:G:T_T        290        621         1245            96   \n",
       "9   chr3:132502295:C:T_T          0         65         1245             0   \n",
       "10  chr3:132502299:G:A_A          0         14         1245             0   \n",
       "11  chr3:132507256:A:G_G          9        201         1245             4   \n",
       "12  chr3:132525718:G:T_T          0          6         1245             0   \n",
       "13  chr3:132528316:T:G_G          0          8         1245             0   \n",
       "14  chr3:132538225:G:A_A          0          0         1245             0   \n",
       "\n",
       "    Het Controls  Total Controls                 SNP  \n",
       "0              0             393  chr3:132434618:A:G  \n",
       "1              0             393  chr3:132447465:G:A  \n",
       "2              0             393  chr3:132475009:C:A  \n",
       "3              0             393  chr3:132492497:A:G  \n",
       "4              0             393  chr3:132494190:A:G  \n",
       "5              0             393  chr3:132499215:T:A  \n",
       "6             15             393  chr3:132499231:C:T  \n",
       "7              1             393  chr3:132499777:G:A  \n",
       "8            190             393  chr3:132499779:G:T  \n",
       "9             21             393  chr3:132502295:C:T  \n",
       "10             4             393  chr3:132502299:G:A  \n",
       "11            80             393  chr3:132507256:A:G  \n",
       "12             1             393  chr3:132525718:G:T  \n",
       "13            10             393  chr3:132528316:T:G  \n",
       "14             0             393  chr3:132538225:G:A  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "#    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "#    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "#        'Freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "#        'Freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Merge with assoc results \n",
    "full_results = pd.merge(assoc_and_freq, df_results, on=\"SNP\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_full_results = full_results[['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR','L95', 'U95', 'BONF', \n",
    "                                   'MAF', 'MAF_A', 'MAF_U', \n",
    "                                   'Hom Cases', 'Het Cases', 'Total Cases', \n",
    "                                   'Hom Controls','Het Controls', 'Total Controls']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15, 19)\n",
      "No. of SNPs: 15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132434618:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.001553</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132447465:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.111600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.002911</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.330300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000971</td>\n",
       "      <td>0.001205</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132492497:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000583</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.426700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132499215:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.574200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.000389</td>\n",
       "      <td>0.000403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "      <td>0.621500</td>\n",
       "      <td>0.8605</td>\n",
       "      <td>0.47370</td>\n",
       "      <td>1.5630</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.015330</td>\n",
       "      <td>0.016470</td>\n",
       "      <td>0.019080</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.055190</td>\n",
       "      <td>5.7160</td>\n",
       "      <td>0.76190</td>\n",
       "      <td>42.8900</td>\n",
       "      <td>0.7727</td>\n",
       "      <td>0.006599</td>\n",
       "      <td>0.007229</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "      <td>0.857300</td>\n",
       "      <td>0.9854</td>\n",
       "      <td>0.83930</td>\n",
       "      <td>1.1570</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.491700</td>\n",
       "      <td>0.482300</td>\n",
       "      <td>0.486000</td>\n",
       "      <td>290</td>\n",
       "      <td>621</td>\n",
       "      <td>1245</td>\n",
       "      <td>96</td>\n",
       "      <td>190</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "      <td>0.925300</td>\n",
       "      <td>0.9764</td>\n",
       "      <td>0.59310</td>\n",
       "      <td>1.6080</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.027170</td>\n",
       "      <td>0.026100</td>\n",
       "      <td>0.026720</td>\n",
       "      <td>0</td>\n",
       "      <td>65</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132502299:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "      <td>0.860000</td>\n",
       "      <td>1.1050</td>\n",
       "      <td>0.36280</td>\n",
       "      <td>3.3680</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.005241</td>\n",
       "      <td>0.005622</td>\n",
       "      <td>0.005089</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>0.044060</td>\n",
       "      <td>0.7649</td>\n",
       "      <td>0.58890</td>\n",
       "      <td>0.9935</td>\n",
       "      <td>0.6168</td>\n",
       "      <td>0.095690</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>9</td>\n",
       "      <td>201</td>\n",
       "      <td>1245</td>\n",
       "      <td>4</td>\n",
       "      <td>80</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132525718:G:T</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.546100</td>\n",
       "      <td>1.8990</td>\n",
       "      <td>0.22830</td>\n",
       "      <td>15.8000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.001943</td>\n",
       "      <td>0.002414</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>0.001664</td>\n",
       "      <td>0.2501</td>\n",
       "      <td>0.09837</td>\n",
       "      <td>0.6360</td>\n",
       "      <td>0.0233</td>\n",
       "      <td>0.004076</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.012720</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000388</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1 A2       F_A       F_U         P      OR      L95  \\\n",
       "0   chr3:132434618:A:G  G  A  0.000803  0.000000  0.426700     NaN      NaN   \n",
       "1   chr3:132447465:G:A  A  G  0.003213  0.000000  0.111600     NaN      NaN   \n",
       "2   chr3:132475009:C:A  A  C  0.001205  0.000000  0.330300     NaN      NaN   \n",
       "3   chr3:132492497:A:G  G  A  0.000803  0.000000  0.426700     NaN      NaN   \n",
       "4   chr3:132494190:A:G  G  A  0.000803  0.000000  0.426700     NaN      NaN   \n",
       "5   chr3:132499215:T:A  A  T  0.000403  0.000000  0.574200     NaN      NaN   \n",
       "6   chr3:132499231:C:T  T  C  0.016470  0.019080  0.621500  0.8605  0.47370   \n",
       "7   chr3:132499777:G:A  A  G  0.007229  0.001272  0.055190  5.7160  0.76190   \n",
       "8   chr3:132499779:G:T  T  G  0.482300  0.486000  0.857300  0.9854  0.83930   \n",
       "9   chr3:132502295:C:T  T  C  0.026100  0.026720  0.925300  0.9764  0.59310   \n",
       "10  chr3:132502299:G:A  A  G  0.005622  0.005089  0.860000  1.1050  0.36280   \n",
       "11  chr3:132507256:A:G  G  A  0.087950  0.112000  0.044060  0.7649  0.58890   \n",
       "12  chr3:132525718:G:T  T  G  0.002414  0.001272  0.546100  1.8990  0.22830   \n",
       "13  chr3:132528316:T:G  G  T  0.003213  0.012720  0.001664  0.2501  0.09837   \n",
       "14  chr3:132538225:G:A  A  G  0.000000  0.000000       NaN     NaN      NaN   \n",
       "\n",
       "        U95    BONF       MAF     MAF_A     MAF_U  Hom Cases  Het Cases  \\\n",
       "0       NaN  1.0000  0.001553  0.000803  0.000000          0          2   \n",
       "1       NaN  1.0000  0.002911  0.003213  0.000000          0          8   \n",
       "2       NaN  1.0000  0.000971  0.001205  0.000000          0          3   \n",
       "3       NaN  1.0000  0.000583  0.000803  0.000000          0          2   \n",
       "4       NaN  1.0000  0.000388  0.000803  0.000000          0          2   \n",
       "5       NaN  1.0000  0.000389  0.000403  0.000000          0          1   \n",
       "6    1.5630  1.0000  0.015330  0.016470  0.019080          0         41   \n",
       "7   42.8900  0.7727  0.006599  0.007229  0.001272          0         18   \n",
       "8    1.1570  1.0000  0.491700  0.482300  0.486000        290        621   \n",
       "9    1.6080  1.0000  0.027170  0.026100  0.026720          0         65   \n",
       "10   3.3680  1.0000  0.005241  0.005622  0.005089          0         14   \n",
       "11   0.9935  0.6168  0.095690  0.087950  0.112000          9        201   \n",
       "12  15.8000  1.0000  0.001943  0.002414  0.001272          0          6   \n",
       "13   0.6360  0.0233  0.004076  0.003213  0.012720          0          8   \n",
       "14      NaN     NaN  0.000388  0.000000  0.000000          0          0   \n",
       "\n",
       "    Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "0          1245             0             0             393  \n",
       "1          1245             0             0             393  \n",
       "2          1245             0             0             393  \n",
       "3          1245             0             0             393  \n",
       "4          1245             0             0             393  \n",
       "5          1245             0             0             393  \n",
       "6          1245             0            15             393  \n",
       "7          1245             0             1             393  \n",
       "8          1245            96           190             393  \n",
       "9          1245             0            21             393  \n",
       "10         1245             0             4             393  \n",
       "11         1245             4            80             393  \n",
       "12         1245             0             1             393  \n",
       "13         1245             0            10             393  \n",
       "14         1245             0             0             393  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clean_full_results.shape)\n",
    "print(f'No. of SNPs: {clean_full_results.shape[0]}')\n",
    "clean_full_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr3:132507256:A:G', 'chr3:132528316:T:G']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_freq = clean_full_results[clean_full_results['P']<0.05]\n",
    "sig_snps = sig_freq['SNP'].tolist()\n",
    "sig_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.11200</td>\n",
       "      <td>0.044060</td>\n",
       "      <td>0.7649</td>\n",
       "      <td>0.58890</td>\n",
       "      <td>0.9935</td>\n",
       "      <td>0.6168</td>\n",
       "      <td>0.095690</td>\n",
       "      <td>0.087950</td>\n",
       "      <td>0.11200</td>\n",
       "      <td>9</td>\n",
       "      <td>201</td>\n",
       "      <td>1245</td>\n",
       "      <td>4</td>\n",
       "      <td>80</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132528316:T:G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.01272</td>\n",
       "      <td>0.001664</td>\n",
       "      <td>0.2501</td>\n",
       "      <td>0.09837</td>\n",
       "      <td>0.6360</td>\n",
       "      <td>0.0233</td>\n",
       "      <td>0.004076</td>\n",
       "      <td>0.003213</td>\n",
       "      <td>0.01272</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1245</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1 A2       F_A      F_U         P      OR      L95  \\\n",
       "11  chr3:132507256:A:G  G  A  0.087950  0.11200  0.044060  0.7649  0.58890   \n",
       "13  chr3:132528316:T:G  G  T  0.003213  0.01272  0.001664  0.2501  0.09837   \n",
       "\n",
       "       U95    BONF       MAF     MAF_A    MAF_U  Hom Cases  Het Cases  \\\n",
       "11  0.9935  0.6168  0.095690  0.087950  0.11200          9        201   \n",
       "13  0.6360  0.0233  0.004076  0.003213  0.01272          0          8   \n",
       "\n",
       "    Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "11         1245             4            80             393  \n",
       "13         1245             0            10             393  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_df_results = clean_full_results[clean_full_results['SNP'].isin(sig_snps)]\n",
    "sig_df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\t132507256\tchr3:132507256:A:G\tA\tG\tTYPED;IMPUTED;AF=0.0960448;MAF=0.0960448;AVG_CS=1;R2=0.999996;ER2=0.998508\n",
      "3\t132528316\tchr3:132528316:T:G\tT\tG\tTYPED;IMPUTED;AF=0.00414313;MAF=0.00414313;AVG_CS=1;R2=1;ER2=1\n"
     ]
    }
   ],
   "source": [
    "for snp in sig_snps:\n",
    "    ! grep {snp} {WORK_DIR}/chr3_{ancestry}_release7.pvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save files to VM\n",
    "clean_full_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt', sep=\"\\t\", index=False)\n",
    "sig_df_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt' , sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
