{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Variants in DNAJC13 and their Association with Parkinson's Disease Across Different Ancestral Backgrounds\n",
    "\n",
    "* **Project**: gp2-hackathon-2023/Team 5 DNAJC13 Gene Assesment\n",
    "* **Last updated**: September 2024\n",
    "* **Version**: Python3.9\n",
    "* **Data**: GP2 Release 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "This notebook is focused on analysing association between DNAJC13 variants in East European (EAS) ancestry and Parkinson's disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_30958/3943855956.py:26: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## AMP-PD v2.5\n",
    "## Explicitly define release v2.5 path \n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2021_v2-5release_0510/wgs'\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_GATK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "print('AMP-PD v2.5')\n",
    "print(f'Path to AMP-PD v2.5 Clinical Data @ `AMP_CLINICAL_RELEASE_PATH`: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v2.5 WGS Data @ `AMP_WGS_RELEASE_PLINK_PATH`: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')\n",
    "\n",
    "## GP2 v7.0\n",
    "## Explicitly define release v7.0 path \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = 'EAS'\n",
    "WORK_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "WORKSPACE_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "\n",
    "! mkdir {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/plink2_linux_amd_avx2_20240704.zip\n",
    "unzip -o plink2_linux_amd_avx2_20240704.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR: Download resources for annotation\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rvtests is already installed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## chr 3\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_IMPUTED_GENO_PATH}/{ancestry}/chr3_{ancestry}_release7.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## master key\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_CLINICAL_RELEASE_PATH}/master_key_release7_final.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/{ancestry}_release7.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## PCs\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RAW_GENO_PATH}/{ancestry}/{ancestry}_release7.eigenvec {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# refFlat\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {WORKSPACE_BUCKET}/notebooks/misc/refFlat_HG38_all_chr.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_EAS_release7.log\tchr3_EAS_release7.pvar\tmaster_key_release7_final.csv\r\n",
      "chr3_EAS_release7.pgen\tEAS_release7.eigenvec\trefFlat_HG38_all_chr.txt\r\n",
      "chr3_EAS_release7.psam\tEAS_release7.related\r\n"
     ]
    }
   ],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Covariate File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(f'{WORK_DIR}/master_key_release7_final.csv')\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'race_for_qc', 'label']]\n",
    "key.rename(columns = {'GP2sampleID': 'IID',\n",
    "                      'baseline_GP2_phenotype':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'race_for_qc':'RACE',\n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = key[key['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals \n",
    "related_df = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.related')\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID1'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(ancestry_key.shape)\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (0/1)\n",
    "    # PD = 1; control = 0\n",
    "pheno_mapping = {\"PD\": 1, \"Control\": 0}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "1       2657\n",
       "0       2572\n",
       "<NA>      44\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get the PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.eigenvec', sep='\\t')\n",
    "\n",
    "# Split the single column into multiple columns (No longer needed with release 7)\n",
    "#pcs_split = pcs['#FID,IID,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10'].str.split(',', expand=True)\n",
    "\n",
    "selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "\n",
    "\n",
    "#pcs = pd.DataFrame(data=pcs_split.iloc[:, 1:7].values, columns=selected_columns)\n",
    "pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "\n",
    "# Drop the first row (since it's now the column names)\n",
    "pcs = pcs.drop(0)\n",
    "\n",
    "# Reset the index to remove any potential issues\n",
    "pcs = pcs.reset_index(drop=True)\n",
    "\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "print(pcs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert sex to binary (0/1)\n",
    "    # Female = 1; male = 0\n",
    "sex_mapping = {\"Female\": 1, \"Male\": 0}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "0       3388\n",
       "1       1879\n",
       "<NA>       6\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX\n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
       "       'AAO', 'RACE', 'label', 'PHENO'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Make covariate file\n",
    "df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['FID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = df[['FID','IID', 'SEX', 'PHENO', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO  SEX\n",
       "0      0      1670\n",
       "       1       675\n",
       "1      0      1434\n",
       "       1      1112\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_EAS_release7.log\tEAS_covariate_file.txt\tmaster_key_release7_final.csv\r\n",
      "chr3_EAS_release7.pgen\tEAS_release7.eigenvec\trefFlat_HG38_all_chr.txt\r\n",
      "chr3_EAS_release7.psam\tEAS_release7.related\r\n",
      "chr3_EAS_release7.pvar\tEAS.samplestoKeep\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}//')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation using ANNOVAR\n",
    "\n",
    "* *DNAJC13* from NCBI gene\n",
    "* hg38 (chr3:132417502-132539032)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --export vcf id-paste=iid\n",
      "  --from-bp 132417502\n",
      "  --mac 2\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13\n",
      "  --pfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Wed Jul 10 00:54:40 2024\n",
      "7450 MiB RAM detected, ~4653 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "5136 samples (1868 females, 3268 males; 5136 founders) loaded from\n",
      "202407_BURDEN_EAS_R7/chr3_EAS_release7.psam.\n",
      "3353374 variants loaded from 202407_BURDEN_EAS_R7/chr3_EAS_release7.pvar.\n",
      "1 binary phenotype loaded (2644 cases, 2452 controls).\n",
      "Calculating allele frequencies... done.\n",
      "562 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1179 variants remaining after main filters.\n",
      "--export vcf to 202407_BURDEN_EAS_R7/EAS_DNAJC13.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Wed Jul 10 00:54:46 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 202407_BURDEN_EAS_R7/EAS_DNAJC13.vcf.gz > 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.avinput>\n",
      "NOTICE: Finished reading 1198 lines from VCF file\n",
      "NOTICE: A total of 1179 locus in VCF file passed QC threshold, representing 1067 SNPs (740 transitions and 327 transversions) and 112 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 5480112 SNP genotypes (3800640 transitions and 1679472 transversions) and 575232 indels/substitutions for 5136 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.refGene -exonsort -nofirstcodondel 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.refGene.variant_function, 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1179 unique variants in 1179 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1179 unique variants in 1179 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 6\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_DNAJC13.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5139</th>\n",
       "      <th>Otherinfo5140</th>\n",
       "      <th>Otherinfo5141</th>\n",
       "      <th>Otherinfo5142</th>\n",
       "      <th>Otherinfo5143</th>\n",
       "      <th>Otherinfo5144</th>\n",
       "      <th>Otherinfo5145</th>\n",
       "      <th>Otherinfo5146</th>\n",
       "      <th>Otherinfo5147</th>\n",
       "      <th>Otherinfo5148</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132417545</td>\n",
       "      <td>132417545</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-17006G&gt;A;NM_015268:c.-17006G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132417611</td>\n",
       "      <td>132417611</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16940A&gt;T;NM_015268:c.-16940A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132417746</td>\n",
       "      <td>132417746</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16805C&gt;T;NM_015268:c.-16805C&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132417766</td>\n",
       "      <td>132417766</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132417820</td>\n",
       "      <td>132417820</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>3</td>\n",
       "      <td>132538575</td>\n",
       "      <td>132538575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*293A&gt;G;NM_015268:c.*293A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>3</td>\n",
       "      <td>132538598</td>\n",
       "      <td>132538598</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*316_*317insT;NM_015268:c.*316_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>3</td>\n",
       "      <td>132538679</td>\n",
       "      <td>132538679</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*397A&gt;T;NM_015268:c.*397A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1177</th>\n",
       "      <td>3</td>\n",
       "      <td>132538755</td>\n",
       "      <td>132538755</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*473C&gt;A;NM_015268:c.*473C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>3</td>\n",
       "      <td>132538903</td>\n",
       "      <td>132538903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*621G&gt;A;NM_015268:c.*621G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1179 rows Ã— 5159 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       3  132417545  132417545   G   A         UTR5      DNAJC13   \n",
       "1       3  132417611  132417611   A   T         UTR5      DNAJC13   \n",
       "2       3  132417746  132417746   C   T         UTR5      DNAJC13   \n",
       "3       3  132417766  132417766   G   A     intronic      DNAJC13   \n",
       "4       3  132417820  132417820   G   -     intronic      DNAJC13   \n",
       "...   ...        ...        ...  ..  ..          ...          ...   \n",
       "1174    3  132538575  132538575   A   G         UTR3      DNAJC13   \n",
       "1175    3  132538598  132538598   -   T         UTR3      DNAJC13   \n",
       "1176    3  132538679  132538679   A   T         UTR3      DNAJC13   \n",
       "1177    3  132538755  132538755   C   A         UTR3      DNAJC13   \n",
       "1178    3  132538903  132538903   G   A         UTR3      DNAJC13   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0        NM_001329126:c.-17006G>A;NM_015268:c.-17006G>A                  .   \n",
       "1        NM_001329126:c.-16940A>T;NM_015268:c.-16940A>T                  .   \n",
       "2        NM_001329126:c.-16805C>T;NM_015268:c.-16805C>T                  .   \n",
       "3                                                     .                  .   \n",
       "4                                                     .                  .   \n",
       "...                                                 ...                ...   \n",
       "1174         NM_001329126:c.*293A>G;NM_015268:c.*293A>G                  .   \n",
       "1175  NM_001329126:c.*316_*317insT;NM_015268:c.*316_...                  .   \n",
       "1176         NM_001329126:c.*397A>T;NM_015268:c.*397A>T                  .   \n",
       "1177         NM_001329126:c.*473C>A;NM_015268:c.*473C>A                  .   \n",
       "1178         NM_001329126:c.*621G>A;NM_015268:c.*621G>A                  .   \n",
       "\n",
       "     AAChange.refGene  ... Otherinfo5139  Otherinfo5140 Otherinfo5141  \\\n",
       "0                   .  ...           0|0            0|0           0|0   \n",
       "1                   .  ...           0|0            0|0           0|0   \n",
       "2                   .  ...           0|0            0|0           0|0   \n",
       "3                   .  ...           0|0            0|0           0|0   \n",
       "4                   .  ...           0|0            0|0           0|0   \n",
       "...               ...  ...           ...            ...           ...   \n",
       "1174                .  ...           0|0            0|1           1|1   \n",
       "1175                .  ...           0|0            0|0           0|0   \n",
       "1176                .  ...           0|0            0|0           0|0   \n",
       "1177                .  ...           0|0            0|0           0|0   \n",
       "1178                .  ...           1|0            0|0           0|0   \n",
       "\n",
       "     Otherinfo5142  Otherinfo5143  Otherinfo5144 Otherinfo5145 Otherinfo5146  \\\n",
       "0              0|0            0|0            0|0           0|0           0|0   \n",
       "1              0|0            0|0            0|0           0|0           0|0   \n",
       "2              0|0            0|0            0|0           0|0           0|0   \n",
       "3              0|0            0|0            0|0           0|0           0|0   \n",
       "4              0|0            0|0            0|0           0|0           0|0   \n",
       "...            ...            ...            ...           ...           ...   \n",
       "1174           0|0            0|0            0|0           0|0           0|0   \n",
       "1175           0|0            0|0            0|0           0|0           0|0   \n",
       "1176           0|0            0|0            0|0           0|0           0|0   \n",
       "1177           0|0            0|0            0|0           0|0           0|0   \n",
       "1178           0|0            1|0            0|1           0|1           0|1   \n",
       "\n",
       "     Otherinfo5147 Otherinfo5148  \n",
       "0              0|0           0|0  \n",
       "1              0|0           0|0  \n",
       "2              0|0           0|0  \n",
       "3              0|0           0|0  \n",
       "4              0|0           0|0  \n",
       "...            ...           ...  \n",
       "1174           0|1           0|0  \n",
       "1175           0|0           0|0  \n",
       "1176           0|0           0|0  \n",
       "1177           0|0           0|0  \n",
       "1178           0|0           0|1  \n",
       "\n",
       "[1179 rows x 5159 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter intronic\n",
    "intronic = gene[(gene['Func.refGene'] == 'intronic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR3\n",
    "utr3 = gene[(gene['Func.refGene'] == 'UTR3')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR5\n",
    "utr5 = gene[(gene['Func.refGene'] == 'UTR5')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter exonic and synonymous variants\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5139</th>\n",
       "      <th>Otherinfo5140</th>\n",
       "      <th>Otherinfo5141</th>\n",
       "      <th>Otherinfo5142</th>\n",
       "      <th>Otherinfo5143</th>\n",
       "      <th>Otherinfo5144</th>\n",
       "      <th>Otherinfo5145</th>\n",
       "      <th>Otherinfo5146</th>\n",
       "      <th>Otherinfo5147</th>\n",
       "      <th>Otherinfo5148</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>3</td>\n",
       "      <td>132454147</td>\n",
       "      <td>132454147</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon9:c.T922A:p.S308T,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>3</td>\n",
       "      <td>132457349</td>\n",
       "      <td>132457349</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon13:c.T1430C:p.M477T,D...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>3</td>\n",
       "      <td>132477807</td>\n",
       "      <td>132477807</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon23:c.G2464A:p.A822T,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>3</td>\n",
       "      <td>132484610</td>\n",
       "      <td>132484610</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon29:c.C3205T:p.R1069W,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>703</th>\n",
       "      <td>3</td>\n",
       "      <td>132492604</td>\n",
       "      <td>132492604</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon33:c.A3814G:p.I1272V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>3</td>\n",
       "      <td>132495159</td>\n",
       "      <td>132495159</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon35:c.A4013G:p.E1338G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>743</th>\n",
       "      <td>3</td>\n",
       "      <td>132496652</td>\n",
       "      <td>132496652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon36:c.G4145A:p.R1382H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>774</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>780</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>781</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>3</td>\n",
       "      <td>132502298</td>\n",
       "      <td>132502298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4546T:p.R1516C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>3</td>\n",
       "      <td>132503217</td>\n",
       "      <td>132503217</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon41:c.G4720A:p.V1574I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>893</th>\n",
       "      <td>3</td>\n",
       "      <td>132511239</td>\n",
       "      <td>132511239</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon44:c.A5288G:p.N1763S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>945</th>\n",
       "      <td>3</td>\n",
       "      <td>132516743</td>\n",
       "      <td>132516743</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon48:c.C5600T:p.T1867I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>3</td>\n",
       "      <td>132516789</td>\n",
       "      <td>132516789</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon48:c.G5646C:p.M1882I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>3</td>\n",
       "      <td>132525761</td>\n",
       "      <td>132525761</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon52:c.G6212A:p.R2071Q,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1168</th>\n",
       "      <td>3</td>\n",
       "      <td>132538179</td>\n",
       "      <td>132538179</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.C6629T:p.P2210L,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1170</th>\n",
       "      <td>3</td>\n",
       "      <td>132538272</td>\n",
       "      <td>132538272</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.A6722G:p.Y2241C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22 rows Ã— 5159 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "371     3  132454147  132454147   T   A       exonic      DNAJC13   \n",
       "398     3  132457349  132457349   T   C       exonic      DNAJC13   \n",
       "571     3  132477807  132477807   G   A       exonic      DNAJC13   \n",
       "629     3  132484610  132484610   C   T       exonic      DNAJC13   \n",
       "703     3  132492604  132492604   A   G       exonic      DNAJC13   \n",
       "717     3  132494190  132494190   A   G       exonic      DNAJC13   \n",
       "726     3  132495159  132495159   A   G       exonic      DNAJC13   \n",
       "743     3  132496652  132496652   G   A       exonic      DNAJC13   \n",
       "774     3  132499231  132499231   C   T       exonic      DNAJC13   \n",
       "780     3  132499777  132499777   G   A       exonic      DNAJC13   \n",
       "781     3  132499779  132499779   G   T       exonic      DNAJC13   \n",
       "805     3  132502295  132502295   C   T       exonic      DNAJC13   \n",
       "806     3  132502298  132502298   C   T       exonic      DNAJC13   \n",
       "810     3  132503217  132503217   G   A       exonic      DNAJC13   \n",
       "846     3  132507256  132507256   A   G       exonic      DNAJC13   \n",
       "893     3  132511239  132511239   A   G       exonic      DNAJC13   \n",
       "945     3  132516743  132516743   C   T       exonic      DNAJC13   \n",
       "946     3  132516789  132516789   G   C       exonic      DNAJC13   \n",
       "1037    3  132525761  132525761   G   A       exonic      DNAJC13   \n",
       "1168    3  132538179  132538179   C   T       exonic      DNAJC13   \n",
       "1169    3  132538225  132538225   G   A       exonic      DNAJC13   \n",
       "1170    3  132538272  132538272   A   G       exonic      DNAJC13   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "371                   .  nonsynonymous SNV   \n",
       "398                   .  nonsynonymous SNV   \n",
       "571                   .  nonsynonymous SNV   \n",
       "629                   .  nonsynonymous SNV   \n",
       "703                   .  nonsynonymous SNV   \n",
       "717                   .  nonsynonymous SNV   \n",
       "726                   .  nonsynonymous SNV   \n",
       "743                   .  nonsynonymous SNV   \n",
       "774                   .  nonsynonymous SNV   \n",
       "780                   .  nonsynonymous SNV   \n",
       "781                   .  nonsynonymous SNV   \n",
       "805                   .  nonsynonymous SNV   \n",
       "806                   .  nonsynonymous SNV   \n",
       "810                   .  nonsynonymous SNV   \n",
       "846                   .  nonsynonymous SNV   \n",
       "893                   .  nonsynonymous SNV   \n",
       "945                   .  nonsynonymous SNV   \n",
       "946                   .  nonsynonymous SNV   \n",
       "1037                  .  nonsynonymous SNV   \n",
       "1168                  .  nonsynonymous SNV   \n",
       "1169                  .  nonsynonymous SNV   \n",
       "1170                  .  nonsynonymous SNV   \n",
       "\n",
       "                                       AAChange.refGene  ... Otherinfo5139  \\\n",
       "371   DNAJC13:NM_001329126:exon9:c.T922A:p.S308T,DNA...  ...           0|0   \n",
       "398   DNAJC13:NM_001329126:exon13:c.T1430C:p.M477T,D...  ...           0|0   \n",
       "571   DNAJC13:NM_015268:exon23:c.G2464A:p.A822T,DNAJ...  ...           0|0   \n",
       "629   DNAJC13:NM_015268:exon29:c.C3205T:p.R1069W,DNA...  ...           0|0   \n",
       "703   DNAJC13:NM_015268:exon33:c.A3814G:p.I1272V,DNA...  ...           0|0   \n",
       "717   DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...  ...           0|0   \n",
       "726   DNAJC13:NM_015268:exon35:c.A4013G:p.E1338G,DNA...  ...           0|0   \n",
       "743   DNAJC13:NM_015268:exon36:c.G4145A:p.R1382H,DNA...  ...           0|0   \n",
       "774   DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...  ...           0|0   \n",
       "780   DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...  ...           0|0   \n",
       "781   DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...  ...           1|1   \n",
       "805   DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...  ...           0|0   \n",
       "806   DNAJC13:NM_015268:exon40:c.C4546T:p.R1516C,DNA...  ...           0|0   \n",
       "810   DNAJC13:NM_015268:exon41:c.G4720A:p.V1574I,DNA...  ...           0|0   \n",
       "846   DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...  ...           0|0   \n",
       "893   DNAJC13:NM_015268:exon44:c.A5288G:p.N1763S,DNA...  ...           0|0   \n",
       "945   DNAJC13:NM_015268:exon48:c.C5600T:p.T1867I,DNA...  ...           0|0   \n",
       "946   DNAJC13:NM_015268:exon48:c.G5646C:p.M1882I,DNA...  ...           0|0   \n",
       "1037  DNAJC13:NM_015268:exon52:c.G6212A:p.R2071Q,DNA...  ...           0|0   \n",
       "1168  DNAJC13:NM_015268:exon56:c.C6629T:p.P2210L,DNA...  ...           0|0   \n",
       "1169  DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...  ...           0|0   \n",
       "1170  DNAJC13:NM_015268:exon56:c.A6722G:p.Y2241C,DNA...  ...           0|0   \n",
       "\n",
       "      Otherinfo5140 Otherinfo5141 Otherinfo5142  Otherinfo5143  Otherinfo5144  \\\n",
       "371             0|0           0|0           0|0            0|0            0|0   \n",
       "398             0|0           0|0           0|0            0|0            0|0   \n",
       "571             0|0           0|0           0|0            0|0            0|0   \n",
       "629             0|0           0|0           0|0            0|0            0|0   \n",
       "703             0|0           0|0           0|0            0|0            0|0   \n",
       "717             0|0           0|0           0|0            0|0            0|0   \n",
       "726             0|0           0|0           0|0            0|0            0|0   \n",
       "743             0|0           0|0           0|0            0|0            0|0   \n",
       "774             0|0           0|0           0|0            0|0            0|0   \n",
       "780             0|0           0|0           0|0            0|0            0|0   \n",
       "781             1|1           1|1           1|0            1|1            1|1   \n",
       "805             0|0           0|0           0|0            0|0            0|0   \n",
       "806             0|0           0|0           0|0            0|0            0|0   \n",
       "810             0|0           0|0           0|0            0|0            0|0   \n",
       "846             0|0           0|0           0|0            0|0            0|0   \n",
       "893             0|0           0|0           0|0            0|0            0|0   \n",
       "945             0|0           0|0           0|0            0|0            0|0   \n",
       "946             0|0           0|0           0|0            0|0            0|0   \n",
       "1037            0|0           0|0           0|0            0|0            0|0   \n",
       "1168            0|0           0|0           0|0            0|0            0|0   \n",
       "1169            0|0           0|0           0|0            0|0            0|0   \n",
       "1170            0|0           0|0           0|0            0|0            0|0   \n",
       "\n",
       "     Otherinfo5145 Otherinfo5146 Otherinfo5147 Otherinfo5148  \n",
       "371            0|0           0|0           0|0           0|0  \n",
       "398            0|0           0|0           0|0           0|0  \n",
       "571            0|0           0|0           0|0           0|0  \n",
       "629            0|0           0|0           0|0           0|0  \n",
       "703            0|0           0|0           0|0           0|0  \n",
       "717            0|0           0|0           0|0           0|0  \n",
       "726            0|0           0|0           0|0           0|0  \n",
       "743            0|0           0|0           0|0           0|0  \n",
       "774            0|0           0|0           0|0           0|0  \n",
       "780            0|0           0|0           0|0           0|0  \n",
       "781            0|1           1|1           1|1           0|1  \n",
       "805            0|0           0|0           0|0           0|0  \n",
       "806            0|0           0|0           0|0           0|0  \n",
       "810            0|0           0|0           0|0           0|0  \n",
       "846            0|0           0|0           0|0           0|0  \n",
       "893            0|0           0|0           0|0           0|0  \n",
       "945            0|0           0|0           0|0           0|0  \n",
       "946            0|0           0|0           0|0           0|0  \n",
       "1037           0|0           0|0           0|0           0|0  \n",
       "1168           0|0           0|0           0|0           0|0  \n",
       "1169           0|0           0|0           0|0           0|0  \n",
       "1170           0|0           0|0           0|0           0|0  \n",
       "\n",
       "[22 rows x 5159 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "totalVariants = len(gene.axes[0])\n",
    "totalIntronic = len(intronic.axes[0])\n",
    "totalUTR3 = len(utr3.axes[0])\n",
    "totalUTR5 = len(utr5.axes[0])\n",
    "totalExonicSyn = len(coding_synonymous.axes[0])\n",
    "totalExonicNonSyn = len(coding_nonsynonymous.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Variants:  1179\n",
      "Intronic: 1140\n",
      "UTR3: 8\n",
      "UTR5: 3\n",
      "Exonic Syn: 6\n",
      "Exonic NonSyn: 22\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Variants: \", totalVariants)\n",
    "print(\"Intronic:\", totalIntronic)\n",
    "print(\"UTR3:\", totalUTR3)\n",
    "print(\"UTR5:\", totalUTR5)\n",
    "print(\"Exonic Syn:\", totalExonicSyn)\n",
    "print(\"Exonic NonSyn:\", totalExonicNonSyn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>3</td>\n",
       "      <td>132454147</td>\n",
       "      <td>132454147</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>3</td>\n",
       "      <td>132457349</td>\n",
       "      <td>132457349</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>3</td>\n",
       "      <td>132477807</td>\n",
       "      <td>132477807</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>3</td>\n",
       "      <td>132484610</td>\n",
       "      <td>132484610</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>703</th>\n",
       "      <td>3</td>\n",
       "      <td>132492604</td>\n",
       "      <td>132492604</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>3</td>\n",
       "      <td>132495159</td>\n",
       "      <td>132495159</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>743</th>\n",
       "      <td>3</td>\n",
       "      <td>132496652</td>\n",
       "      <td>132496652</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>774</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>780</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>781</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>3</td>\n",
       "      <td>132502298</td>\n",
       "      <td>132502298</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>3</td>\n",
       "      <td>132503217</td>\n",
       "      <td>132503217</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>893</th>\n",
       "      <td>3</td>\n",
       "      <td>132511239</td>\n",
       "      <td>132511239</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>945</th>\n",
       "      <td>3</td>\n",
       "      <td>132516743</td>\n",
       "      <td>132516743</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>3</td>\n",
       "      <td>132516789</td>\n",
       "      <td>132516789</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>3</td>\n",
       "      <td>132525761</td>\n",
       "      <td>132525761</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1168</th>\n",
       "      <td>3</td>\n",
       "      <td>132538179</td>\n",
       "      <td>132538179</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1170</th>\n",
       "      <td>3</td>\n",
       "      <td>132538272</td>\n",
       "      <td>132538272</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Gene.refGene\n",
       "371     3  132454147  132454147      DNAJC13\n",
       "398     3  132457349  132457349      DNAJC13\n",
       "571     3  132477807  132477807      DNAJC13\n",
       "629     3  132484610  132484610      DNAJC13\n",
       "703     3  132492604  132492604      DNAJC13\n",
       "717     3  132494190  132494190      DNAJC13\n",
       "726     3  132495159  132495159      DNAJC13\n",
       "743     3  132496652  132496652      DNAJC13\n",
       "774     3  132499231  132499231      DNAJC13\n",
       "780     3  132499777  132499777      DNAJC13\n",
       "781     3  132499779  132499779      DNAJC13\n",
       "805     3  132502295  132502295      DNAJC13\n",
       "806     3  132502298  132502298      DNAJC13\n",
       "810     3  132503217  132503217      DNAJC13\n",
       "846     3  132507256  132507256      DNAJC13\n",
       "893     3  132511239  132511239      DNAJC13\n",
       "945     3  132516743  132516743      DNAJC13\n",
       "946     3  132516789  132516789      DNAJC13\n",
       "1037    3  132525761  132525761      DNAJC13\n",
       "1168    3  132538179  132538179      DNAJC13\n",
       "1169    3  132538225  132538225      DNAJC13\n",
       "1170    3  132538272  132538272      DNAJC13"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_EAS_release7.log\t\t\t      EAS_DNAJC13.log\r\n",
      "chr3_EAS_release7.pgen\t\t\t      EAS_DNAJC13.vcf.gz\r\n",
      "chr3_EAS_release7.psam\t\t\t      EAS_DNAJC13.vcf.gz.tbi\r\n",
      "chr3_EAS_release7.pvar\t\t\t      EAS_release7.eigenvec\r\n",
      "EAS_covariate_file.txt\t\t\t      EAS_release7.related\r\n",
      "EAS_DNAJC13.all_coding_nonsyn.variantstoKeep  EAS.samplestoKeep\r\n",
      "EAS_DNAJC13.annovar.avinput\t\t      master_key_release7_final.csv\r\n",
      "EAS_DNAJC13.annovar.hg38_multianno.txt\t      refFlat_HG38_all_chr.txt\r\n",
      "EAS_DNAJC13.annovar.hg38_multianno.vcf\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep {WORKSPACE_BUCKET}//{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/chr3_EAS_release7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --pfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "\n",
      "Start time: Wed Jul 10 00:56:04 2024\n",
      "7450 MiB RAM detected, ~4495 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "5136 samples (1868 females, 3268 males; 5136 founders) loaded from\n",
      "202407_BURDEN_EAS_R7/chr3_EAS_release7.psam.\n",
      "3353374 variants loaded from 202407_BURDEN_EAS_R7/chr3_EAS_release7.pvar.\n",
      "1 binary phenotype loaded (2644 cases, 2452 controls).\n",
      "Writing 202407_BURDEN_EAS_R7/chr3_EAS_release7.fam ... done.\n",
      "Writing 202407_BURDEN_EAS_R7/chr3_EAS_release7.bim ... done.\n",
      "Writing 202407_BURDEN_EAS_R7/chr3_EAS_release7.bed ... 11131517192123252729313335373941424446485052545658606264666870727476788082848587899193959799done.\n",
      "End time: Wed Jul 10 00:58:43 2024\n"
     ]
    }
   ],
   "source": [
    "# Convert the files from Plink 2.0 to Plink 1.9 format\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}_release7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding\n",
      "  --recode vcf-iid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "22 variants and 5135 people pass filters and QC.\n",
      "Among remaining phenotypes, 2643 are cases and 2452 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "--recode vcf-iid to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "    # later do based on class and frequency\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--recode vcf-iid \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [202407_BURDEN_EAS_R7/EAS_DNAJC13.coding.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [202407_BURDEN_EAS_R7/EAS_DNAJC13.burden.coding]\n",
      "                          --outputRaw\n",
      "       Specify Covariate: --covar [202407_BURDEN_EAS_R7/EAS_covariate_file.txt]\n",
      "                          --covar-name [SEX,PC1,PC2,PC3,PC4], --sex\n",
      "       Specify Phenotype: --pheno [202407_BURDEN_EAS_R7/EAS_covariate_file.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype\n",
      "                          --mpheno [4], --pheno-name [], --qtl\n",
      "                          --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skat\n",
      "                         skato], --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [202407_BURDEN_EAS_R7/refFlat_HG38_all_chr.txt]\n",
      "                          --gene [DNAJC13], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding.vcf.gz\n",
      "    --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.burden.coding\n",
      "    --covar 202407_BURDEN_EAS_R7/EAS_covariate_file.txt\n",
      "    --covar-name SEX,PC1,PC2,PC3,PC4\n",
      "    --pheno 202407_BURDEN_EAS_R7/EAS_covariate_file.txt\n",
      "    --mpheno 4\n",
      "    --kernel skat,skato\n",
      "    --geneFile 202407_BURDEN_EAS_R7/refFlat_HG38_all_chr.txt\n",
      "    --gene DNAJC13\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Wed Jul 10 00:58:52 2024\n",
      "[INFO]\tLoaded [ 5135 ] samples from genotype files\n",
      "[WARN]\tSkip line 127 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 128 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 129 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 130 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 131 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 132 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 133 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 134 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 135 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 154 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 341 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 602 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 663 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 673 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 690 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 753 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 757 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 764 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 777 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 782 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 801 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 807 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 827 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 841 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 842 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 845 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 868 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 878 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 958 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 964 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 973 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 986 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 996 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1003 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1039 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1045 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1054 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1055 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1064 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1105 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1117 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1146 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1155 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1159 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1170 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1171 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1175 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1177 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1179 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1187 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1261 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1287 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1294 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1323 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1326 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1409 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1416 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1428 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1444 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1458 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1460 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1467 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1473 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1507 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1513 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1526 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1532 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1534 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1545 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1546 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1568 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1598 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1620 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1621 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1637 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1653 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1654 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1672 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1677 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1692 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1699 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1702 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1817 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1837 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1952 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 1998 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2002 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2006 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2009 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2010 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2011 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2015 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2017 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2019 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2021 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2023 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2025 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2027 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2029 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2031 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2033 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2035 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2037 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2041 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2042 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2044 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2045 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2046 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2047 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2048 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2050 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2051 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2053 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2054 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2055 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2056 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2057 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2059 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2061 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2065 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2067 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2072 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2074 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2076 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2078 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2080 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2082 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2084 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2086 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2090 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2094 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2095 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2097 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2098 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2100 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2101 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2103 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2104 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2382 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2482 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2484 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2485 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2497 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2499 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2516 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2526 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2563 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2693 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2709 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2782 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2795 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2815 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2818 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2822 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2829 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2833 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2836 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2841 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2842 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2846 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2848 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2862 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2865 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2867 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2876 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2879 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2885 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2888 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2906 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2910 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2920 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2932 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2937 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2954 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2972 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 2991 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3022 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3043 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3054 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3166 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3224 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3340 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3341 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3356 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3362 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3364 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3366 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3368 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3429 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3435 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3436 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3438 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3441 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3444 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3586 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3648 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3654 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3655 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3678 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3679 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3680 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3681 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3682 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3683 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3768 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3803 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3912 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3924 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3935 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 3992 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4105 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4220 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4221 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4222 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4223 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4224 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4225 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4226 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4227 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4228 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4229 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4230 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4231 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4232 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4233 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4234 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4235 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4236 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4237 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4264 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4475 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4543 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4680 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4782 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4783 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4817 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4826 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4841 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4845 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4852 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 4919 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 5006 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 5021 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[WARN]\tSkip line 5116 (short of columns) in phenotype file [ 202407_BURDEN_EAS_R7/EAS_covariate_file.txt ]\n",
      "[INFO]\tLoaded [ 4891 ] sample phenotypes\n",
      "[WARN]\tTotal [ 244 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ CORIELL_008702_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008775_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008810_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008867_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008922_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008940_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008942_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008999_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_009025_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ DODONA_000655_s1 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 234 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[WARN]\tTotal [ 244 ] samples are skipped from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008702_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008775_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008810_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008867_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008922_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008940_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008942_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_008999_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ CORIELL_009025_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ DODONA_000655_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip sample [ IPDGCEA-JU_000198_s1 ] from covariate file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 234 ] samples from covariate file with missing phenotypes\n",
      "[INFO]\tLoaded 1 male, 0 female and 4890 sex-unknown samples from 202407_BURDEN_EAS_R7/EAS_covariate_file.txt\n",
      "[INFO]\tAnalysis begins with [ 4891 ] samples...\n",
      "[INFO]\tSKAT test significance will be evaluated using 10000 permutations at alpha = 0.05 weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 22 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Wed Jul 10 00:58:53 2024\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "## RVtests with covariates \n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.burden.coding \\\n",
    "--kernel skat,skato \\\n",
    "--inVcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--mpheno 4 \\\n",
    "--gene DNAJC13 \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38_all_chr.txt \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4 #[WARN]Covariate [ PC5 ] has strong correlation [ r^2 = 1 ] with the response!\n",
    "\n",
    "\n",
    "# --out : Name of output \n",
    "# --burden cmc --kernel skato: tests to run \n",
    "# --inVcf : VCF file \n",
    "# --gene: gene name (if only looking at one or a few)\n",
    "# --geneFile refFlat.txt\n",
    "# --pheno :  covar file\n",
    "# --mpheno : # column that has phenotype information\n",
    "# --pheno-name : column name with phenotype in file\n",
    "# --covar : covar file\n",
    "# --freqUpper : optional, MAF cut-off\n",
    "# --covar-name : covariates, listed by column name, separated by commas (no spaces between commas)\n",
    "## 0=controls; 1=cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t4891\t22\t17\t0.113714\t0.999986\t10000\t1000\t0.113714\t1000\t0\t1\r\n"
     ]
    }
   ],
   "source": [
    "## look at results \n",
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t4891\t22\t17\t653.777\t1\t1\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_EAS_release7.bed\r\n",
      "chr3_EAS_release7.bim\r\n",
      "chr3_EAS_release7.fam\r\n",
      "chr3_EAS_release7.log\r\n",
      "chr3_EAS_release7.pgen\r\n",
      "chr3_EAS_release7.psam\r\n",
      "chr3_EAS_release7.pvar\r\n",
      "EAS_covariate_file.txt\r\n",
      "EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "EAS_DNAJC13.annovar.avinput\r\n",
      "EAS_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "EAS_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "EAS_DNAJC13.burden.coding.log\r\n",
      "EAS_DNAJC13.burden.coding.Skat.assoc\r\n",
      "EAS_DNAJC13.burden.coding.SkatO.assoc\r\n",
      "EAS_DNAJC13.coding.log\r\n",
      "EAS_DNAJC13.coding.vcf.gz\r\n",
      "EAS_DNAJC13.coding.vcf.gz.tbi\r\n",
      "EAS_DNAJC13.log\r\n",
      "EAS_DNAJC13.vcf.gz\r\n",
      "EAS_DNAJC13.vcf.gz.tbi\r\n",
      "EAS_release7.eigenvec\r\n",
      "EAS_release7.related\r\n",
      "EAS.samplestoKeep\r\n",
      "master_key_release7_final.csv\r\n",
      "refFlat_HG38_all_chr.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.*.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --ci 0.95\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "22 variants and 5135 people pass filters and QC.\n",
      "Among remaining phenotypes, 2643 are cases and 2452 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR           SE          L95          U95 \r\n",
      "   3   chr3:132454147:T:A  132454147    A        0 0.000204    T        1.078       0.2991            0          inf            0          nan \r\n",
      "   3   chr3:132457349:T:C  132457349    C        0 0.0002042    T        1.077       0.2993            0          inf            0          nan \r\n",
      "   3   chr3:132477807:G:A  132477807    A 0.0009509 0.001847    G        1.472        0.225       0.5143       0.5581       0.1722        1.536 \r\n",
      "   3   chr3:132484610:C:T  132484610    T        0 0.0002039    C        1.078       0.2992            0          inf            0          nan \r\n",
      "   3   chr3:132492604:A:G  132492604    G 0.0003784        0    A        1.856       0.1731           NA           NA           NA           NA \r\n",
      "   3   chr3:132494190:A:G  132494190    G 0.0003785        0    A        1.857        0.173           NA           NA           NA           NA \r\n",
      "   3   chr3:132495159:A:G  132495159    G        0        0    A           NA           NA           NA           NA           NA           NA \r\n",
      "   3   chr3:132496652:G:A  132496652    A        0        0    G           NA           NA           NA           NA           NA           NA \r\n",
      "   3   chr3:132499231:C:T  132499231    T        0 0.0004078    C        2.156        0.142            0          inf            0          nan \r\n",
      "   3   chr3:132499777:G:A  132499777    A 0.0001892  0.00102    G        2.981      0.08425       0.1854        1.096      0.02165        1.587 \r\n",
      "   3   chr3:132499779:G:T  132499779    G  0.09946   0.1525    T        58.93    1.635e-14       0.6138        0.064       0.5415       0.6959 \r\n",
      "   3   chr3:132502295:C:T  132502295    T 0.0003849 0.0004177    C     0.006694       0.9348       0.9214            1       0.1297        6.544 \r\n",
      "   3   chr3:132502298:C:T  132502298    T 0.0001893 0.0008167    C        2.037       0.1535       0.2317        1.118      0.02589        2.074 \r\n",
      "   3   chr3:132503217:G:A  132503217    A 0.0001895        0    G       0.9292       0.3351           NA           NA           NA           NA \r\n",
      "   3   chr3:132507256:A:G  132507256    G 0.0009463 0.001427    A       0.5005       0.4793       0.6626       0.5859       0.2102        2.089 \r\n",
      "   3   chr3:132511239:A:G  132511239    G 0.0001893        0    A       0.9247       0.3362           NA           NA           NA           NA \r\n",
      "   3   chr3:132516743:C:T  132516743    T        0        0    C           NA           NA           NA           NA           NA           NA \r\n",
      "   3   chr3:132516789:G:C  132516789    C 0.0005688 0.0002047    G       0.8547       0.3552         2.78        1.155       0.2891        26.74 \r\n",
      "   3   chr3:132525761:G:A  132525761    A 0.0001894 0.0002039    G     0.002731       0.9583       0.9288        1.414      0.05808        14.85 \r\n",
      "   3   chr3:132538179:C:T  132538179    T 0.0001894 0.0002041    C     0.002791       0.9579        0.928        1.414      0.05803        14.84 \r\n",
      "   3   chr3:132538225:G:A  132538225    A 0.001135 0.001224    G      0.01709        0.896       0.9273       0.5777       0.2989        2.877 \r\n",
      "   3   chr3:132538272:A:G  132538272    G        0        0    A           NA           NA           NA           NA           NA           NA \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>2.991000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>2.993000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>G</td>\n",
       "      <td>2.250000e-01</td>\n",
       "      <td>0.5143</td>\n",
       "      <td>0.17220</td>\n",
       "      <td>1.5360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>2.992000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.731000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.730000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132495159:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132496652:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>C</td>\n",
       "      <td>1.420000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>G</td>\n",
       "      <td>8.425000e-02</td>\n",
       "      <td>0.1854</td>\n",
       "      <td>0.02165</td>\n",
       "      <td>1.5870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "      <td>T</td>\n",
       "      <td>1.635000e-14</td>\n",
       "      <td>0.6138</td>\n",
       "      <td>0.54150</td>\n",
       "      <td>0.6959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>C</td>\n",
       "      <td>9.348000e-01</td>\n",
       "      <td>0.9214</td>\n",
       "      <td>0.12970</td>\n",
       "      <td>6.5440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>C</td>\n",
       "      <td>1.535000e-01</td>\n",
       "      <td>0.2317</td>\n",
       "      <td>0.02589</td>\n",
       "      <td>2.0740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>3.351000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "      <td>A</td>\n",
       "      <td>4.793000e-01</td>\n",
       "      <td>0.6626</td>\n",
       "      <td>0.21020</td>\n",
       "      <td>2.0890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>3.362000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132516743:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>G</td>\n",
       "      <td>3.552000e-01</td>\n",
       "      <td>2.7800</td>\n",
       "      <td>0.28910</td>\n",
       "      <td>26.7400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>9.583000e-01</td>\n",
       "      <td>0.9288</td>\n",
       "      <td>0.05808</td>\n",
       "      <td>14.8500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>9.579000e-01</td>\n",
       "      <td>0.9280</td>\n",
       "      <td>0.05803</td>\n",
       "      <td>14.8400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>G</td>\n",
       "      <td>8.960000e-01</td>\n",
       "      <td>0.9273</td>\n",
       "      <td>0.29890</td>\n",
       "      <td>2.8770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr3:132538272:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1       F_A       F_U A2             P      OR  \\\n",
       "0   chr3:132454147:T:A  A  0.000000  0.000204  T  2.991000e-01  0.0000   \n",
       "1   chr3:132457349:T:C  C  0.000000  0.000204  T  2.993000e-01  0.0000   \n",
       "2   chr3:132477807:G:A  A  0.000951  0.001847  G  2.250000e-01  0.5143   \n",
       "3   chr3:132484610:C:T  T  0.000000  0.000204  C  2.992000e-01  0.0000   \n",
       "4   chr3:132492604:A:G  G  0.000378  0.000000  A  1.731000e-01     NaN   \n",
       "5   chr3:132494190:A:G  G  0.000378  0.000000  A  1.730000e-01     NaN   \n",
       "6   chr3:132495159:A:G  G  0.000000  0.000000  A           NaN     NaN   \n",
       "7   chr3:132496652:G:A  A  0.000000  0.000000  G           NaN     NaN   \n",
       "8   chr3:132499231:C:T  T  0.000000  0.000408  C  1.420000e-01  0.0000   \n",
       "9   chr3:132499777:G:A  A  0.000189  0.001020  G  8.425000e-02  0.1854   \n",
       "10  chr3:132499779:G:T  G  0.099460  0.152500  T  1.635000e-14  0.6138   \n",
       "11  chr3:132502295:C:T  T  0.000385  0.000418  C  9.348000e-01  0.9214   \n",
       "12  chr3:132502298:C:T  T  0.000189  0.000817  C  1.535000e-01  0.2317   \n",
       "13  chr3:132503217:G:A  A  0.000189  0.000000  G  3.351000e-01     NaN   \n",
       "14  chr3:132507256:A:G  G  0.000946  0.001427  A  4.793000e-01  0.6626   \n",
       "15  chr3:132511239:A:G  G  0.000189  0.000000  A  3.362000e-01     NaN   \n",
       "16  chr3:132516743:C:T  T  0.000000  0.000000  C           NaN     NaN   \n",
       "17  chr3:132516789:G:C  C  0.000569  0.000205  G  3.552000e-01  2.7800   \n",
       "18  chr3:132525761:G:A  A  0.000189  0.000204  G  9.583000e-01  0.9288   \n",
       "19  chr3:132538179:C:T  T  0.000189  0.000204  C  9.579000e-01  0.9280   \n",
       "20  chr3:132538225:G:A  A  0.001135  0.001224  G  8.960000e-01  0.9273   \n",
       "21  chr3:132538272:A:G  G  0.000000  0.000000  A           NaN     NaN   \n",
       "\n",
       "        L95      U95  \n",
       "0   0.00000      NaN  \n",
       "1   0.00000      NaN  \n",
       "2   0.17220   1.5360  \n",
       "3   0.00000      NaN  \n",
       "4       NaN      NaN  \n",
       "5       NaN      NaN  \n",
       "6       NaN      NaN  \n",
       "7       NaN      NaN  \n",
       "8   0.00000      NaN  \n",
       "9   0.02165   1.5870  \n",
       "10  0.54150   0.6959  \n",
       "11  0.12970   6.5440  \n",
       "12  0.02589   2.0740  \n",
       "13      NaN      NaN  \n",
       "14  0.21020   2.0890  \n",
       "15      NaN      NaN  \n",
       "16      NaN      NaN  \n",
       "17  0.28910  26.7400  \n",
       "18  0.05808  14.8500  \n",
       "19  0.05803  14.8400  \n",
       "20  0.29890   2.8770  \n",
       "21      NaN      NaN  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc', delim_whitespace=True, usecols=['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR', 'L95', 'U95'])\n",
    "assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "22 variants and 5135 people pass filters and QC.\n",
      "Among remaining phenotypes, 2643 are cases and 2452 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 2.3628.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (18 variants) written to\n",
      "202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "## repeat association study applying multiple testing corrections for the raw p-values (like Bonferroni post-hoc)\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP      UNADJ         GC       BONF       HOLM   SIDAK_SS   SIDAK_SD     FDR_BH     FDR_BY\r\n",
      "   3   chr3:132499779:G:T  1.635e-14  5.913e-07  2.942e-13  2.942e-13  2.942e-13  2.942e-13  2.942e-13  1.028e-12 \r\n",
      "   3   chr3:132499777:G:A    0.08425     0.2613          1          1     0.7949      0.776     0.4919          1 \r\n",
      "   3   chr3:132499231:C:T      0.142     0.3394          1          1     0.9365     0.9137     0.4919          1 \r\n",
      "   3   chr3:132502298:C:T     0.1535     0.3531          1          1     0.9502     0.9179     0.4919          1 \r\n",
      "   3   chr3:132494190:A:G      0.173     0.3754          1          1     0.9673       0.93     0.4919          1 \r\n",
      "   3   chr3:132492604:A:G     0.1731     0.3755          1          1     0.9673       0.93     0.4919          1 \r\n",
      "   3   chr3:132477807:G:A      0.225     0.4299          1          1     0.9898     0.9531     0.4919          1 \r\n",
      "   3   chr3:132454147:T:A     0.2991     0.4994          1          1     0.9983       0.98     0.4919          1 \r\n",
      "   3   chr3:132484610:C:T     0.2992     0.4995          1          1     0.9983       0.98     0.4919          1 \r\n",
      "   3   chr3:132457349:T:C     0.2993     0.4995          1          1     0.9983       0.98     0.4919          1 \r\n",
      "   3   chr3:132503217:G:A     0.3351     0.5306          1          1     0.9994       0.98     0.4919          1 \r\n",
      "   3   chr3:132511239:A:G     0.3362     0.5316          1          1     0.9994       0.98     0.4919          1 \r\n",
      "   3   chr3:132516789:G:C     0.3552     0.5476          1          1     0.9996       0.98     0.4919          1 \r\n",
      "   3   chr3:132507256:A:G     0.4793     0.6453          1          1          1       0.98     0.6162          1 \r\n",
      "   3   chr3:132538225:G:A      0.896     0.9322          1          1          1     0.9999     0.9583          1 \r\n",
      "   3   chr3:132502295:C:T     0.9348     0.9576          1          1          1     0.9999     0.9583          1 \r\n",
      "   3   chr3:132538179:C:T     0.9579     0.9726          1          1          1     0.9999     0.9583          1 \r\n",
      "   3   chr3:132525761:G:A     0.9583     0.9729          1          1          1     0.9999     0.9583          1 \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>2.942000e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP          BONF\n",
       "0   chr3:132499779:G:T  2.942000e-13\n",
       "1   chr3:132499777:G:A  1.000000e+00\n",
       "2   chr3:132499231:C:T  1.000000e+00\n",
       "3   chr3:132502298:C:T  1.000000e+00\n",
       "4   chr3:132494190:A:G  1.000000e+00\n",
       "5   chr3:132492604:A:G  1.000000e+00\n",
       "6   chr3:132477807:G:A  1.000000e+00\n",
       "7   chr3:132454147:T:A  1.000000e+00\n",
       "8   chr3:132484610:C:T  1.000000e+00\n",
       "9   chr3:132457349:T:C  1.000000e+00\n",
       "10  chr3:132503217:G:A  1.000000e+00\n",
       "11  chr3:132511239:A:G  1.000000e+00\n",
       "12  chr3:132516789:G:C  1.000000e+00\n",
       "13  chr3:132507256:A:G  1.000000e+00\n",
       "14  chr3:132538225:G:A  1.000000e+00\n",
       "15  chr3:132502295:C:T  1.000000e+00\n",
       "16  chr3:132538179:C:T  1.000000e+00\n",
       "17  chr3:132525761:G:A  1.000000e+00"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_adjusted = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted', delim_whitespace=True, usecols=['SNP', 'BONF'])\n",
    "assoc_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.frq .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2          MAF  NCHROBS\r\n",
      "   3   chr3:132454147:T:A    A    T    9.741e-05    10266\r\n",
      "   3   chr3:132457349:T:C    C    T    9.752e-05    10254\r\n",
      "   3   chr3:132477807:G:A    A    G     0.001371    10210\r\n",
      "   3   chr3:132484610:C:T    T    C    9.739e-05    10268\r\n",
      "   3   chr3:132492604:A:G    G    A    0.0001947    10270\r\n",
      "   3   chr3:132494190:A:G    G    A    0.0001948    10268\r\n",
      "   3   chr3:132495159:A:G    G    A            0    10262\r\n",
      "   3   chr3:132496652:G:A    A    G            0    10254\r\n",
      "   3   chr3:132499231:C:T    T    C    0.0001947    10270\r\n",
      "   3   chr3:132499777:G:A    A    G    0.0005842    10270\r\n",
      "   3   chr3:132499779:G:T    G    T       0.1241     9242\r\n",
      "   3   chr3:132502295:C:T    T    C    0.0003975    10064\r\n",
      "   3   chr3:132502298:C:T    T    C    0.0004873    10260\r\n",
      "   3   chr3:132503217:G:A    A    G    9.748e-05    10258\r\n",
      "   3   chr3:132507256:A:G    G    A     0.001169    10268\r\n",
      "   3   chr3:132511239:A:G    G    A    9.762e-05    10244\r\n",
      "   3   chr3:132516743:C:T    T    C            0    10258\r\n",
      "   3   chr3:132516789:G:C    C    G    0.0003906    10240\r\n",
      "   3   chr3:132525761:G:A    A    G    0.0001949    10264\r\n",
      "   3   chr3:132538179:C:T    T    C    0.0002924    10260\r\n",
      "   3   chr3:132538225:G:A    A    G     0.001169    10268\r\n",
      "   3   chr3:132538272:A:G    G    A            0    10262\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>0.000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>0.000098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>0.001371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>0.000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF\n",
       "0  chr3:132454147:T:A  0.000097\n",
       "1  chr3:132457349:T:C  0.000098\n",
       "2  chr3:132477807:G:A  0.001371\n",
       "3  chr3:132484610:C:T  0.000097\n",
       "4  chr3:132492604:A:G  0.000195"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq', delim_whitespace=True, usecols=['SNP', 'MAF'])\n",
    "freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq case-control\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.frq.cc .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq case-control \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2        MAF_A        MAF_U  NCHROBS_A  NCHROBS_U\r\n",
      "   3   chr3:132454147:T:A    A    T            0     0.000204       5284       4902\r\n",
      "   3   chr3:132457349:T:C    C    T            0    0.0002042       5276       4898\r\n",
      "   3   chr3:132477807:G:A    A    G    0.0009509     0.001847       5258       4872\r\n",
      "   3   chr3:132484610:C:T    T    C            0    0.0002039       5284       4904\r\n",
      "   3   chr3:132492604:A:G    G    A    0.0003784            0       5286       4904\r\n",
      "   3   chr3:132494190:A:G    G    A    0.0003785            0       5284       4904\r\n",
      "   3   chr3:132495159:A:G    G    A            0            0       5280       4902\r\n",
      "   3   chr3:132496652:G:A    A    G            0            0       5274       4900\r\n",
      "   3   chr3:132499231:C:T    T    C            0    0.0004078       5286       4904\r\n",
      "   3   chr3:132499777:G:A    A    G    0.0001892      0.00102       5286       4904\r\n",
      "   3   chr3:132499779:G:T    G    T      0.09946       0.1525       4836       4328\r\n",
      "   3   chr3:132502295:C:T    T    C    0.0003849    0.0004177       5196       4788\r\n",
      "   3   chr3:132502298:C:T    T    C    0.0001893    0.0008167       5282       4898\r\n",
      "   3   chr3:132503217:G:A    A    G    0.0001895            0       5276       4902\r\n",
      "   3   chr3:132507256:A:G    G    A    0.0009463     0.001427       5284       4904\r\n",
      "   3   chr3:132511239:A:G    G    A    0.0001893            0       5282       4884\r\n",
      "   3   chr3:132516743:C:T    T    C            0            0       5276       4902\r\n",
      "   3   chr3:132516789:G:C    C    G    0.0005688    0.0002047       5274       4886\r\n",
      "   3   chr3:132525761:G:A    A    G    0.0001894    0.0002039       5280       4904\r\n",
      "   3   chr3:132538179:C:T    T    C    0.0001894    0.0002041       5280       4900\r\n",
      "   3   chr3:132538225:G:A    A    G     0.001135     0.001224       5286       4902\r\n",
      "   3   chr3:132538272:A:G    G    A            0            0       5280       4902\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP     MAF_A     MAF_U\n",
       "0  chr3:132454147:T:A  0.000000  0.000204\n",
       "1  chr3:132457349:T:C  0.000000  0.000204\n",
       "2  chr3:132477807:G:A  0.000951  0.001847\n",
       "3  chr3:132484610:C:T  0.000000  0.000204\n",
       "4  chr3:132492604:A:G  0.000378  0.000000"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq_cc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc', delim_whitespace=True, usecols=['SNP', 'MAF_A', 'MAF_U'])\n",
    "freq_cc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>0.001371</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132495159:A:G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132496652:G:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.000584</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.124100</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.000398</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "      <td>0.000487</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132516743:C:T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr3:132538272:A:G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP       MAF     MAF_A     MAF_U\n",
       "0   chr3:132454147:T:A  0.000097  0.000000  0.000204\n",
       "1   chr3:132457349:T:C  0.000098  0.000000  0.000204\n",
       "2   chr3:132477807:G:A  0.001371  0.000951  0.001847\n",
       "3   chr3:132484610:C:T  0.000097  0.000000  0.000204\n",
       "4   chr3:132492604:A:G  0.000195  0.000378  0.000000\n",
       "5   chr3:132494190:A:G  0.000195  0.000378  0.000000\n",
       "6   chr3:132495159:A:G  0.000000  0.000000  0.000000\n",
       "7   chr3:132496652:G:A  0.000000  0.000000  0.000000\n",
       "8   chr3:132499231:C:T  0.000195  0.000000  0.000408\n",
       "9   chr3:132499777:G:A  0.000584  0.000189  0.001020\n",
       "10  chr3:132499779:G:T  0.124100  0.099460  0.152500\n",
       "11  chr3:132502295:C:T  0.000398  0.000385  0.000418\n",
       "12  chr3:132502298:C:T  0.000487  0.000189  0.000817\n",
       "13  chr3:132503217:G:A  0.000097  0.000189  0.000000\n",
       "14  chr3:132507256:A:G  0.001169  0.000946  0.001427\n",
       "15  chr3:132511239:A:G  0.000098  0.000189  0.000000\n",
       "16  chr3:132516743:C:T  0.000000  0.000000  0.000000\n",
       "17  chr3:132516789:G:C  0.000391  0.000569  0.000205\n",
       "18  chr3:132525761:G:A  0.000195  0.000189  0.000204\n",
       "19  chr3:132538179:C:T  0.000292  0.000189  0.000204\n",
       "20  chr3:132538225:G:A  0.001169  0.001135  0.001224\n",
       "21  chr3:132538272:A:G  0.000000  0.000000  0.000000"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_freq = pd.merge(freq, freq_cc, on=\"SNP\", how=\"left\")\n",
    "all_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>2.991000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>2.993000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>0.001371</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>G</td>\n",
       "      <td>2.250000e-01</td>\n",
       "      <td>0.5143</td>\n",
       "      <td>0.17220</td>\n",
       "      <td>1.5360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>2.992000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.731000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>1.730000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132495159:A:G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132496652:G:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>C</td>\n",
       "      <td>1.420000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.000584</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>G</td>\n",
       "      <td>8.425000e-02</td>\n",
       "      <td>0.1854</td>\n",
       "      <td>0.02165</td>\n",
       "      <td>1.5870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.124100</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "      <td>G</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "      <td>T</td>\n",
       "      <td>1.635000e-14</td>\n",
       "      <td>0.6138</td>\n",
       "      <td>0.54150</td>\n",
       "      <td>0.6959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.000398</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>C</td>\n",
       "      <td>9.348000e-01</td>\n",
       "      <td>0.9214</td>\n",
       "      <td>0.12970</td>\n",
       "      <td>6.5440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "      <td>0.000487</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>C</td>\n",
       "      <td>1.535000e-01</td>\n",
       "      <td>0.2317</td>\n",
       "      <td>0.02589</td>\n",
       "      <td>2.0740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>3.351000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "      <td>A</td>\n",
       "      <td>4.793000e-01</td>\n",
       "      <td>0.6626</td>\n",
       "      <td>0.21020</td>\n",
       "      <td>2.0890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>3.362000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132516743:C:T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>G</td>\n",
       "      <td>3.552000e-01</td>\n",
       "      <td>2.7800</td>\n",
       "      <td>0.28910</td>\n",
       "      <td>26.7400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>G</td>\n",
       "      <td>9.583000e-01</td>\n",
       "      <td>0.9288</td>\n",
       "      <td>0.05808</td>\n",
       "      <td>14.8500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>9.579000e-01</td>\n",
       "      <td>0.9280</td>\n",
       "      <td>0.05803</td>\n",
       "      <td>14.8400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>G</td>\n",
       "      <td>8.960000e-01</td>\n",
       "      <td>0.9273</td>\n",
       "      <td>0.29890</td>\n",
       "      <td>2.8770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr3:132538272:A:G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP       MAF     MAF_A     MAF_U A1       F_A       F_U  \\\n",
       "0   chr3:132454147:T:A  0.000097  0.000000  0.000204  A  0.000000  0.000204   \n",
       "1   chr3:132457349:T:C  0.000098  0.000000  0.000204  C  0.000000  0.000204   \n",
       "2   chr3:132477807:G:A  0.001371  0.000951  0.001847  A  0.000951  0.001847   \n",
       "3   chr3:132484610:C:T  0.000097  0.000000  0.000204  T  0.000000  0.000204   \n",
       "4   chr3:132492604:A:G  0.000195  0.000378  0.000000  G  0.000378  0.000000   \n",
       "5   chr3:132494190:A:G  0.000195  0.000378  0.000000  G  0.000378  0.000000   \n",
       "6   chr3:132495159:A:G  0.000000  0.000000  0.000000  G  0.000000  0.000000   \n",
       "7   chr3:132496652:G:A  0.000000  0.000000  0.000000  A  0.000000  0.000000   \n",
       "8   chr3:132499231:C:T  0.000195  0.000000  0.000408  T  0.000000  0.000408   \n",
       "9   chr3:132499777:G:A  0.000584  0.000189  0.001020  A  0.000189  0.001020   \n",
       "10  chr3:132499779:G:T  0.124100  0.099460  0.152500  G  0.099460  0.152500   \n",
       "11  chr3:132502295:C:T  0.000398  0.000385  0.000418  T  0.000385  0.000418   \n",
       "12  chr3:132502298:C:T  0.000487  0.000189  0.000817  T  0.000189  0.000817   \n",
       "13  chr3:132503217:G:A  0.000097  0.000189  0.000000  A  0.000189  0.000000   \n",
       "14  chr3:132507256:A:G  0.001169  0.000946  0.001427  G  0.000946  0.001427   \n",
       "15  chr3:132511239:A:G  0.000098  0.000189  0.000000  G  0.000189  0.000000   \n",
       "16  chr3:132516743:C:T  0.000000  0.000000  0.000000  T  0.000000  0.000000   \n",
       "17  chr3:132516789:G:C  0.000391  0.000569  0.000205  C  0.000569  0.000205   \n",
       "18  chr3:132525761:G:A  0.000195  0.000189  0.000204  A  0.000189  0.000204   \n",
       "19  chr3:132538179:C:T  0.000292  0.000189  0.000204  T  0.000189  0.000204   \n",
       "20  chr3:132538225:G:A  0.001169  0.001135  0.001224  A  0.001135  0.001224   \n",
       "21  chr3:132538272:A:G  0.000000  0.000000  0.000000  G  0.000000  0.000000   \n",
       "\n",
       "   A2             P      OR      L95      U95  \n",
       "0   T  2.991000e-01  0.0000  0.00000      NaN  \n",
       "1   T  2.993000e-01  0.0000  0.00000      NaN  \n",
       "2   G  2.250000e-01  0.5143  0.17220   1.5360  \n",
       "3   C  2.992000e-01  0.0000  0.00000      NaN  \n",
       "4   A  1.731000e-01     NaN      NaN      NaN  \n",
       "5   A  1.730000e-01     NaN      NaN      NaN  \n",
       "6   A           NaN     NaN      NaN      NaN  \n",
       "7   G           NaN     NaN      NaN      NaN  \n",
       "8   C  1.420000e-01  0.0000  0.00000      NaN  \n",
       "9   G  8.425000e-02  0.1854  0.02165   1.5870  \n",
       "10  T  1.635000e-14  0.6138  0.54150   0.6959  \n",
       "11  C  9.348000e-01  0.9214  0.12970   6.5440  \n",
       "12  C  1.535000e-01  0.2317  0.02589   2.0740  \n",
       "13  G  3.351000e-01     NaN      NaN      NaN  \n",
       "14  A  4.793000e-01  0.6626  0.21020   2.0890  \n",
       "15  A  3.362000e-01     NaN      NaN      NaN  \n",
       "16  C           NaN     NaN      NaN      NaN  \n",
       "17  G  3.552000e-01  2.7800  0.28910  26.7400  \n",
       "18  G  9.583000e-01  0.9288  0.05808  14.8500  \n",
       "19  C  9.579000e-01  0.9280  0.05803  14.8400  \n",
       "20  G  8.960000e-01  0.9273  0.29890   2.8770  \n",
       "21  A           NaN     NaN      NaN      NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_and_freq = pd.merge(all_freq, assoc, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>0.2991</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>0.2993</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>0.001371</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2250</td>\n",
       "      <td>0.5143</td>\n",
       "      <td>0.1722</td>\n",
       "      <td>1.536</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2992</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.1731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF     MAF_A     MAF_U A1       F_A       F_U A2  \\\n",
       "0  chr3:132454147:T:A  0.000097  0.000000  0.000204  A  0.000000  0.000204  T   \n",
       "1  chr3:132457349:T:C  0.000098  0.000000  0.000204  C  0.000000  0.000204  T   \n",
       "2  chr3:132477807:G:A  0.001371  0.000951  0.001847  A  0.000951  0.001847  G   \n",
       "3  chr3:132484610:C:T  0.000097  0.000000  0.000204  T  0.000000  0.000204  C   \n",
       "4  chr3:132492604:A:G  0.000195  0.000378  0.000000  G  0.000378  0.000000  A   \n",
       "\n",
       "        P      OR     L95    U95  BONF  \n",
       "0  0.2991  0.0000  0.0000    NaN   1.0  \n",
       "1  0.2993  0.0000  0.0000    NaN   1.0  \n",
       "2  0.2250  0.5143  0.1722  1.536   1.0  \n",
       "3  0.2992  0.0000  0.0000    NaN   1.0  \n",
       "4  0.1731     NaN     NaN    NaN   1.0  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the results\n",
    "assoc_and_freq = pd.merge(assoc_and_freq, assoc_adjusted, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_EAS_R7/chr3_EAS_release7\n",
      "  --extract range 202407_BURDEN_EAS_R7/EAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_EAS_R7/EAS.samplestoKeep\n",
      "  --out 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn\n",
      "  --recode A\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "3353374 variants loaded from .bim file.\n",
      "5136 people (3268 males, 1868 females) loaded from .fam.\n",
      "5096 phenotype values loaded from .fam.\n",
      "--extract range: 3353352 variants excluded.\n",
      "--extract range: 22 variants remaining.\n",
      "--keep: 5135 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5135 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.993538.\n",
      "22 variants and 5135 people pass filters and QC.\n",
      "Among remaining phenotypes, 2643 are cases and 2452 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "--recode A to 202407_BURDEN_EAS_R7/EAS_DNAJC13.coding_nonsyn.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EAS for DNAJC13: 22\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr3:132454147:T:A_A',\n",
       " 'chr3:132457349:T:C_C',\n",
       " 'chr3:132477807:G:A_A',\n",
       " 'chr3:132484610:C:T_T',\n",
       " 'chr3:132492604:A:G_G',\n",
       " 'chr3:132494190:A:G_G',\n",
       " 'chr3:132495159:A:G_G',\n",
       " 'chr3:132496652:G:A_A',\n",
       " 'chr3:132499231:C:T_T',\n",
       " 'chr3:132499777:G:A_A',\n",
       " 'chr3:132499779:G:T_G',\n",
       " 'chr3:132502295:C:T_T',\n",
       " 'chr3:132502298:C:T_T',\n",
       " 'chr3:132503217:G:A_A',\n",
       " 'chr3:132507256:A:G_G',\n",
       " 'chr3:132511239:A:G_G',\n",
       " 'chr3:132516743:C:T_T',\n",
       " 'chr3:132516789:G:C_C',\n",
       " 'chr3:132525761:G:A_A',\n",
       " 'chr3:132538179:C:T_T',\n",
       " 'chr3:132538225:G:A_A',\n",
       " 'chr3:132538272:A:G_G']"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for DNAJC13: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132495159:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132495159:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132496652:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132496652:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499231:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499777:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T_G</td>\n",
       "      <td>24</td>\n",
       "      <td>433</td>\n",
       "      <td>2643</td>\n",
       "      <td>51</td>\n",
       "      <td>558</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132502295:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502298:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132503217:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132507256:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132511239:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132516743:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132516743:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132516789:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr3:132525761:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr3:132538179:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr3:132538225:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr3:132538272:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "      <td>chr3:132538272:A:G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Variant  Hom Cases  Het Cases  Total Cases  Hom Controls  \\\n",
       "0   chr3:132454147:T:A_A          0          0         2643             0   \n",
       "1   chr3:132457349:T:C_C          0          0         2643             0   \n",
       "2   chr3:132477807:G:A_A          0          5         2643             0   \n",
       "3   chr3:132484610:C:T_T          0          0         2643             0   \n",
       "4   chr3:132492604:A:G_G          0          2         2643             0   \n",
       "5   chr3:132494190:A:G_G          0          2         2643             0   \n",
       "6   chr3:132495159:A:G_G          0          0         2643             0   \n",
       "7   chr3:132496652:G:A_A          0          0         2643             0   \n",
       "8   chr3:132499231:C:T_T          0          0         2643             0   \n",
       "9   chr3:132499777:G:A_A          0          1         2643             0   \n",
       "10  chr3:132499779:G:T_G         24        433         2643            51   \n",
       "11  chr3:132502295:C:T_T          0          2         2643             0   \n",
       "12  chr3:132502298:C:T_T          0          1         2643             0   \n",
       "13  chr3:132503217:G:A_A          0          1         2643             0   \n",
       "14  chr3:132507256:A:G_G          0          5         2643             0   \n",
       "15  chr3:132511239:A:G_G          0          1         2643             0   \n",
       "16  chr3:132516743:C:T_T          0          0         2643             0   \n",
       "17  chr3:132516789:G:C_C          0          3         2643             0   \n",
       "18  chr3:132525761:G:A_A          0          1         2643             0   \n",
       "19  chr3:132538179:C:T_T          0          1         2643             0   \n",
       "20  chr3:132538225:G:A_A          0          6         2643             0   \n",
       "21  chr3:132538272:A:G_G          0          0         2643             0   \n",
       "\n",
       "    Het Controls  Total Controls                 SNP  \n",
       "0              1            2452  chr3:132454147:T:A  \n",
       "1              1            2452  chr3:132457349:T:C  \n",
       "2              9            2452  chr3:132477807:G:A  \n",
       "3              1            2452  chr3:132484610:C:T  \n",
       "4              0            2452  chr3:132492604:A:G  \n",
       "5              0            2452  chr3:132494190:A:G  \n",
       "6              0            2452  chr3:132495159:A:G  \n",
       "7              0            2452  chr3:132496652:G:A  \n",
       "8              2            2452  chr3:132499231:C:T  \n",
       "9              5            2452  chr3:132499777:G:A  \n",
       "10           558            2452  chr3:132499779:G:T  \n",
       "11             2            2452  chr3:132502295:C:T  \n",
       "12             4            2452  chr3:132502298:C:T  \n",
       "13             0            2452  chr3:132503217:G:A  \n",
       "14             7            2452  chr3:132507256:A:G  \n",
       "15             0            2452  chr3:132511239:A:G  \n",
       "16             0            2452  chr3:132516743:C:T  \n",
       "17             1            2452  chr3:132516789:G:C  \n",
       "18             1            2452  chr3:132525761:G:A  \n",
       "19             1            2452  chr3:132538179:C:T  \n",
       "20             6            2452  chr3:132538225:G:A  \n",
       "21             0            2452  chr3:132538272:A:G  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "#    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "#    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "#        'Freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "#        'Freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Merge with assoc results \n",
    "full_results = pd.merge(assoc_and_freq, df_results, on=\"SNP\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_full_results = full_results[['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR','L95', 'U95', 'BONF', \n",
    "                                   'MAF', 'MAF_A', 'MAF_U', \n",
    "                                   'Hom Cases', 'Het Cases', 'Total Cases', \n",
    "                                   'Hom Controls','Het Controls', 'Total Controls']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(22, 19)\n",
      "No. of SNPs: 22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132454147:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>2.991000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132457349:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>2.993000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132477807:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>2.250000e-01</td>\n",
       "      <td>0.5143</td>\n",
       "      <td>0.17220</td>\n",
       "      <td>1.5360</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.001371</td>\n",
       "      <td>0.000951</td>\n",
       "      <td>0.001847</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132484610:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>2.992000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132492604:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.731000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.730000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132495159:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132496652:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>1.420000e-01</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000408</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>8.425000e-02</td>\n",
       "      <td>0.1854</td>\n",
       "      <td>0.02165</td>\n",
       "      <td>1.5870</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000584</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.001020</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "      <td>1.635000e-14</td>\n",
       "      <td>0.6138</td>\n",
       "      <td>0.54150</td>\n",
       "      <td>0.6959</td>\n",
       "      <td>2.942000e-13</td>\n",
       "      <td>0.124100</td>\n",
       "      <td>0.099460</td>\n",
       "      <td>0.152500</td>\n",
       "      <td>24</td>\n",
       "      <td>433</td>\n",
       "      <td>2643</td>\n",
       "      <td>51</td>\n",
       "      <td>558</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>9.348000e-01</td>\n",
       "      <td>0.9214</td>\n",
       "      <td>0.12970</td>\n",
       "      <td>6.5440</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000398</td>\n",
       "      <td>0.000385</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr3:132502298:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>1.535000e-01</td>\n",
       "      <td>0.2317</td>\n",
       "      <td>0.02589</td>\n",
       "      <td>2.0740</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000487</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr3:132503217:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.351000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000097</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "      <td>4.793000e-01</td>\n",
       "      <td>0.6626</td>\n",
       "      <td>0.21020</td>\n",
       "      <td>2.0890</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001427</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr3:132511239:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.362000e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr3:132516743:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>3.552000e-01</td>\n",
       "      <td>2.7800</td>\n",
       "      <td>0.28910</td>\n",
       "      <td>26.7400</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000569</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr3:132525761:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>9.583000e-01</td>\n",
       "      <td>0.9288</td>\n",
       "      <td>0.05808</td>\n",
       "      <td>14.8500</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr3:132538179:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>9.579000e-01</td>\n",
       "      <td>0.9280</td>\n",
       "      <td>0.05803</td>\n",
       "      <td>14.8400</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.000189</td>\n",
       "      <td>0.000204</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>8.960000e-01</td>\n",
       "      <td>0.9273</td>\n",
       "      <td>0.29890</td>\n",
       "      <td>2.8770</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.001169</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr3:132538272:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2643</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1 A2       F_A       F_U             P      OR  \\\n",
       "0   chr3:132454147:T:A  A  T  0.000000  0.000204  2.991000e-01  0.0000   \n",
       "1   chr3:132457349:T:C  C  T  0.000000  0.000204  2.993000e-01  0.0000   \n",
       "2   chr3:132477807:G:A  A  G  0.000951  0.001847  2.250000e-01  0.5143   \n",
       "3   chr3:132484610:C:T  T  C  0.000000  0.000204  2.992000e-01  0.0000   \n",
       "4   chr3:132492604:A:G  G  A  0.000378  0.000000  1.731000e-01     NaN   \n",
       "5   chr3:132494190:A:G  G  A  0.000378  0.000000  1.730000e-01     NaN   \n",
       "6   chr3:132495159:A:G  G  A  0.000000  0.000000           NaN     NaN   \n",
       "7   chr3:132496652:G:A  A  G  0.000000  0.000000           NaN     NaN   \n",
       "8   chr3:132499231:C:T  T  C  0.000000  0.000408  1.420000e-01  0.0000   \n",
       "9   chr3:132499777:G:A  A  G  0.000189  0.001020  8.425000e-02  0.1854   \n",
       "10  chr3:132499779:G:T  G  T  0.099460  0.152500  1.635000e-14  0.6138   \n",
       "11  chr3:132502295:C:T  T  C  0.000385  0.000418  9.348000e-01  0.9214   \n",
       "12  chr3:132502298:C:T  T  C  0.000189  0.000817  1.535000e-01  0.2317   \n",
       "13  chr3:132503217:G:A  A  G  0.000189  0.000000  3.351000e-01     NaN   \n",
       "14  chr3:132507256:A:G  G  A  0.000946  0.001427  4.793000e-01  0.6626   \n",
       "15  chr3:132511239:A:G  G  A  0.000189  0.000000  3.362000e-01     NaN   \n",
       "16  chr3:132516743:C:T  T  C  0.000000  0.000000           NaN     NaN   \n",
       "17  chr3:132516789:G:C  C  G  0.000569  0.000205  3.552000e-01  2.7800   \n",
       "18  chr3:132525761:G:A  A  G  0.000189  0.000204  9.583000e-01  0.9288   \n",
       "19  chr3:132538179:C:T  T  C  0.000189  0.000204  9.579000e-01  0.9280   \n",
       "20  chr3:132538225:G:A  A  G  0.001135  0.001224  8.960000e-01  0.9273   \n",
       "21  chr3:132538272:A:G  G  A  0.000000  0.000000           NaN     NaN   \n",
       "\n",
       "        L95      U95          BONF       MAF     MAF_A     MAF_U  Hom Cases  \\\n",
       "0   0.00000      NaN  1.000000e+00  0.000097  0.000000  0.000204          0   \n",
       "1   0.00000      NaN  1.000000e+00  0.000098  0.000000  0.000204          0   \n",
       "2   0.17220   1.5360  1.000000e+00  0.001371  0.000951  0.001847          0   \n",
       "3   0.00000      NaN  1.000000e+00  0.000097  0.000000  0.000204          0   \n",
       "4       NaN      NaN  1.000000e+00  0.000195  0.000378  0.000000          0   \n",
       "5       NaN      NaN  1.000000e+00  0.000195  0.000378  0.000000          0   \n",
       "6       NaN      NaN           NaN  0.000000  0.000000  0.000000          0   \n",
       "7       NaN      NaN           NaN  0.000000  0.000000  0.000000          0   \n",
       "8   0.00000      NaN  1.000000e+00  0.000195  0.000000  0.000408          0   \n",
       "9   0.02165   1.5870  1.000000e+00  0.000584  0.000189  0.001020          0   \n",
       "10  0.54150   0.6959  2.942000e-13  0.124100  0.099460  0.152500         24   \n",
       "11  0.12970   6.5440  1.000000e+00  0.000398  0.000385  0.000418          0   \n",
       "12  0.02589   2.0740  1.000000e+00  0.000487  0.000189  0.000817          0   \n",
       "13      NaN      NaN  1.000000e+00  0.000097  0.000189  0.000000          0   \n",
       "14  0.21020   2.0890  1.000000e+00  0.001169  0.000946  0.001427          0   \n",
       "15      NaN      NaN  1.000000e+00  0.000098  0.000189  0.000000          0   \n",
       "16      NaN      NaN           NaN  0.000000  0.000000  0.000000          0   \n",
       "17  0.28910  26.7400  1.000000e+00  0.000391  0.000569  0.000205          0   \n",
       "18  0.05808  14.8500  1.000000e+00  0.000195  0.000189  0.000204          0   \n",
       "19  0.05803  14.8400  1.000000e+00  0.000292  0.000189  0.000204          0   \n",
       "20  0.29890   2.8770  1.000000e+00  0.001169  0.001135  0.001224          0   \n",
       "21      NaN      NaN           NaN  0.000000  0.000000  0.000000          0   \n",
       "\n",
       "    Het Cases  Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "0           0         2643             0             1            2452  \n",
       "1           0         2643             0             1            2452  \n",
       "2           5         2643             0             9            2452  \n",
       "3           0         2643             0             1            2452  \n",
       "4           2         2643             0             0            2452  \n",
       "5           2         2643             0             0            2452  \n",
       "6           0         2643             0             0            2452  \n",
       "7           0         2643             0             0            2452  \n",
       "8           0         2643             0             2            2452  \n",
       "9           1         2643             0             5            2452  \n",
       "10        433         2643            51           558            2452  \n",
       "11          2         2643             0             2            2452  \n",
       "12          1         2643             0             4            2452  \n",
       "13          1         2643             0             0            2452  \n",
       "14          5         2643             0             7            2452  \n",
       "15          1         2643             0             0            2452  \n",
       "16          0         2643             0             0            2452  \n",
       "17          3         2643             0             1            2452  \n",
       "18          1         2643             0             1            2452  \n",
       "19          1         2643             0             1            2452  \n",
       "20          6         2643             0             6            2452  \n",
       "21          0         2643             0             0            2452  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clean_full_results.shape)\n",
    "print(f'No. of SNPs: {clean_full_results.shape[0]}')\n",
    "clean_full_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr3:132499779:G:T']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_freq = clean_full_results[clean_full_results['P']<0.05]\n",
    "sig_snps = sig_freq['SNP'].tolist()\n",
    "sig_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.09946</td>\n",
       "      <td>0.1525</td>\n",
       "      <td>1.635000e-14</td>\n",
       "      <td>0.6138</td>\n",
       "      <td>0.5415</td>\n",
       "      <td>0.6959</td>\n",
       "      <td>2.942000e-13</td>\n",
       "      <td>0.1241</td>\n",
       "      <td>0.09946</td>\n",
       "      <td>0.1525</td>\n",
       "      <td>24</td>\n",
       "      <td>433</td>\n",
       "      <td>2643</td>\n",
       "      <td>51</td>\n",
       "      <td>558</td>\n",
       "      <td>2452</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   SNP A1 A2      F_A     F_U             P      OR     L95  \\\n",
       "10  chr3:132499779:G:T  G  T  0.09946  0.1525  1.635000e-14  0.6138  0.5415   \n",
       "\n",
       "       U95          BONF     MAF    MAF_A   MAF_U  Hom Cases  Het Cases  \\\n",
       "10  0.6959  2.942000e-13  0.1241  0.09946  0.1525         24        433   \n",
       "\n",
       "    Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "10         2643            51           558            2452  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_df_results = clean_full_results[clean_full_results['SNP'].isin(sig_snps)]\n",
    "sig_df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\t132499779\tchr3:132499779:G:T\tG\tT\tIMPUTED;AF=0.855382;MAF=0.144618;AVG_CS=0.978315;R2=0.852045\r\n"
     ]
    }
   ],
   "source": [
    "for snp in sig_snps:\n",
    "    ! grep {snp} {WORK_DIR}/chr3_{ancestry}_release7.pvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save files to VM\n",
    "clean_full_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt', sep=\"\\t\", index=False)\n",
    "sig_df_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt' , sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
