{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Variants in DNAJC13 and their Association with Parkinson's Disease Across Different Ancestral Backgrounds\n",
    "\n",
    "* **Project**: DNAJC13 Gene Assesment\n",
    "* **Last updated**: September 2024\n",
    "* **Version**: Python 3.9\n",
    "* **Data**: GP2 Release 7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "This notebook is focused on analysing association between DNAJC13 variants in South Asian (SAS) ancestry and Parkinson's disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1915/3943855956.py:26: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## AMP-PD v2.5\n",
    "## Explicitly define release v2.5 path \n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_GATK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "print('AMP-PD v2.5')\n",
    "print(f'Path to AMP-PD v2.5 Clinical Data @ `AMP_CLINICAL_RELEASE_PATH`: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v2.5 WGS Data @ `AMP_WGS_RELEASE_PLINK_PATH`: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')\n",
    "\n",
    "## GP2 v7.0\n",
    "## Explicitly define release v7.0 path \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘202407_BURDEN_SAS_R7’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "ancestry = 'SAS'\n",
    "WORK_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "WORKSPACE_DIR = f'202407_BURDEN_{ancestry}_R7'\n",
    "\n",
    "! mkdir {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/plink2_linux_amd_avx2_20240704.zip\n",
    "unzip -o plink2_linux_amd_avx2_20240704.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20140902.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the 'humandb' directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR: Download resources for annotation\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rvtests is already installed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## chr 3\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_IMPUTED_GENO_PATH}/{ancestry}/chr3_{ancestry}_release7.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## master key\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_CLINICAL_RELEASE_PATH}/master_key_release7_final.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/{ancestry}_release7.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-f03d16c4 -m cp gs://gp2tier2/release7_30042024/raw_genotypes/SAS/SAS_release7.eigenvec 202407_BURDEN_SAS_R7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://gp2tier2/release7_30042024/raw_genotypes/SAS/SAS_release7.eigenvec...\n",
      "/ [1/1 files][ 72.2 KiB/ 72.2 KiB] 100% Done                                    \n",
      "Operation completed over 1 objects/72.2 KiB.                                     \n"
     ]
    }
   ],
   "source": [
    "## PCs\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RAW_GENO_PATH}/{ancestry}/{ancestry}_release7.eigenvec {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# refFlat\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {WORKSPACE_BUCKET}/notebooks/misc/refFlat_HG38_all_chr.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Covariate File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(f'{WORK_DIR}/master_key_release7_final.csv')\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'race_for_qc', 'label']]\n",
    "key.rename(columns = {'GP2sampleID': 'IID',\n",
    "                      'baseline_GP2_phenotype':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'race_for_qc':'RACE',\n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = key[key['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals \n",
    "related_df = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.related')\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID1'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(ancestry_key.shape)\n",
    "ancestry_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (0/1)\n",
    "    # PD = 1; control = 0\n",
    "pheno_mapping = {\"PD\": 1, \"Control\": 0}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "1       335\n",
       "0       206\n",
       "<NA>     23\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get the PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.eigenvec', sep='\\t')\n",
    "\n",
    "# Split the single column into multiple columns (No longer needed with release 7)\n",
    "#pcs_split = pcs['#FID,IID,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10'].str.split(',', expand=True)\n",
    "\n",
    "selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "\n",
    "\n",
    "#pcs = pd.DataFrame(data=pcs_split.iloc[:, 1:7].values, columns=selected_columns)\n",
    "pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "\n",
    "# Drop the first row (since it's now the column names)\n",
    "pcs = pcs.drop(0)\n",
    "\n",
    "# Reset the index to remove any potential issues\n",
    "pcs = pcs.reset_index(drop=True)\n",
    "\n",
    "\n",
    "# Display the resulting DataFrame\n",
    "print(pcs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert sex to binary (0/1)\n",
    "    # Female = 1; male = 0\n",
    "sex_mapping = {\"Female\": 1, \"Male\": 0}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "0       364\n",
       "1       199\n",
       "<NA>      1\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX\n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
       "       'AAO', 'RACE', 'label', 'PHENO'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Make covariate file\n",
    "df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['FID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = df[['FID','IID', 'SEX', 'PHENO', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO  SEX\n",
       "0      0      133\n",
       "       1       65\n",
       "1      0      202\n",
       "       1      116\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_SAS_release7.log\tmaster_key_release7_final.csv  SAS_release7.related\r\n",
      "chr3_SAS_release7.pgen\trefFlat_HG38_all_chr.txt       SAS.samplestoKeep\r\n",
      "chr3_SAS_release7.psam\tSAS_covariate_file.txt\r\n",
      "chr3_SAS_release7.pvar\tSAS_release7.eigenvec\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}//')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation using ANNOVAR\n",
    "\n",
    "* *DNAJC13* from NCBI gene\n",
    "* hg38 (chr3:132417502-132539032)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --export vcf id-paste=iid\n",
      "  --from-bp 132417502\n",
      "  --mac 2\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13\n",
      "  --pfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Wed Jul 10 01:19:49 2024\n",
      "7450 MiB RAM detected, ~4290 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "576 samples (200 females, 376 males; 576 founders) loaded from\n",
      "202407_BURDEN_SAS_R7/chr3_SAS_release7.psam.\n",
      "1959689 variants loaded from 202407_BURDEN_SAS_R7/chr3_SAS_release7.pvar.\n",
      "1 binary phenotype loaded (338 cases, 216 controls).\n",
      "Calculating allele frequencies... done.\n",
      "462 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "540 variants remaining after main filters.\n",
      "--export vcf to 202407_BURDEN_SAS_R7/SAS_DNAJC13.vcf ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n",
      "End time: Wed Jul 10 01:19:49 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 202407_BURDEN_SAS_R7/SAS_DNAJC13.vcf.gz > 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.avinput>\n",
      "NOTICE: Finished reading 559 lines from VCF file\n",
      "NOTICE: A total of 540 locus in VCF file passed QC threshold, representing 495 SNPs (347 transitions and 148 transversions) and 45 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 285120 SNP genotypes (199872 transitions and 85248 transversions) and 25920 indels/substitutions for 576 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.refGene -exonsort -nofirstcodondel 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.refGene.variant_function, 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 540 unique variants in 540 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 540 unique variants in 540 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 6\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_DNAJC13.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo579</th>\n",
       "      <th>Otherinfo580</th>\n",
       "      <th>Otherinfo581</th>\n",
       "      <th>Otherinfo582</th>\n",
       "      <th>Otherinfo583</th>\n",
       "      <th>Otherinfo584</th>\n",
       "      <th>Otherinfo585</th>\n",
       "      <th>Otherinfo586</th>\n",
       "      <th>Otherinfo587</th>\n",
       "      <th>Otherinfo588</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132417623</td>\n",
       "      <td>132417623</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16928del-;NM_015268:c.-16928del-</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132418181</td>\n",
       "      <td>132418181</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132418296</td>\n",
       "      <td>132418296</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132418405</td>\n",
       "      <td>132418405</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132418415</td>\n",
       "      <td>132418415</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>3</td>\n",
       "      <td>132538575</td>\n",
       "      <td>132538575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*293A&gt;G;NM_015268:c.*293A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>3</td>\n",
       "      <td>132538679</td>\n",
       "      <td>132538679</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*397A&gt;T;NM_015268:c.*397A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>3</td>\n",
       "      <td>132538734</td>\n",
       "      <td>132538734</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*452G&gt;A;NM_015268:c.*452G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>3</td>\n",
       "      <td>132538890</td>\n",
       "      <td>132538890</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*608G&gt;A;NM_015268:c.*608G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>.|.</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>3</td>\n",
       "      <td>132538903</td>\n",
       "      <td>132538903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*621G&gt;A;NM_015268:c.*621G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0|1</td>\n",
       "      <td>.|.</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>540 rows × 599 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      3  132417623  132417623   C   -         UTR5      DNAJC13   \n",
       "1      3  132418181  132418181   A   G     intronic      DNAJC13   \n",
       "2      3  132418296  132418296   C   T     intronic      DNAJC13   \n",
       "3      3  132418405  132418405   G   A     intronic      DNAJC13   \n",
       "4      3  132418415  132418415   A   G     intronic      DNAJC13   \n",
       "..   ...        ...        ...  ..  ..          ...          ...   \n",
       "535    3  132538575  132538575   A   G         UTR3      DNAJC13   \n",
       "536    3  132538679  132538679   A   T         UTR3      DNAJC13   \n",
       "537    3  132538734  132538734   G   A         UTR3      DNAJC13   \n",
       "538    3  132538890  132538890   G   A         UTR3      DNAJC13   \n",
       "539    3  132538903  132538903   G   A         UTR3      DNAJC13   \n",
       "\n",
       "                                   GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0    NM_001329126:c.-16928del-;NM_015268:c.-16928del-                  .   \n",
       "1                                                   .                  .   \n",
       "2                                                   .                  .   \n",
       "3                                                   .                  .   \n",
       "4                                                   .                  .   \n",
       "..                                                ...                ...   \n",
       "535        NM_001329126:c.*293A>G;NM_015268:c.*293A>G                  .   \n",
       "536        NM_001329126:c.*397A>T;NM_015268:c.*397A>T                  .   \n",
       "537        NM_001329126:c.*452G>A;NM_015268:c.*452G>A                  .   \n",
       "538        NM_001329126:c.*608G>A;NM_015268:c.*608G>A                  .   \n",
       "539        NM_001329126:c.*621G>A;NM_015268:c.*621G>A                  .   \n",
       "\n",
       "    AAChange.refGene  ... Otherinfo579  Otherinfo580 Otherinfo581  \\\n",
       "0                  .  ...          0|0           0|0          0|0   \n",
       "1                  .  ...          0|0           0|0          0|0   \n",
       "2                  .  ...          0|0           0|0          0|0   \n",
       "3                  .  ...          0|0           0|0          0|0   \n",
       "4                  .  ...          0|0           0|1          0|0   \n",
       "..               ...  ...          ...           ...          ...   \n",
       "535                .  ...          0|0           0|0          0|0   \n",
       "536                .  ...          0|0           0|0          0|0   \n",
       "537                .  ...          0|0           0|0          0|0   \n",
       "538                .  ...          0|0           .|.          0|0   \n",
       "539                .  ...          0|1           .|.          0|0   \n",
       "\n",
       "    Otherinfo582  Otherinfo583  Otherinfo584 Otherinfo585 Otherinfo586  \\\n",
       "0            0|0           0|0           0|0          0|0          0|0   \n",
       "1            0|0           0|0           0|0          0|0          0|0   \n",
       "2            0|0           0|0           0|0          0|0          0|0   \n",
       "3            0|0           0|0           0|0          0|0          0|0   \n",
       "4            0|0           1|0           0|0          0|0          0|0   \n",
       "..           ...           ...           ...          ...          ...   \n",
       "535          0|0           1|0           0|0          0|0          0|0   \n",
       "536          0|0           0|0           0|0          0|0          0|0   \n",
       "537          0|0           0|0           0|0          0|0          0|0   \n",
       "538          0|0           0|0           0|0          0|0          0|0   \n",
       "539          1|0           0|1           1|0          0|1          0|0   \n",
       "\n",
       "    Otherinfo587 Otherinfo588  \n",
       "0            0|0          0|0  \n",
       "1            0|0          0|0  \n",
       "2            0|0          0|0  \n",
       "3            0|0          0|0  \n",
       "4            0|1          0|1  \n",
       "..           ...          ...  \n",
       "535          0|1          0|1  \n",
       "536          0|0          0|0  \n",
       "537          0|0          0|0  \n",
       "538          0|0          0|0  \n",
       "539          0|0          0|0  \n",
       "\n",
       "[540 rows x 599 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter intronic\n",
    "intronic = gene[(gene['Func.refGene'] == 'intronic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR3\n",
    "utr3 = gene[(gene['Func.refGene'] == 'UTR3')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR5\n",
    "utr5 = gene[(gene['Func.refGene'] == 'UTR5')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter exonic and synonymous variants\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo579</th>\n",
       "      <th>Otherinfo580</th>\n",
       "      <th>Otherinfo581</th>\n",
       "      <th>Otherinfo582</th>\n",
       "      <th>Otherinfo583</th>\n",
       "      <th>Otherinfo584</th>\n",
       "      <th>Otherinfo585</th>\n",
       "      <th>Otherinfo586</th>\n",
       "      <th>Otherinfo587</th>\n",
       "      <th>Otherinfo588</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>3</td>\n",
       "      <td>132516789</td>\n",
       "      <td>132516789</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon48:c.G5646C:p.M1882I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>3</td>\n",
       "      <td>132523636</td>\n",
       "      <td>132523636</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon51:c.G5983C:p.V1995L,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 599 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "235    3  132475009  132475009   C   A       exonic      DNAJC13   \n",
       "326    3  132494190  132494190   A   G       exonic      DNAJC13   \n",
       "349    3  132499231  132499231   C   T       exonic      DNAJC13   \n",
       "353    3  132499777  132499777   G   A       exonic      DNAJC13   \n",
       "354    3  132499779  132499779   G   T       exonic      DNAJC13   \n",
       "367    3  132502295  132502295   C   T       exonic      DNAJC13   \n",
       "388    3  132507256  132507256   A   G       exonic      DNAJC13   \n",
       "437    3  132516789  132516789   G   C       exonic      DNAJC13   \n",
       "467    3  132523636  132523636   G   C       exonic      DNAJC13   \n",
       "532    3  132538225  132538225   G   A       exonic      DNAJC13   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "235                  .  nonsynonymous SNV   \n",
       "326                  .  nonsynonymous SNV   \n",
       "349                  .  nonsynonymous SNV   \n",
       "353                  .  nonsynonymous SNV   \n",
       "354                  .  nonsynonymous SNV   \n",
       "367                  .  nonsynonymous SNV   \n",
       "388                  .  nonsynonymous SNV   \n",
       "437                  .  nonsynonymous SNV   \n",
       "467                  .  nonsynonymous SNV   \n",
       "532                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo579  \\\n",
       "235  DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...  ...          0|0   \n",
       "326  DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...  ...          0|0   \n",
       "349  DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...  ...          0|0   \n",
       "353  DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...  ...          0|0   \n",
       "354  DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...  ...          1|1   \n",
       "367  DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...  ...          0|0   \n",
       "388  DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...  ...          0|0   \n",
       "437  DNAJC13:NM_015268:exon48:c.G5646C:p.M1882I,DNA...  ...          0|0   \n",
       "467  DNAJC13:NM_015268:exon51:c.G5983C:p.V1995L,DNA...  ...          0|0   \n",
       "532  DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...  ...          0|0   \n",
       "\n",
       "     Otherinfo580 Otherinfo581 Otherinfo582  Otherinfo583  Otherinfo584  \\\n",
       "235           0|0          0|0          0|0           0|0           0|0   \n",
       "326           0|0          0|0          0|0           0|0           0|0   \n",
       "349           0|0          0|0          0|0           0|0           0|0   \n",
       "353           0|0          0|0          0|0           0|0           0|0   \n",
       "354           1|1          1|0          1|0           1|1           1|1   \n",
       "367           0|0          0|0          0|0           0|0           0|0   \n",
       "388           0|0          0|0          0|0           0|0           0|0   \n",
       "437           0|0          0|0          0|0           0|0           0|0   \n",
       "467           0|0          0|0          0|0           0|0           0|0   \n",
       "532           0|0          0|0          0|0           0|0           0|0   \n",
       "\n",
       "    Otherinfo585 Otherinfo586 Otherinfo587 Otherinfo588  \n",
       "235          0|0          0|0          0|0          0|0  \n",
       "326          0|0          0|0          0|0          0|0  \n",
       "349          0|0          0|0          0|0          0|0  \n",
       "353          0|0          0|0          0|0          0|0  \n",
       "354          0|1          0|0          1|1          0|1  \n",
       "367          0|0          0|0          0|0          0|0  \n",
       "388          0|0          0|0          0|0          0|0  \n",
       "437          0|0          0|0          0|0          0|0  \n",
       "467          0|0          0|0          0|0          0|0  \n",
       "532          0|0          0|0          0|0          0|0  \n",
       "\n",
       "[10 rows x 599 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "totalVariants = len(gene.axes[0])\n",
    "totalIntronic = len(intronic.axes[0])\n",
    "totalUTR3 = len(utr3.axes[0])\n",
    "totalUTR5 = len(utr5.axes[0])\n",
    "totalExonicSyn = len(coding_synonymous.axes[0])\n",
    "totalExonicNonSyn = len(coding_nonsynonymous.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Variants:  540\n",
      "Intronic: 519\n",
      "UTR3: 7\n",
      "UTR5: 1\n",
      "Exonic Syn: 3\n",
      "Exonic NonSyn: 10\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Variants: \", totalVariants)\n",
    "print(\"Intronic:\", totalIntronic)\n",
    "print(\"UTR3:\", totalUTR3)\n",
    "print(\"UTR5:\", totalUTR5)\n",
    "print(\"Exonic Syn:\", totalExonicSyn)\n",
    "print(\"Exonic NonSyn:\", totalExonicNonSyn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>3</td>\n",
       "      <td>132516789</td>\n",
       "      <td>132516789</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>3</td>\n",
       "      <td>132523636</td>\n",
       "      <td>132523636</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "235    3  132475009  132475009      DNAJC13\n",
       "326    3  132494190  132494190      DNAJC13\n",
       "349    3  132499231  132499231      DNAJC13\n",
       "353    3  132499777  132499777      DNAJC13\n",
       "354    3  132499779  132499779      DNAJC13\n",
       "367    3  132502295  132502295      DNAJC13\n",
       "388    3  132507256  132507256      DNAJC13\n",
       "437    3  132516789  132516789      DNAJC13\n",
       "467    3  132523636  132523636      DNAJC13\n",
       "532    3  132538225  132538225      DNAJC13"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_SAS_release7.log\r\n",
      "chr3_SAS_release7.pgen\r\n",
      "chr3_SAS_release7.psam\r\n",
      "chr3_SAS_release7.pvar\r\n",
      "master_key_release7_final.csv\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "SAS_covariate_file.txt\r\n",
      "SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "SAS_DNAJC13.annovar.avinput\r\n",
      "SAS_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "SAS_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "SAS_DNAJC13.log\r\n",
      "SAS_DNAJC13.vcf.gz\r\n",
      "SAS_DNAJC13.vcf.gz.tbi\r\n",
      "SAS_release7.eigenvec\r\n",
      "SAS_release7.related\r\n",
      "SAS.samplestoKeep\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep {WORKSPACE_BUCKET}//{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/chr3_SAS_release7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --pfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "\n",
      "Start time: Wed Jul 10 01:20:15 2024\n",
      "7450 MiB RAM detected, ~4293 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "576 samples (200 females, 376 males; 576 founders) loaded from\n",
      "202407_BURDEN_SAS_R7/chr3_SAS_release7.psam.\n",
      "1959689 variants loaded from 202407_BURDEN_SAS_R7/chr3_SAS_release7.pvar.\n",
      "1 binary phenotype loaded (338 cases, 216 controls).\n",
      "Writing 202407_BURDEN_SAS_R7/chr3_SAS_release7.fam ... done.\n",
      "Writing 202407_BURDEN_SAS_R7/chr3_SAS_release7.bim ... done.\n",
      "Writing 202407_BURDEN_SAS_R7/chr3_SAS_release7.bed ... 101316202326303336404346505356606366707376808386909396done.\n",
      "End time: Wed Jul 10 01:20:37 2024\n"
     ]
    }
   ],
   "source": [
    "# Convert the files from Plink 2.0 to Plink 1.9 format\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--make-bed \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}_release7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding\n",
      "  --recode vcf-iid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "10 variants and 575 people pass filters and QC.\n",
      "Among remaining phenotypes, 337 are cases and 216 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "--recode vcf-iid to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "    # later do based on class and frequency\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--recode vcf-iid \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\r\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\r\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\r\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\r\n",
      "\r\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\r\n",
      "\r\n",
      "Available Options\r\n",
      "      Basic Input/Output:\r\n",
      "                          --inVcf [202407_BURDEN_SAS_R7/SAS_DNAJC13.coding.vcf.gz]\r\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\r\n",
      "                          --out [202407_BURDEN_SAS_R7/SAS_DNAJC13.burden.coding]\r\n",
      "                          --outputRaw\r\n",
      "       Specify Covariate: --covar [202407_BURDEN_SAS_R7/SAS_covariate_file.txt]\r\n",
      "                          --covar-name [SEX,PC1,PC2,PC3,PC4], --sex\r\n",
      "       Specify Phenotype: --pheno [202407_BURDEN_SAS_R7/SAS_covariate_file.txt]\r\n",
      "                          --inverseNormal, --useResidualAsPhenotype\r\n",
      "                          --mpheno [4], --pheno-name [], --qtl\r\n",
      "                          --multiplePheno []\r\n",
      "        Specify Genotype: --dosage [], --multipleAllele\r\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\r\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\r\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\r\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\r\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\r\n",
      "                          --annoType []\r\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\r\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skat\r\n",
      "                         skato], --meta []\r\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\r\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\r\n",
      "                          --boltPlinkNoCheck\r\n",
      "          Grouping Unit :\r\n",
      "                          --geneFile [202407_BURDEN_SAS_R7/refFlat_HG38_all_chr.txt]\r\n",
      "                          --gene [DNAJC13], --setList [], --setFile [], --set []\r\n",
      "        Frequency Cutoff: --freqUpper [], --freqLower []\r\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\r\n",
      "    Conditional Analysis: --condition []\r\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\r\n",
      "                          --outputID, --help\r\n",
      "\r\n",
      "Effective Options\r\n",
      "    --inVcf 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding.vcf.gz\r\n",
      "    --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.burden.coding\r\n",
      "    --covar 202407_BURDEN_SAS_R7/SAS_covariate_file.txt\r\n",
      "    --covar-name SEX,PC1,PC2,PC3,PC4\r\n",
      "    --pheno 202407_BURDEN_SAS_R7/SAS_covariate_file.txt\r\n",
      "    --mpheno 4\r\n",
      "    --kernel skat,skato\r\n",
      "    --geneFile 202407_BURDEN_SAS_R7/refFlat_HG38_all_chr.txt\r\n",
      "    --gene DNAJC13\r\n",
      "    --noweb\r\n",
      "    --hide-covar\r\n",
      "\r\n",
      "[INFO]\tProgram version: 20190205\r\n",
      "[INFO]\tAnalysis started at: Wed Jul 10 01:20:43 2024\r\n",
      "[INFO]\tLoaded [ 575 ] samples from genotype files\r\n",
      "[WARN]\tSkip line 30 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 56 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 72 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 73 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 74 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 75 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 76 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 77 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 122 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 124 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 127 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 129 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 131 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 133 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 135 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 137 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 141 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 199 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 200 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 202 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 210 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 245 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 264 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 271 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 272 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 275 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 279 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 282 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 288 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 289 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 291 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 292 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 293 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 295 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 296 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 297 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 298 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 299 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 300 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 301 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 302 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 303 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 304 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 305 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 306 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 363 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 364 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 367 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 372 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 373 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 407 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 419 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 483 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 484 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 485 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 486 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 533 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 559 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[WARN]\tSkip line 568 (short of columns) in phenotype file [ 202407_BURDEN_SAS_R7/SAS_covariate_file.txt ]\r\n",
      "[INFO]\tLoaded [ 516 ] sample phenotypes\r\n",
      "[WARN]\tTotal [ 59 ] samples are dropped from VCF file due to missing phenotype.\r\n",
      "[WARN]\tDrop sample [ BCM_000801_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_003435_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_008898_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_008971_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_008985_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_009004_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_009035_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ CORIELL_009060_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ m-KUL_000003_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tDrop sample [ m-KUL_000023_s1 ] from VCF file due to missing phenotype\r\n",
      "[WARN]\tSkip outputting additional [ 49 ] samples due to missing phenotype.\r\n",
      "[INFO]\tBegin to read covariate file\r\n",
      "[WARN]\tTotal [ 59 ] samples are skipped from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ BCM_000801_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_003435_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_008898_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_008971_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_008985_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_009004_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_009035_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ CORIELL_009060_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ m-KUL_000003_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ m-KUL_000023_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip sample [ m-KUL_000033_s1 ] from covariate file due to missing phenotype\r\n",
      "[WARN]\tSkip outputting additional [ 49 ] samples from covariate file with missing phenotypes\r\n",
      "[INFO]\tLoaded 0 male, 1 female and 515 sex-unknown samples from 202407_BURDEN_SAS_R7/SAS_covariate_file.txt\r\n",
      "[INFO]\tAnalysis begins with [ 516 ] samples...\r\n",
      "[INFO]\tSKAT test significance will be evaluated using 10000 permutations at alpha = 0.05 weight = Beta[beta1 = 1.00, beta2 = 25.00]\r\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\r\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 10 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Wed Jul 10 01:20:43 2024\n",
      "[INFO]\tAnalysis took 0 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "## RVtests with covariates \n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.burden.coding \\\n",
    "--kernel skat,skato \\\n",
    "--inVcf {WORK_DIR}/{ancestry}_DNAJC13.coding.vcf.gz \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--mpheno 4 \\\n",
    "--gene DNAJC13 \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38_all_chr.txt \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4 #[WARN]Covariate [ PC5 ] has strong correlation [ r^2 = 1 ] with the response!\n",
    "\n",
    "\n",
    "# --out : Name of output \n",
    "# --burden cmc --kernel skato: tests to run \n",
    "# --inVcf : VCF file \n",
    "# --gene: gene name (if only looking at one or a few)\n",
    "# --geneFile refFlat.txt\n",
    "# --pheno :  covar file\n",
    "# --mpheno : # column that has phenotype information\n",
    "# --pheno-name : column name with phenotype in file\n",
    "# --covar : covar file\n",
    "# --freqUpper : optional, MAF cut-off\n",
    "# --covar-name : covariates, listed by column name, separated by commas (no spaces between commas)\n",
    "## 0=controls; 1=cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t516\t10\t9\t12.1983\t0.650295\t10000\t2757\t12.1983\t1000\t0\t0.362713\r\n"
     ]
    }
   ],
   "source": [
    "## look at results \n",
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t516\t10\t9\t14422.1\t1\t0.354309\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr3_SAS_release7.bed\r\n",
      "chr3_SAS_release7.bim\r\n",
      "chr3_SAS_release7.fam\r\n",
      "chr3_SAS_release7.log\r\n",
      "chr3_SAS_release7.pgen\r\n",
      "chr3_SAS_release7.psam\r\n",
      "chr3_SAS_release7.pvar\r\n",
      "master_key_release7_final.csv\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "SAS_covariate_file.txt\r\n",
      "SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "SAS_DNAJC13.annovar.avinput\r\n",
      "SAS_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "SAS_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "SAS_DNAJC13.burden.coding.log\r\n",
      "SAS_DNAJC13.burden.coding.Skat.assoc\r\n",
      "SAS_DNAJC13.burden.coding.SkatO.assoc\r\n",
      "SAS_DNAJC13.coding.log\r\n",
      "SAS_DNAJC13.coding.vcf.gz\r\n",
      "SAS_DNAJC13.coding.vcf.gz.tbi\r\n",
      "SAS_DNAJC13.log\r\n",
      "SAS_DNAJC13.vcf.gz\r\n",
      "SAS_DNAJC13.vcf.gz.tbi\r\n",
      "SAS_release7.eigenvec\r\n",
      "SAS_release7.related\r\n",
      "SAS.samplestoKeep\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.burden.coding.*.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --ci 0.95\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "10 variants and 575 people pass filters and QC.\n",
      "Among remaining phenotypes, 337 are cases and 216 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file_fullAGE.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR           SE          L95          U95 \r\n",
      "   3   chr3:132475009:C:A  132475009    A        0 0.009259    C        6.263      0.01233            0          inf            0          nan \r\n",
      "   3   chr3:132494190:A:G  132494190    G 0.004451 0.006944    A       0.3033       0.5818       0.6393       0.8188       0.1285        3.182 \r\n",
      "   3   chr3:132499231:C:T  132499231    T 0.004451        0    C        1.928        0.165           NA           NA           NA           NA \r\n",
      "   3   chr3:132499777:G:A  132499777    A 0.004451  0.03472    G        15.07    0.0001037       0.1243       0.6355      0.03577       0.4319 \r\n",
      "   3   chr3:132499779:G:T  132499779    G   0.3591   0.3495    T        0.104       0.7471        1.042        0.129       0.8097        1.342 \r\n",
      "   3   chr3:132502295:C:T  132502295    T 0.001484 0.002315    C       0.1007       0.7509       0.6404        1.416      0.03995        10.27 \r\n",
      "   3   chr3:132507256:A:G  132507256    G  0.02226  0.03704    A        2.111       0.1462       0.5918       0.3648       0.2895         1.21 \r\n",
      "   3   chr3:132516789:G:C  132516789    C 0.004478 0.004651    G     0.001743       0.9667       0.9625        0.915       0.1602        5.784 \r\n",
      "   3   chr3:132523636:G:C  132523636    C 0.001488        0    G       0.6434       0.4225           NA           NA           NA           NA \r\n",
      "   3   chr3:132538225:G:A  132538225    A        0        0    G           NA           NA           NA           NA           NA           NA \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>C</td>\n",
       "      <td>0.012330</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>A</td>\n",
       "      <td>0.581800</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>0.12850</td>\n",
       "      <td>3.1820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.165000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.03577</td>\n",
       "      <td>0.4319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>T</td>\n",
       "      <td>0.747100</td>\n",
       "      <td>1.0420</td>\n",
       "      <td>0.80970</td>\n",
       "      <td>1.3420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "      <td>C</td>\n",
       "      <td>0.750900</td>\n",
       "      <td>0.6404</td>\n",
       "      <td>0.03995</td>\n",
       "      <td>10.2700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "      <td>A</td>\n",
       "      <td>0.146200</td>\n",
       "      <td>0.5918</td>\n",
       "      <td>0.28950</td>\n",
       "      <td>1.2100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>G</td>\n",
       "      <td>0.966700</td>\n",
       "      <td>0.9625</td>\n",
       "      <td>0.16020</td>\n",
       "      <td>5.7840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.422500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP A1       F_A       F_U A2         P      OR      L95  \\\n",
       "0  chr3:132475009:C:A  A  0.000000  0.009259  C  0.012330  0.0000  0.00000   \n",
       "1  chr3:132494190:A:G  G  0.004451  0.006944  A  0.581800  0.6393  0.12850   \n",
       "2  chr3:132499231:C:T  T  0.004451  0.000000  C  0.165000     NaN      NaN   \n",
       "3  chr3:132499777:G:A  A  0.004451  0.034720  G  0.000104  0.1243  0.03577   \n",
       "4  chr3:132499779:G:T  G  0.359100  0.349500  T  0.747100  1.0420  0.80970   \n",
       "5  chr3:132502295:C:T  T  0.001484  0.002315  C  0.750900  0.6404  0.03995   \n",
       "6  chr3:132507256:A:G  G  0.022260  0.037040  A  0.146200  0.5918  0.28950   \n",
       "7  chr3:132516789:G:C  C  0.004478  0.004651  G  0.966700  0.9625  0.16020   \n",
       "8  chr3:132523636:G:C  C  0.001488  0.000000  G  0.422500     NaN      NaN   \n",
       "9  chr3:132538225:G:A  A  0.000000  0.000000  G       NaN     NaN      NaN   \n",
       "\n",
       "       U95  \n",
       "0      NaN  \n",
       "1   3.1820  \n",
       "2      NaN  \n",
       "3   0.4319  \n",
       "4   1.3420  \n",
       "5  10.2700  \n",
       "6   1.2100  \n",
       "7   5.7840  \n",
       "8      NaN  \n",
       "9      NaN  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc', delim_whitespace=True, usecols=['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR', 'L95', 'U95'])\n",
    "assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "10 variants and 575 people pass filters and QC.\n",
      "Among remaining phenotypes, 337 are cases and 216 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.41105.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (9 variants) written to\n",
      "202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "## repeat association study applying multiple testing corrections for the raw p-values (like Bonferroni post-hoc)\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --allow-no-sex --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP      UNADJ         GC       BONF       HOLM   SIDAK_SS   SIDAK_SD     FDR_BH     FDR_BY\r\n",
      "   3   chr3:132499777:G:A  0.0001037   0.001084  0.0009333  0.0009333   0.000933   0.000933  0.0009333    0.00264 \r\n",
      "   3   chr3:132475009:C:A    0.01233    0.03513     0.1109    0.09861     0.1056    0.09446    0.05547     0.1569 \r\n",
      "   3   chr3:132507256:A:G     0.1462     0.2212          1          1     0.7589     0.6693     0.3712          1 \r\n",
      "   3   chr3:132499231:C:T      0.165     0.2424          1          1     0.8026     0.6693     0.3712          1 \r\n",
      "   3   chr3:132523636:G:C     0.4225     0.4995          1          1     0.9929     0.9357     0.7604          1 \r\n",
      "   3   chr3:132494190:A:G     0.5818     0.6429          1          1     0.9996     0.9694     0.8448          1 \r\n",
      "   3   chr3:132499779:G:T     0.7471      0.786          1          1          1     0.9838     0.8448          1 \r\n",
      "   3   chr3:132502295:C:T     0.7509     0.7893          1          1          1     0.9838     0.8448          1 \r\n",
      "   3   chr3:132516789:G:C     0.9667      0.972          1          1          1     0.9838     0.9667          1 \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.000933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.110900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP      BONF\n",
       "0  chr3:132499777:G:A  0.000933\n",
       "1  chr3:132475009:C:A  0.110900\n",
       "2  chr3:132507256:A:G  1.000000\n",
       "3  chr3:132499231:C:T  1.000000\n",
       "4  chr3:132523636:G:C  1.000000\n",
       "5  chr3:132494190:A:G  1.000000\n",
       "6  chr3:132499779:G:T  1.000000\n",
       "7  chr3:132502295:C:T  1.000000\n",
       "8  chr3:132516789:G:C  1.000000"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_adjusted = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted', delim_whitespace=True, usecols=['SNP', 'BONF'])\n",
    "assoc_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.frq .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2          MAF  NCHROBS\r\n",
      "   3   chr3:132475009:C:A    A    C     0.003478     1150\r\n",
      "   3   chr3:132494190:A:G    G    A     0.005217     1150\r\n",
      "   3   chr3:132499231:C:T    T    C     0.002609     1150\r\n",
      "   3   chr3:132499777:G:A    A    G      0.01565     1150\r\n",
      "   3   chr3:132499779:G:T    G    T       0.3583     1150\r\n",
      "   3   chr3:132502295:C:T    T    C     0.001739     1150\r\n",
      "   3   chr3:132507256:A:G    G    A      0.02696     1150\r\n",
      "   3   chr3:132516789:G:C    C    G     0.004371     1144\r\n",
      "   3   chr3:132523636:G:C    C    G    0.0008711     1148\r\n",
      "   3   chr3:132538225:G:A    A    G            0     1102\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.003478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.005217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.002609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.015650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.358300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF\n",
       "0  chr3:132475009:C:A  0.003478\n",
       "1  chr3:132494190:A:G  0.005217\n",
       "2  chr3:132499231:C:T  0.002609\n",
       "3  chr3:132499777:G:A  0.015650\n",
       "4  chr3:132499779:G:T  0.358300"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq', delim_whitespace=True, usecols=['SNP', 'MAF'])\n",
    "freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --freq case-control\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.frq.cc .\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq case-control \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR                  SNP   A1   A2        MAF_A        MAF_U  NCHROBS_A  NCHROBS_U\r\n",
      "   3   chr3:132475009:C:A    A    C            0     0.009259        674        432\r\n",
      "   3   chr3:132494190:A:G    G    A     0.004451     0.006944        674        432\r\n",
      "   3   chr3:132499231:C:T    T    C     0.004451            0        674        432\r\n",
      "   3   chr3:132499777:G:A    A    G     0.004451      0.03472        674        432\r\n",
      "   3   chr3:132499779:G:T    G    T       0.3591       0.3495        674        432\r\n",
      "   3   chr3:132502295:C:T    T    C     0.001484     0.002315        674        432\r\n",
      "   3   chr3:132507256:A:G    G    A      0.02226      0.03704        674        432\r\n",
      "   3   chr3:132516789:G:C    C    G     0.004478     0.004651        670        430\r\n",
      "   3   chr3:132523636:G:C    C    G     0.001488            0        672        432\r\n",
      "   3   chr3:132538225:G:A    A    G            0            0        638        424\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP     MAF_A     MAF_U\n",
       "0  chr3:132475009:C:A  0.000000  0.009259\n",
       "1  chr3:132494190:A:G  0.004451  0.006944\n",
       "2  chr3:132499231:C:T  0.004451  0.000000\n",
       "3  chr3:132499777:G:A  0.004451  0.034720\n",
       "4  chr3:132499779:G:T  0.359100  0.349500"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq_cc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc', delim_whitespace=True, usecols=['SNP', 'MAF_A', 'MAF_U'])\n",
    "freq_cc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.002609</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.015650</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.358300</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>0.004371</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "      <td>0.000871</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF     MAF_A     MAF_U\n",
       "0  chr3:132475009:C:A  0.003478  0.000000  0.009259\n",
       "1  chr3:132494190:A:G  0.005217  0.004451  0.006944\n",
       "2  chr3:132499231:C:T  0.002609  0.004451  0.000000\n",
       "3  chr3:132499777:G:A  0.015650  0.004451  0.034720\n",
       "4  chr3:132499779:G:T  0.358300  0.359100  0.349500\n",
       "5  chr3:132502295:C:T  0.001739  0.001484  0.002315\n",
       "6  chr3:132507256:A:G  0.026960  0.022260  0.037040\n",
       "7  chr3:132516789:G:C  0.004371  0.004478  0.004651\n",
       "8  chr3:132523636:G:C  0.000871  0.001488  0.000000\n",
       "9  chr3:132538225:G:A  0.000000  0.000000  0.000000"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_freq = pd.merge(freq, freq_cc, on=\"SNP\", how=\"left\")\n",
    "all_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>C</td>\n",
       "      <td>0.012330</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>A</td>\n",
       "      <td>0.581800</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>0.12850</td>\n",
       "      <td>3.1820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.002609</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.165000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.015650</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.03577</td>\n",
       "      <td>0.4319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.358300</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>G</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>T</td>\n",
       "      <td>0.747100</td>\n",
       "      <td>1.0420</td>\n",
       "      <td>0.80970</td>\n",
       "      <td>1.3420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "      <td>C</td>\n",
       "      <td>0.750900</td>\n",
       "      <td>0.6404</td>\n",
       "      <td>0.03995</td>\n",
       "      <td>10.2700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "      <td>G</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "      <td>A</td>\n",
       "      <td>0.146200</td>\n",
       "      <td>0.5918</td>\n",
       "      <td>0.28950</td>\n",
       "      <td>1.2100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>0.004371</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>G</td>\n",
       "      <td>0.966700</td>\n",
       "      <td>0.9625</td>\n",
       "      <td>0.16020</td>\n",
       "      <td>5.7840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "      <td>0.000871</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>0.422500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF     MAF_A     MAF_U A1       F_A       F_U A2  \\\n",
       "0  chr3:132475009:C:A  0.003478  0.000000  0.009259  A  0.000000  0.009259  C   \n",
       "1  chr3:132494190:A:G  0.005217  0.004451  0.006944  G  0.004451  0.006944  A   \n",
       "2  chr3:132499231:C:T  0.002609  0.004451  0.000000  T  0.004451  0.000000  C   \n",
       "3  chr3:132499777:G:A  0.015650  0.004451  0.034720  A  0.004451  0.034720  G   \n",
       "4  chr3:132499779:G:T  0.358300  0.359100  0.349500  G  0.359100  0.349500  T   \n",
       "5  chr3:132502295:C:T  0.001739  0.001484  0.002315  T  0.001484  0.002315  C   \n",
       "6  chr3:132507256:A:G  0.026960  0.022260  0.037040  G  0.022260  0.037040  A   \n",
       "7  chr3:132516789:G:C  0.004371  0.004478  0.004651  C  0.004478  0.004651  G   \n",
       "8  chr3:132523636:G:C  0.000871  0.001488  0.000000  C  0.001488  0.000000  G   \n",
       "9  chr3:132538225:G:A  0.000000  0.000000  0.000000  A  0.000000  0.000000  G   \n",
       "\n",
       "          P      OR      L95      U95  \n",
       "0  0.012330  0.0000  0.00000      NaN  \n",
       "1  0.581800  0.6393  0.12850   3.1820  \n",
       "2  0.165000     NaN      NaN      NaN  \n",
       "3  0.000104  0.1243  0.03577   0.4319  \n",
       "4  0.747100  1.0420  0.80970   1.3420  \n",
       "5  0.750900  0.6404  0.03995  10.2700  \n",
       "6  0.146200  0.5918  0.28950   1.2100  \n",
       "7  0.966700  0.9625  0.16020   5.7840  \n",
       "8  0.422500     NaN      NaN      NaN  \n",
       "9       NaN     NaN      NaN      NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_and_freq = pd.merge(all_freq, assoc, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>C</td>\n",
       "      <td>0.012330</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>A</td>\n",
       "      <td>0.581800</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>0.12850</td>\n",
       "      <td>3.1820</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>0.002609</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.165000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>0.015650</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.03577</td>\n",
       "      <td>0.4319</td>\n",
       "      <td>0.000933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>0.358300</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>G</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>T</td>\n",
       "      <td>0.747100</td>\n",
       "      <td>1.0420</td>\n",
       "      <td>0.80970</td>\n",
       "      <td>1.3420</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP       MAF     MAF_A     MAF_U A1       F_A       F_U A2  \\\n",
       "0  chr3:132475009:C:A  0.003478  0.000000  0.009259  A  0.000000  0.009259  C   \n",
       "1  chr3:132494190:A:G  0.005217  0.004451  0.006944  G  0.004451  0.006944  A   \n",
       "2  chr3:132499231:C:T  0.002609  0.004451  0.000000  T  0.004451  0.000000  C   \n",
       "3  chr3:132499777:G:A  0.015650  0.004451  0.034720  A  0.004451  0.034720  G   \n",
       "4  chr3:132499779:G:T  0.358300  0.359100  0.349500  G  0.359100  0.349500  T   \n",
       "\n",
       "          P      OR      L95     U95      BONF  \n",
       "0  0.012330  0.0000  0.00000     NaN  0.110900  \n",
       "1  0.581800  0.6393  0.12850  3.1820  1.000000  \n",
       "2  0.165000     NaN      NaN     NaN  1.000000  \n",
       "3  0.000104  0.1243  0.03577  0.4319  0.000933  \n",
       "4  0.747100  1.0420  0.80970  1.3420  1.000000  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the results\n",
    "assoc_and_freq = pd.merge(assoc_and_freq, assoc_adjusted, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 202407_BURDEN_SAS_R7/chr3_SAS_release7\n",
      "  --extract range 202407_BURDEN_SAS_R7/SAS_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 202407_BURDEN_SAS_R7/SAS.samplestoKeep\n",
      "  --out 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn\n",
      "  --recode A\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "1959689 variants loaded from .bim file.\n",
      "576 people (376 males, 200 females) loaded from .fam.\n",
      "554 phenotype values loaded from .fam.\n",
      "--extract range: 1959679 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 575 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 575 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99513.\n",
      "10 variants and 575 people pass filters and QC.\n",
      "Among remaining phenotypes, 337 are cases and 216 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "--recode A to 202407_BURDEN_SAS_R7/SAS_DNAJC13.coding_nonsyn.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_release7 \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in SAS for DNAJC13: 10\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['chr3:132475009:C:A_A',\n",
       " 'chr3:132494190:A:G_G',\n",
       " 'chr3:132499231:C:T_T',\n",
       " 'chr3:132499777:G:A_A',\n",
       " 'chr3:132499779:G:T_G',\n",
       " 'chr3:132502295:C:T_T',\n",
       " 'chr3:132507256:A:G_G',\n",
       " 'chr3:132516789:G:C_C',\n",
       " 'chr3:132523636:G:C_C',\n",
       " 'chr3:132538225:G:A_A']"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for DNAJC13: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T_G</td>\n",
       "      <td>43</td>\n",
       "      <td>156</td>\n",
       "      <td>337</td>\n",
       "      <td>24</td>\n",
       "      <td>103</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132502295:C:T_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132507256:A:G_G</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132516789:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132523636:G:C_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132538225:G:A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Variant  Hom Cases  Het Cases  Total Cases  Hom Controls  \\\n",
       "0  chr3:132475009:C:A_A          0          0          337             0   \n",
       "1  chr3:132494190:A:G_G          0          3          337             0   \n",
       "2  chr3:132499231:C:T_T          0          3          337             0   \n",
       "3  chr3:132499777:G:A_A          0          3          337             1   \n",
       "4  chr3:132499779:G:T_G         43        156          337            24   \n",
       "5  chr3:132502295:C:T_T          0          1          337             0   \n",
       "6  chr3:132507256:A:G_G          0         15          337             1   \n",
       "7  chr3:132516789:G:C_C          0          3          337             0   \n",
       "8  chr3:132523636:G:C_C          0          1          337             0   \n",
       "9  chr3:132538225:G:A_A          0          0          337             0   \n",
       "\n",
       "   Het Controls  Total Controls                 SNP  \n",
       "0             4             216  chr3:132475009:C:A  \n",
       "1             3             216  chr3:132494190:A:G  \n",
       "2             0             216  chr3:132499231:C:T  \n",
       "3            13             216  chr3:132499777:G:A  \n",
       "4           103             216  chr3:132499779:G:T  \n",
       "5             1             216  chr3:132502295:C:T  \n",
       "6            14             216  chr3:132507256:A:G  \n",
       "7             2             216  chr3:132516789:G:C  \n",
       "8             0             216  chr3:132523636:G:C  \n",
       "9             0             216  chr3:132538225:G:A  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "#    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "#    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "#        'Freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "#        'Freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Merge with assoc results \n",
    "full_results = pd.merge(assoc_and_freq, df_results, on=\"SNP\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_full_results = full_results[['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR','L95', 'U95', 'BONF', \n",
    "                                   'MAF', 'MAF_A', 'MAF_U', \n",
    "                                   'Hom Cases', 'Het Cases', 'Total Cases', \n",
    "                                   'Hom Controls','Het Controls', 'Total Controls']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10, 19)\n",
      "No. of SNPs: 10\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>0.012330</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110900</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr3:132494190:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>0.581800</td>\n",
       "      <td>0.6393</td>\n",
       "      <td>0.12850</td>\n",
       "      <td>3.1820</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr3:132499231:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.165000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.002609</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.03577</td>\n",
       "      <td>0.4319</td>\n",
       "      <td>0.000933</td>\n",
       "      <td>0.015650</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:132499779:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>0.747100</td>\n",
       "      <td>1.0420</td>\n",
       "      <td>0.80970</td>\n",
       "      <td>1.3420</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.358300</td>\n",
       "      <td>0.359100</td>\n",
       "      <td>0.349500</td>\n",
       "      <td>43</td>\n",
       "      <td>156</td>\n",
       "      <td>337</td>\n",
       "      <td>24</td>\n",
       "      <td>103</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr3:132502295:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "      <td>0.750900</td>\n",
       "      <td>0.6404</td>\n",
       "      <td>0.03995</td>\n",
       "      <td>10.2700</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.001739</td>\n",
       "      <td>0.001484</td>\n",
       "      <td>0.002315</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr3:132507256:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "      <td>0.146200</td>\n",
       "      <td>0.5918</td>\n",
       "      <td>0.28950</td>\n",
       "      <td>1.2100</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.026960</td>\n",
       "      <td>0.022260</td>\n",
       "      <td>0.037040</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr3:132516789:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>0.966700</td>\n",
       "      <td>0.9625</td>\n",
       "      <td>0.16020</td>\n",
       "      <td>5.7840</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.004371</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3:132523636:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.422500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000871</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr3:132538225:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP A1 A2       F_A       F_U         P      OR      L95  \\\n",
       "0  chr3:132475009:C:A  A  C  0.000000  0.009259  0.012330  0.0000  0.00000   \n",
       "1  chr3:132494190:A:G  G  A  0.004451  0.006944  0.581800  0.6393  0.12850   \n",
       "2  chr3:132499231:C:T  T  C  0.004451  0.000000  0.165000     NaN      NaN   \n",
       "3  chr3:132499777:G:A  A  G  0.004451  0.034720  0.000104  0.1243  0.03577   \n",
       "4  chr3:132499779:G:T  G  T  0.359100  0.349500  0.747100  1.0420  0.80970   \n",
       "5  chr3:132502295:C:T  T  C  0.001484  0.002315  0.750900  0.6404  0.03995   \n",
       "6  chr3:132507256:A:G  G  A  0.022260  0.037040  0.146200  0.5918  0.28950   \n",
       "7  chr3:132516789:G:C  C  G  0.004478  0.004651  0.966700  0.9625  0.16020   \n",
       "8  chr3:132523636:G:C  C  G  0.001488  0.000000  0.422500     NaN      NaN   \n",
       "9  chr3:132538225:G:A  A  G  0.000000  0.000000       NaN     NaN      NaN   \n",
       "\n",
       "       U95      BONF       MAF     MAF_A     MAF_U  Hom Cases  Het Cases  \\\n",
       "0      NaN  0.110900  0.003478  0.000000  0.009259          0          0   \n",
       "1   3.1820  1.000000  0.005217  0.004451  0.006944          0          3   \n",
       "2      NaN  1.000000  0.002609  0.004451  0.000000          0          3   \n",
       "3   0.4319  0.000933  0.015650  0.004451  0.034720          0          3   \n",
       "4   1.3420  1.000000  0.358300  0.359100  0.349500         43        156   \n",
       "5  10.2700  1.000000  0.001739  0.001484  0.002315          0          1   \n",
       "6   1.2100  1.000000  0.026960  0.022260  0.037040          0         15   \n",
       "7   5.7840  1.000000  0.004371  0.004478  0.004651          0          3   \n",
       "8      NaN  1.000000  0.000871  0.001488  0.000000          0          1   \n",
       "9      NaN       NaN  0.000000  0.000000  0.000000          0          0   \n",
       "\n",
       "   Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "0          337             0             4             216  \n",
       "1          337             0             3             216  \n",
       "2          337             0             0             216  \n",
       "3          337             1            13             216  \n",
       "4          337            24           103             216  \n",
       "5          337             0             1             216  \n",
       "6          337             1            14             216  \n",
       "7          337             0             2             216  \n",
       "8          337             0             0             216  \n",
       "9          337             0             0             216  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clean_full_results.shape)\n",
    "print(f'No. of SNPs: {clean_full_results.shape[0]}')\n",
    "clean_full_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr3:132475009:C:A', 'chr3:132499777:G:A']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_freq = clean_full_results[clean_full_results['P']<0.05]\n",
    "sig_snps = sig_freq['SNP'].tolist()\n",
    "sig_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3:132475009:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>0.012330</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110900</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009259</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr3:132499777:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.1243</td>\n",
       "      <td>0.03577</td>\n",
       "      <td>0.4319</td>\n",
       "      <td>0.000933</td>\n",
       "      <td>0.015650</td>\n",
       "      <td>0.004451</td>\n",
       "      <td>0.034720</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  SNP A1 A2       F_A       F_U         P      OR      L95  \\\n",
       "0  chr3:132475009:C:A  A  C  0.000000  0.009259  0.012330  0.0000  0.00000   \n",
       "3  chr3:132499777:G:A  A  G  0.004451  0.034720  0.000104  0.1243  0.03577   \n",
       "\n",
       "      U95      BONF       MAF     MAF_A     MAF_U  Hom Cases  Het Cases  \\\n",
       "0     NaN  0.110900  0.003478  0.000000  0.009259          0          0   \n",
       "3  0.4319  0.000933  0.015650  0.004451  0.034720          0          3   \n",
       "\n",
       "   Total Cases  Hom Controls  Het Controls  Total Controls  \n",
       "0          337             0             4             216  \n",
       "3          337             1            13             216  "
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_df_results = clean_full_results[clean_full_results['SNP'].isin(sig_snps)]\n",
    "sig_df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\t132475009\tchr3:132475009:C:A\tC\tA\tTYPED;IMPUTED;AF=0.00318583;MAF=0.00318583;AVG_CS=0.999964;R2=0.988774;ER2=0.777405\n",
      "3\t132499777\tchr3:132499777:G:A\tG\tA\tTYPED;IMPUTED;AF=0.0150827;MAF=0.0150827;AVG_CS=0.999878;R2=0.991941;ER2=0.809756\n"
     ]
    }
   ],
   "source": [
    "for snp in sig_snps:\n",
    "    ! grep {snp} {WORK_DIR}/chr3_{ancestry}_release7.pvar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save files to VM\n",
    "clean_full_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt', sep=\"\\t\", index=False)\n",
    "sig_df_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt' , sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/{WORKSPACE_DIR}/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
