{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variants in DNAJC13 and their Association with Parkinson's Disease Across Different Ancestral Backgrounds\n",
    "\n",
    "* **Project**: DNAJC13 Gene Assesment\n",
    "* **Last updated**: September 2024\n",
    "* **Version**: Python 3.9\n",
    "* **Data**: AMP-PD Release 3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "This notebook is focused on analysing association between DNAJC13 variants in European (EUR) ancestry and Parkinson's disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_103/3943855956.py:26: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## AMP-PD v2.5\n",
    "## Explicitly define release v2.5 path \n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_GATK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "print('AMP-PD v2.5')\n",
    "print(f'Path to AMP-PD v2.5 Clinical Data @ `AMP_CLINICAL_RELEASE_PATH`: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v2.5 WGS Data @ `AMP_WGS_RELEASE_PLINK_PATH`: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v6.0')\n",
    "print(f'Path to GP2 v6.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v6.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v6.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v6.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = \"EUR\"\n",
    "WORK_DIR = f'2024_BURDEN_AMP_{ancestry}'\n",
    "! mkdir {WORK_DIR}\n",
    "\n",
    "AMP_DATA_DIR = f'AMP_DATA_DIR'\n",
    "! mkdir {AMP_DATA_DIR}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR: Download resources for annotation\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## chr 3\n",
    "#shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_WGS_RELEASE_PLINK_PATH}/pfiles/chr3.* {AMP_DATA_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_WGS_RELEASE_PLINK_PATH}/pfiles/chr3.pvar {AMP_DATA_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## I have already downloaded it in a different folder. Just linking\n",
    "#shell_do(f'ln -s /home/jupyter/Team 5 DNAJC13 Gene Assessment/edit/2024_BURDEN_AMP/chr3.* {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## demographics\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_CLINICAL_RELEASE_PATH}/Demographics.csv {WORK_DIR}')\n",
    "\n",
    "## enrollment\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_CLINICAL_RELEASE_PATH}/Enrollment.csv {WORK_DIR}')\n",
    "\n",
    "## duplicates\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_RELEASE_PATH}/amp_pd_participant_wgs_duplicates.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "#shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/release6_{ancestry}.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# refFlat\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {WORKSPACE_BUCKET}/notebooks/misc/refFlat_HG38_all_chr.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Predicted ancestries uploaded by Mary\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMPPD_v25_GenoTools_Predictions.txt {WORK_DIR}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Additional files uploaded by Mary\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/PCA.FILTERED.AMP_PD_{ancestry}.PD.eigenvec {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/toRemove_1stand2ndDegree_Relateds_{ancestry}.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "amp_pd_participant_wgs_duplicates.csv\r\n",
      "AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "chr3.pgen\r\n",
      "chr3.psam\r\n",
      "chr3.pvar\r\n",
      "Demographics.csv\r\n",
      "Enrollment.csv\r\n",
      "PCA.FILTERED.AMP_PD_EUR.PD.eigenvec\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "toRemove_1stand2ndDegree_Relateds_EUR.txt\r\n"
     ]
    }
   ],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Covariate File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the disease status\n",
    "enrollment = pd.read_csv(f'{WORK_DIR}/Enrollment.csv')\n",
    "print(enrollment.shape)\n",
    "enrollment.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the demographic data\n",
    "demographics = pd.read_csv(f'{WORK_DIR}/Demographics.csv')\n",
    "print(demographics.shape)\n",
    "demographics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the predicted ancestry\n",
    "predictedAncestry= pd.read_csv(f'{WORK_DIR}/AMPPD_v25_GenoTools_Predictions.txt', sep='\\s+' )\n",
    "print(predictedAncestry.shape)\n",
    "predictedAncestry.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get just the needed info from Enrollment\n",
    "enrollment = enrollment[['participant_id', 'study_arm']]\n",
    "enrollment.rename(columns = {'participant_id': 'IID',\n",
    "                      'study_arm':'phenotype'}, inplace = True)\n",
    "print(enrollment.shape)\n",
    "enrollment.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get just the needed info from Demographics\n",
    "demographics = demographics[['participant_id', 'sex', 'age_at_baseline']]\n",
    "demographics.rename(columns = {'participant_id': 'IID',\n",
    "                      'sex':'SEX',\n",
    "                      'age_at_baseline':'AGE'}, inplace = True)\n",
    "print(demographics.shape)\n",
    "demographics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Enrollment with Demographics\n",
    "merged_key = pd.merge(enrollment, demographics, on='IID', how='left')\n",
    "print(merged_key.shape)\n",
    "merged_key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Enrollment/Demographics with Ancestry\n",
    "merged_key2 = pd.merge(merged_key, predictedAncestry, on='IID', how='left')\n",
    "print(merged_key2.shape)\n",
    "merged_key2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "label\n",
       "EUR    8370\n",
       "AJ     1312\n",
       "NaN     316\n",
       "AAC     106\n",
       "AMR      86\n",
       "MDE      42\n",
       "EAS      33\n",
       "CAS      15\n",
       "AFR      13\n",
       "SAS      10\n",
       "FIN       2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_key2['label'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phenotype\n",
       "Healthy Control                3377\n",
       "PD                             2987\n",
       "LBD                            2521\n",
       "Genetic Cohort Unaffected       403\n",
       "Genetic Cohort PD               272\n",
       "Genetic Registry Unaffected     245\n",
       "Genetic Registry PD             196\n",
       "Disease Control                 159\n",
       "SWEDD                            77\n",
       "Prodromal                        64\n",
       "NaN                               3\n",
       "Unknown                           1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_key2['phenotype'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_key2[merged_key2['label']==ancestry]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = merged_key2[merged_key2['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (0/1)\n",
    "    # PD = 1; control = 0\n",
    "pheno_mapping = {\"PD\": 2, \"Healthy Control\": 1}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "<NA>    2938\n",
       "1       2920\n",
       "2       2512\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert sex to binary (0/1)\n",
    "    # Female = 1; male = 0\n",
    "sex_mapping = {'Female': 2, 'Male': 1}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SEX\n",
       "1       4812\n",
       "2       3558\n",
       "<NA>       0\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check value counts of SEX\n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = ancestry_key[['FID','IID', 'SEX', 'PHENO', 'AGE']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove duplicate values in IID column\n",
    "final_df = final_df.drop_duplicates(subset=['IID'], keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO  SEX\n",
       "1      2      1504\n",
       "       1      1416\n",
       "2      1      1445\n",
       "       2       835\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6818/2709626079.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  final_df.dropna(inplace=True)\n"
     ]
    }
   ],
   "source": [
    "final_df.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(final_df.shape)\n",
    "final_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals \n",
    "column_names = ['FID','IID']\n",
    "related_df = pd.read_csv(f'{WORK_DIR}/toRemove_1stand2ndDegree_Relateds_{ancestry}.txt', sep='\\t', header = None, names=column_names)\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "final_df = final_df[~final_df[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(final_df.shape)\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID','IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! head {WORK_DIR}/{ancestry}.samplestoKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#final_df['FID'] = 0\n",
    "final_df['FATID'] = 0\n",
    "final_df['MATID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df2=final_df[['FID','IID','MATID','FATID','SEX','PHENO', 'AGE']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/PCA.FILTERED.AMP_PD_{ancestry}.PD.eigenvec', sep='\\t')\n",
    "print(pcs.shape)\n",
    "pcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Cov and PCs\n",
    "final_df3 = pd.merge(final_df2, pcs, on='IID', how='left')\n",
    "print(final_df3.shape)\n",
    "final_df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df3.drop(['#FID', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10'], axis='columns', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(final_df3.shape)\n",
    "final_df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO  SEX\n",
       "1      2      1459\n",
       "       1      1379\n",
       "2      1      1442\n",
       "       2       835\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df3.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df3.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!head {WORK_DIR}/{ancestry}_covariate_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 23M\r\n",
      "-rw-rw-r-- 1 jupyter users 1.9K May  5 15:43 amp_pd_participant_wgs_duplicates.csv\r\n",
      "-rw-rw-r-- 1 jupyter users 262K May  5 15:43 AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "-rw-rw-r-- 1 jupyter users 924K May  5 15:43 AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "lrwxrwxrwx 1 jupyter users   25 May  5 15:43 chr3.pgen -> ../AMP_DATA_DIR/chr3.pgen\r\n",
      "lrwxrwxrwx 1 jupyter users   25 May  5 15:43 chr3.psam -> ../AMP_DATA_DIR/chr3.psam\r\n",
      "lrwxrwxrwx 1 jupyter users   25 May  5 15:43 chr3.pvar -> ../AMP_DATA_DIR/chr3.pvar\r\n",
      "-rw-rw-r-- 1 jupyter users 636K May  5 15:43 Demographics.csv\r\n",
      "-rw-rw-r-- 1 jupyter users 633K May  5 15:43 Enrollment.csv\r\n",
      "-rw-rw-r-- 1 jupyter users 448K May  5 15:43 EUR_covariate_file.txt\r\n",
      "-rw-rw-r-- 1 jupyter users 121K May  5 15:43 EUR.samplestoKeep\r\n",
      "-rw-rw-r-- 1 jupyter users 667K May  5 15:43 PCA.FILTERED.AMP_PD_EUR.PD.eigenvec\r\n",
      "-rw-rw-r-- 1 jupyter users  20M May  5 15:43 refFlat_HG38_all_chr.txt\r\n",
      "-rw-rw-r-- 1 jupyter users  12K May  5 15:43 toRemove_1stand2ndDegree_Relateds_EUR.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls -lh {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a5.10LM AVX2 Intel (5 Jan 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_EUR/chr3_EUR.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --from-bp 132417502\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\n",
      "  --make-pgen\n",
      "  --out 2024_BURDEN_AMP_EUR/chr3_EUR\n",
      "  --pfile 2024_BURDEN_AMP_EUR/chr3\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Sun May  5 15:43:56 2024\n",
      "7450 MiB RAM detected, ~5378 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "10418 samples (0 females, 0 males, 10418 ambiguous; 10418 founders) loaded from\n",
      "2024_BURDEN_AMP_EUR/chr3.psam.\n",
      "10972242 variants loaded from 2024_BURDEN_AMP_EUR/chr3.pvar.\n",
      "Note: No phenotype data present.\n",
      "--keep: 5115 samples remaining.\n",
      "5115 samples (0 females, 0 males, 5115 ambiguous; 5115 founders) remaining\n",
      "after main filters.\n",
      "6005 variants remaining after main filters.\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.psam ... done.\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.pvar ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.pgen ... done.\n",
      "End time: Sun May  5 15:50:57 2024\n"
     ]
    }
   ],
   "source": [
    "## extract ancestry\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {WORK_DIR}/chr3 \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--make-pgen \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket) \n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_covariate_file.txt')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}.samplestoKeep')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation using ANNOVAR\n",
    "\n",
    "* *DNAJC13* from NCBI gene\n",
    "* hg38 (chr3:132417502-132539032)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a5.10LM AVX2 Intel (5 Jan 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --export vcf id-paste=iid\n",
      "  --from-bp 132417502\n",
      "  --mac 2\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13\n",
      "  --pfile 2024_BURDEN_AMP_EUR/chr3_EUR\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Sun May  5 15:56:58 2024\n",
      "7450 MiB RAM detected, ~5386 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "5115 samples (0 females, 0 males, 5115 ambiguous; 5115 founders) loaded from\n",
      "2024_BURDEN_AMP_EUR/chr3_EUR.psam.\n",
      "6005 variants loaded from 2024_BURDEN_AMP_EUR/chr3_EUR.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "4360 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1645 variants remaining after main filters.\n",
      "--export vcf to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.vcf ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Sun May  5 15:56:58 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 2024_BURDEN_AMP_EUR/EUR_DNAJC13.vcf.gz > 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.avinput>\n",
      "NOTICE: Finished reading 1706 lines from VCF file\n",
      "NOTICE: A total of 1645 locus in VCF file passed QC threshold, representing 1430 SNPs (941 transitions and 489 transversions) and 500 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 7314450 SNP genotypes (4813215 transitions and 2501235 transversions) and 2557500 indels/substitutions for 5115 samples\n",
      "WARNING: 172 invalid alternative alleles found in input file\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.refGene -exonsort -nofirstcodondel 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.refGene.variant_function, 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2102 unique variants in 2102 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2102 unique variants in 2102 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 6\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_DNAJC13.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5118</th>\n",
       "      <th>Otherinfo5119</th>\n",
       "      <th>Otherinfo5120</th>\n",
       "      <th>Otherinfo5121</th>\n",
       "      <th>Otherinfo5122</th>\n",
       "      <th>Otherinfo5123</th>\n",
       "      <th>Otherinfo5124</th>\n",
       "      <th>Otherinfo5125</th>\n",
       "      <th>Otherinfo5126</th>\n",
       "      <th>Otherinfo5127</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132417567</td>\n",
       "      <td>132417567</td>\n",
       "      <td>-</td>\n",
       "      <td>GGAGGAGGC</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16984_-16983insGGAGGAGGC;NM_01...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132417567</td>\n",
       "      <td>132417567</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16984G&gt;0;NM_015268:c.-16984G&gt;0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132417589</td>\n",
       "      <td>132417589</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16962G&gt;A;NM_015268:c.-16962G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132417590</td>\n",
       "      <td>132417590</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16961A&gt;G;NM_015268:c.-16961A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132417594</td>\n",
       "      <td>132417594</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.-16957A&gt;G;NM_015268:c.-16957A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2097</th>\n",
       "      <td>3</td>\n",
       "      <td>132538679</td>\n",
       "      <td>132538679</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*397A&gt;T;NM_015268:c.*397A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2098</th>\n",
       "      <td>3</td>\n",
       "      <td>132538708</td>\n",
       "      <td>132538708</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*426G&gt;A;NM_015268:c.*426G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2099</th>\n",
       "      <td>3</td>\n",
       "      <td>132538723</td>\n",
       "      <td>132538723</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*441C&gt;G;NM_015268:c.*441C&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2100</th>\n",
       "      <td>3</td>\n",
       "      <td>132538890</td>\n",
       "      <td>132538890</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*608G&gt;A;NM_015268:c.*608G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2101</th>\n",
       "      <td>3</td>\n",
       "      <td>132538903</td>\n",
       "      <td>132538903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*621G&gt;A;NM_015268:c.*621G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2102 rows × 5138 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref        Alt Func.refGene Gene.refGene  \\\n",
       "0       3  132417567  132417567   -  GGAGGAGGC         UTR5      DNAJC13   \n",
       "1       3  132417567  132417567   G          0         UTR5      DNAJC13   \n",
       "2       3  132417589  132417589   G          A         UTR5      DNAJC13   \n",
       "3       3  132417590  132417590   A          G         UTR5      DNAJC13   \n",
       "4       3  132417594  132417594   A          G         UTR5      DNAJC13   \n",
       "...   ...        ...        ...  ..        ...          ...          ...   \n",
       "2097    3  132538679  132538679   A          T         UTR3      DNAJC13   \n",
       "2098    3  132538708  132538708   G          A         UTR3      DNAJC13   \n",
       "2099    3  132538723  132538723   C          G         UTR3      DNAJC13   \n",
       "2100    3  132538890  132538890   G          A         UTR3      DNAJC13   \n",
       "2101    3  132538903  132538903   G          A         UTR3      DNAJC13   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0     NM_001329126:c.-16984_-16983insGGAGGAGGC;NM_01...                  .   \n",
       "1        NM_001329126:c.-16984G>0;NM_015268:c.-16984G>0                  .   \n",
       "2        NM_001329126:c.-16962G>A;NM_015268:c.-16962G>A                  .   \n",
       "3        NM_001329126:c.-16961A>G;NM_015268:c.-16961A>G                  .   \n",
       "4        NM_001329126:c.-16957A>G;NM_015268:c.-16957A>G                  .   \n",
       "...                                                 ...                ...   \n",
       "2097         NM_001329126:c.*397A>T;NM_015268:c.*397A>T                  .   \n",
       "2098         NM_001329126:c.*426G>A;NM_015268:c.*426G>A                  .   \n",
       "2099         NM_001329126:c.*441C>G;NM_015268:c.*441C>G                  .   \n",
       "2100         NM_001329126:c.*608G>A;NM_015268:c.*608G>A                  .   \n",
       "2101         NM_001329126:c.*621G>A;NM_015268:c.*621G>A                  .   \n",
       "\n",
       "     AAChange.refGene  ... Otherinfo5118  Otherinfo5119  Otherinfo5120  \\\n",
       "0                   .  ...           0/0            0/0            0/0   \n",
       "1                   .  ...           0/0            0/0            0/0   \n",
       "2                   .  ...           0/0            0/0            0/0   \n",
       "3                   .  ...           0/0            0/0            0/0   \n",
       "4                   .  ...           0/0            0/0            0/0   \n",
       "...               ...  ...           ...            ...            ...   \n",
       "2097                .  ...           0/0            0/0            0/0   \n",
       "2098                .  ...           0/0            0/0            0/0   \n",
       "2099                .  ...           0/0            0/0            0/0   \n",
       "2100                .  ...           0/0            0/0            0/0   \n",
       "2101                .  ...           0/0            0/1            0/0   \n",
       "\n",
       "      Otherinfo5121  Otherinfo5122  Otherinfo5123 Otherinfo5124 Otherinfo5125  \\\n",
       "0               0/0            0/0            0/0           0/0           0/0   \n",
       "1               0/0            0/0            0/0           0/0           0/0   \n",
       "2               0/0            0/0            0/0           0/0           0/0   \n",
       "3               0/0            0/0            0/0           0/0           0/0   \n",
       "4               0/0            0/0            0/0           0/0           0/0   \n",
       "...             ...            ...            ...           ...           ...   \n",
       "2097            0/0            0/0            0/0           0/0           0/0   \n",
       "2098            0/0            0/0            0/0           0/0           0/0   \n",
       "2099            0/0            0/0            0/0           0/0           0/0   \n",
       "2100            0/0            0/0            0/0           0/0           0/0   \n",
       "2101            0/0            0/1            0/1           0/0           0/0   \n",
       "\n",
       "     Otherinfo5126  Otherinfo5127  \n",
       "0              0/0            0/0  \n",
       "1              0/0            0/0  \n",
       "2              0/0            0/0  \n",
       "3              0/0            0/0  \n",
       "4              0/0            0/0  \n",
       "...            ...            ...  \n",
       "2097           0/0            0/0  \n",
       "2098           0/0            0/0  \n",
       "2099           0/0            0/0  \n",
       "2100           0/0            0/0  \n",
       "2101           0/0            0/1  \n",
       "\n",
       "[2102 rows x 5138 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter intronic\n",
    "intronic = gene[(gene['Func.refGene'] == 'intronic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR3\n",
    "utr3 = gene[(gene['Func.refGene'] == 'UTR3')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR5\n",
    "utr5 = gene[(gene['Func.refGene'] == 'UTR5')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter exonic and synonymous variants\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo5118</th>\n",
       "      <th>Otherinfo5119</th>\n",
       "      <th>Otherinfo5120</th>\n",
       "      <th>Otherinfo5121</th>\n",
       "      <th>Otherinfo5122</th>\n",
       "      <th>Otherinfo5123</th>\n",
       "      <th>Otherinfo5124</th>\n",
       "      <th>Otherinfo5125</th>\n",
       "      <th>Otherinfo5126</th>\n",
       "      <th>Otherinfo5127</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>525</th>\n",
       "      <td>3</td>\n",
       "      <td>132450732</td>\n",
       "      <td>132450732</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon6:c.A422T:p.N141I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>3</td>\n",
       "      <td>132456751</td>\n",
       "      <td>132456751</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon12:c.C1268T:p.A423V,D...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>3</td>\n",
       "      <td>132456763</td>\n",
       "      <td>132456763</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon12:c.G1280C:p.S427T,D...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666</th>\n",
       "      <td>3</td>\n",
       "      <td>132461159</td>\n",
       "      <td>132461159</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon15:c.T1667C:p.L556S,D...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>743</th>\n",
       "      <td>3</td>\n",
       "      <td>132466338</td>\n",
       "      <td>132466338</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon19:c.G2008A:p.V670I,D...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1077</th>\n",
       "      <td>3</td>\n",
       "      <td>132473173</td>\n",
       "      <td>132473173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon21:c.G2237A:p.R746Q,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>3</td>\n",
       "      <td>132477841</td>\n",
       "      <td>132477841</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon23:c.G2498A:p.R833K,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1183</th>\n",
       "      <td>3</td>\n",
       "      <td>132478139</td>\n",
       "      <td>132478139</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon24:c.G2708A:p.R903K,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>3</td>\n",
       "      <td>132480415</td>\n",
       "      <td>132480415</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon26:c.T2819C:p.V940A,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>3</td>\n",
       "      <td>132480415</td>\n",
       "      <td>132480415</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon26:c.T2819G:p.V940G,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>3</td>\n",
       "      <td>132484650</td>\n",
       "      <td>132484650</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon29:c.C3245T:p.T1082I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1329</th>\n",
       "      <td>3</td>\n",
       "      <td>132488428</td>\n",
       "      <td>132488428</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon30:c.G3398A:p.G1133D,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>3</td>\n",
       "      <td>132492497</td>\n",
       "      <td>132492497</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon33:c.A3707G:p.Y1236C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>3</td>\n",
       "      <td>132492604</td>\n",
       "      <td>132492604</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon33:c.A3814G:p.I1272V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1408</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1475</th>\n",
       "      <td>3</td>\n",
       "      <td>132499168</td>\n",
       "      <td>132499168</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.G4199A:p.R1400Q,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1485</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1486</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1500</th>\n",
       "      <td>3</td>\n",
       "      <td>132500821</td>\n",
       "      <td>132500821</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon39:c.A4444G:p.S1482G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1521</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1522</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1586</th>\n",
       "      <td>3</td>\n",
       "      <td>132505378</td>\n",
       "      <td>132505378</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon42:c.A4961C:p.E1654A,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1587</th>\n",
       "      <td>3</td>\n",
       "      <td>132505383</td>\n",
       "      <td>132505383</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon42:c.C4966A:p.L1656I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1692</th>\n",
       "      <td>3</td>\n",
       "      <td>132511169</td>\n",
       "      <td>132511169</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon44:c.G5218C:p.E1740Q,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1713</th>\n",
       "      <td>3</td>\n",
       "      <td>132513022</td>\n",
       "      <td>132513022</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon45:c.T5308G:p.C1770G,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1850</th>\n",
       "      <td>3</td>\n",
       "      <td>132522838</td>\n",
       "      <td>132522838</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon49:c.C5684T:p.T1895M,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1851</th>\n",
       "      <td>3</td>\n",
       "      <td>132522950</td>\n",
       "      <td>132522950</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon49:c.T5796G:p.D1932E,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1858</th>\n",
       "      <td>3</td>\n",
       "      <td>132523636</td>\n",
       "      <td>132523636</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon51:c.G5983C:p.V1995L,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1884</th>\n",
       "      <td>3</td>\n",
       "      <td>132525718</td>\n",
       "      <td>132525718</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon52:c.G6169T:p.A2057S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>3</td>\n",
       "      <td>132528264</td>\n",
       "      <td>132528264</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon54:c.G6457A:p.A2153T,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1912</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2089</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35 rows × 5138 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "525     3  132450732  132450732   A   T       exonic      DNAJC13   \n",
       "602     3  132456751  132456751   C   T       exonic      DNAJC13   \n",
       "604     3  132456763  132456763   G   C       exonic      DNAJC13   \n",
       "666     3  132461159  132461159   T   C       exonic      DNAJC13   \n",
       "743     3  132466338  132466338   G   A       exonic      DNAJC13   \n",
       "1077    3  132473173  132473173   G   A       exonic      DNAJC13   \n",
       "1138    3  132475009  132475009   C   A       exonic      DNAJC13   \n",
       "1178    3  132477841  132477841   G   A       exonic      DNAJC13   \n",
       "1183    3  132478139  132478139   G   A       exonic      DNAJC13   \n",
       "1224    3  132480415  132480415   T   C       exonic      DNAJC13   \n",
       "1225    3  132480415  132480415   T   G       exonic      DNAJC13   \n",
       "1269    3  132484650  132484650   C   T       exonic      DNAJC13   \n",
       "1329    3  132488428  132488428   G   A       exonic      DNAJC13   \n",
       "1376    3  132492497  132492497   A   G       exonic      DNAJC13   \n",
       "1377    3  132492604  132492604   A   G       exonic      DNAJC13   \n",
       "1408    3  132494190  132494190   A   G       exonic      DNAJC13   \n",
       "1475    3  132499168  132499168   G   A       exonic      DNAJC13   \n",
       "1476    3  132499231  132499231   C   T       exonic      DNAJC13   \n",
       "1485    3  132499777  132499777   G   A       exonic      DNAJC13   \n",
       "1486    3  132499779  132499779   G   T       exonic      DNAJC13   \n",
       "1500    3  132500821  132500821   A   G       exonic      DNAJC13   \n",
       "1521    3  132502295  132502295   C   T       exonic      DNAJC13   \n",
       "1522    3  132502299  132502299   G   A       exonic      DNAJC13   \n",
       "1586    3  132505378  132505378   A   C       exonic      DNAJC13   \n",
       "1587    3  132505383  132505383   C   A       exonic      DNAJC13   \n",
       "1642    3  132507256  132507256   A   G       exonic      DNAJC13   \n",
       "1692    3  132511169  132511169   G   C       exonic      DNAJC13   \n",
       "1713    3  132513022  132513022   T   G       exonic      DNAJC13   \n",
       "1850    3  132522838  132522838   C   T       exonic      DNAJC13   \n",
       "1851    3  132522950  132522950   T   G       exonic      DNAJC13   \n",
       "1858    3  132523636  132523636   G   C       exonic      DNAJC13   \n",
       "1884    3  132525718  132525718   G   T       exonic      DNAJC13   \n",
       "1911    3  132528264  132528264   G   A       exonic      DNAJC13   \n",
       "1912    3  132528316  132528316   T   G       exonic      DNAJC13   \n",
       "2089    3  132538225  132538225   G   A       exonic      DNAJC13   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "525                   .  nonsynonymous SNV   \n",
       "602                   .  nonsynonymous SNV   \n",
       "604                   .  nonsynonymous SNV   \n",
       "666                   .  nonsynonymous SNV   \n",
       "743                   .  nonsynonymous SNV   \n",
       "1077                  .  nonsynonymous SNV   \n",
       "1138                  .  nonsynonymous SNV   \n",
       "1178                  .  nonsynonymous SNV   \n",
       "1183                  .  nonsynonymous SNV   \n",
       "1224                  .  nonsynonymous SNV   \n",
       "1225                  .  nonsynonymous SNV   \n",
       "1269                  .  nonsynonymous SNV   \n",
       "1329                  .  nonsynonymous SNV   \n",
       "1376                  .  nonsynonymous SNV   \n",
       "1377                  .  nonsynonymous SNV   \n",
       "1408                  .  nonsynonymous SNV   \n",
       "1475                  .  nonsynonymous SNV   \n",
       "1476                  .  nonsynonymous SNV   \n",
       "1485                  .  nonsynonymous SNV   \n",
       "1486                  .  nonsynonymous SNV   \n",
       "1500                  .  nonsynonymous SNV   \n",
       "1521                  .  nonsynonymous SNV   \n",
       "1522                  .  nonsynonymous SNV   \n",
       "1586                  .  nonsynonymous SNV   \n",
       "1587                  .  nonsynonymous SNV   \n",
       "1642                  .  nonsynonymous SNV   \n",
       "1692                  .  nonsynonymous SNV   \n",
       "1713                  .  nonsynonymous SNV   \n",
       "1850                  .  nonsynonymous SNV   \n",
       "1851                  .  nonsynonymous SNV   \n",
       "1858                  .  nonsynonymous SNV   \n",
       "1884                  .  nonsynonymous SNV   \n",
       "1911                  .  nonsynonymous SNV   \n",
       "1912                  .  nonsynonymous SNV   \n",
       "2089                  .  nonsynonymous SNV   \n",
       "\n",
       "                                       AAChange.refGene  ... Otherinfo5118  \\\n",
       "525   DNAJC13:NM_001329126:exon6:c.A422T:p.N141I,DNA...  ...           0/0   \n",
       "602   DNAJC13:NM_001329126:exon12:c.C1268T:p.A423V,D...  ...           0/0   \n",
       "604   DNAJC13:NM_001329126:exon12:c.G1280C:p.S427T,D...  ...           0/0   \n",
       "666   DNAJC13:NM_001329126:exon15:c.T1667C:p.L556S,D...  ...           0/0   \n",
       "743   DNAJC13:NM_001329126:exon19:c.G2008A:p.V670I,D...  ...           0/0   \n",
       "1077  DNAJC13:NM_015268:exon21:c.G2237A:p.R746Q,DNAJ...  ...           0/0   \n",
       "1138  DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...  ...           0/0   \n",
       "1178  DNAJC13:NM_015268:exon23:c.G2498A:p.R833K,DNAJ...  ...           0/0   \n",
       "1183  DNAJC13:NM_015268:exon24:c.G2708A:p.R903K,DNAJ...  ...           0/0   \n",
       "1224  DNAJC13:NM_015268:exon26:c.T2819C:p.V940A,DNAJ...  ...           0/0   \n",
       "1225  DNAJC13:NM_015268:exon26:c.T2819G:p.V940G,DNAJ...  ...           0/0   \n",
       "1269  DNAJC13:NM_015268:exon29:c.C3245T:p.T1082I,DNA...  ...           0/0   \n",
       "1329  DNAJC13:NM_015268:exon30:c.G3398A:p.G1133D,DNA...  ...           0/0   \n",
       "1376  DNAJC13:NM_015268:exon33:c.A3707G:p.Y1236C,DNA...  ...           0/0   \n",
       "1377  DNAJC13:NM_015268:exon33:c.A3814G:p.I1272V,DNA...  ...           0/0   \n",
       "1408  DNAJC13:NM_015268:exon34:c.A3872G:p.E1291G,DNA...  ...           0/0   \n",
       "1475  DNAJC13:NM_015268:exon37:c.G4199A:p.R1400Q,DNA...  ...           0/0   \n",
       "1476  DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...  ...           0/0   \n",
       "1485  DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...  ...           0/0   \n",
       "1486  DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...  ...           1/1   \n",
       "1500  DNAJC13:NM_015268:exon39:c.A4444G:p.S1482G,DNA...  ...           0/0   \n",
       "1521  DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...  ...           0/0   \n",
       "1522  DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...  ...           0/0   \n",
       "1586  DNAJC13:NM_015268:exon42:c.A4961C:p.E1654A,DNA...  ...           0/0   \n",
       "1587  DNAJC13:NM_015268:exon42:c.C4966A:p.L1656I,DNA...  ...           0/0   \n",
       "1642  DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...  ...           0/1   \n",
       "1692  DNAJC13:NM_015268:exon44:c.G5218C:p.E1740Q,DNA...  ...           0/0   \n",
       "1713  DNAJC13:NM_015268:exon45:c.T5308G:p.C1770G,DNA...  ...           0/0   \n",
       "1850  DNAJC13:NM_015268:exon49:c.C5684T:p.T1895M,DNA...  ...           0/0   \n",
       "1851  DNAJC13:NM_015268:exon49:c.T5796G:p.D1932E,DNA...  ...           0/0   \n",
       "1858  DNAJC13:NM_015268:exon51:c.G5983C:p.V1995L,DNA...  ...           0/0   \n",
       "1884  DNAJC13:NM_015268:exon52:c.G6169T:p.A2057S,DNA...  ...           0/0   \n",
       "1911  DNAJC13:NM_015268:exon54:c.G6457A:p.A2153T,DNA...  ...           0/0   \n",
       "1912  DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...  ...           0/0   \n",
       "2089  DNAJC13:NM_015268:exon56:c.G6675A:p.M2225I,DNA...  ...           0/0   \n",
       "\n",
       "      Otherinfo5119  Otherinfo5120  Otherinfo5121  Otherinfo5122  \\\n",
       "525             0/0            0/0            0/0            0/0   \n",
       "602             0/0            0/0            0/0            0/0   \n",
       "604             0/0            0/0            0/0            0/0   \n",
       "666             0/0            0/0            0/0            0/0   \n",
       "743             0/0            0/0            0/0            0/0   \n",
       "1077            0/0            0/0            0/0            0/0   \n",
       "1138            0/0            0/0            0/0            0/0   \n",
       "1178            0/0            0/0            0/0            0/0   \n",
       "1183            0/0            0/0            0/0            0/0   \n",
       "1224            0/0            0/0            0/0            0/0   \n",
       "1225            0/0            0/0            0/0            0/0   \n",
       "1269            0/0            0/0            0/0            0/0   \n",
       "1329            0/0            0/0            0/0            0/0   \n",
       "1376            0/0            0/0            0/0            0/0   \n",
       "1377            0/0            0/0            0/0            0/0   \n",
       "1408            0/0            0/0            0/0            0/0   \n",
       "1475            0/0            0/0            0/0            0/0   \n",
       "1476            0/0            0/0            0/0            0/0   \n",
       "1485            0/0            0/0            0/0            0/0   \n",
       "1486            1/1            0/1            0/1            0/1   \n",
       "1500            0/0            0/0            0/0            0/0   \n",
       "1521            0/1            0/0            0/0            0/1   \n",
       "1522            0/0            0/0            0/0            0/0   \n",
       "1586            0/0            0/0            0/0            0/0   \n",
       "1587            0/0            0/0            0/0            0/0   \n",
       "1642            0/0            0/1            0/0            0/0   \n",
       "1692            0/0            0/0            0/0            0/0   \n",
       "1713            0/0            0/0            0/0            0/0   \n",
       "1850            0/0            0/0            0/0            0/0   \n",
       "1851            0/0            0/0            0/0            0/0   \n",
       "1858            0/0            0/0            0/0            0/0   \n",
       "1884            0/0            0/0            0/0            0/0   \n",
       "1911            0/0            0/0            0/0            0/0   \n",
       "1912            0/0            0/0            0/0            0/0   \n",
       "2089            0/0            0/0            0/0            0/0   \n",
       "\n",
       "      Otherinfo5123 Otherinfo5124 Otherinfo5125 Otherinfo5126  Otherinfo5127  \n",
       "525             0/0           0/0           0/0           0/0            0/0  \n",
       "602             0/0           0/0           0/0           0/0            0/0  \n",
       "604             0/0           0/0           0/0           0/0            0/0  \n",
       "666             0/0           0/0           0/0           0/0            0/0  \n",
       "743             0/0           0/0           0/0           0/0            0/0  \n",
       "1077            0/0           0/0           0/0           0/0            0/0  \n",
       "1138            0/0           0/0           0/0           0/0            0/0  \n",
       "1178            0/0           0/0           0/0           0/0            0/0  \n",
       "1183            0/0           0/0           0/0           0/0            0/0  \n",
       "1224            0/0           0/0           0/0           0/0            0/0  \n",
       "1225            0/0           0/0           0/0           0/0            0/0  \n",
       "1269            0/0           0/0           0/0           0/0            0/0  \n",
       "1329            0/0           0/0           0/0           0/0            0/0  \n",
       "1376            0/0           0/0           0/0           0/0            0/0  \n",
       "1377            0/0           0/0           0/0           0/0            0/0  \n",
       "1408            0/0           0/0           0/0           0/0            0/0  \n",
       "1475            0/0           0/0           0/0           0/0            0/0  \n",
       "1476            0/0           0/0           0/0           0/0            0/0  \n",
       "1485            0/0           0/0           0/0           0/0            0/0  \n",
       "1486            0/1           0/1           0/0           0/0            0/1  \n",
       "1500            0/0           0/0           0/0           0/0            0/0  \n",
       "1521            0/0           0/0           0/0           0/0            0/0  \n",
       "1522            0/0           0/0           0/0           0/0            0/0  \n",
       "1586            0/0           0/0           0/0           0/0            0/0  \n",
       "1587            0/0           0/0           0/0           0/0            0/0  \n",
       "1642            0/0           0/0           0/0           0/0            0/0  \n",
       "1692            0/0           0/1           0/0           0/0            0/0  \n",
       "1713            0/0           0/0           0/0           0/0            0/0  \n",
       "1850            0/0           0/0           0/0           0/0            0/0  \n",
       "1851            0/0           0/0           0/0           0/0            0/0  \n",
       "1858            0/0           0/0           0/0           0/0            0/0  \n",
       "1884            0/0           0/0           0/0           0/0            0/0  \n",
       "1911            0/0           0/0           0/0           0/0            0/0  \n",
       "1912            0/0           0/0           0/0           0/0            0/0  \n",
       "2089            0/0           0/0           0/0           0/0            0/0  \n",
       "\n",
       "[35 rows x 5138 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "totalVariants = len(gene.axes[0])\n",
    "totalIntronic = len(intronic.axes[0])\n",
    "totalUTR3 = len(utr3.axes[0])\n",
    "totalUTR5 = len(utr5.axes[0])\n",
    "totalExonicSyn = len(coding_synonymous.axes[0])\n",
    "totalExonicNonSyn = len(coding_nonsynonymous.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Variants:  2102\n",
      "Intronic: 2032\n",
      "UTR3: 12\n",
      "UTR5: 5\n",
      "Exonic Syn: 18\n",
      "Exonic NonSyn: 35\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Variants: \", totalVariants)\n",
    "print(\"Intronic:\", totalIntronic)\n",
    "print(\"UTR3:\", totalUTR3)\n",
    "print(\"UTR5:\", totalUTR5)\n",
    "print(\"Exonic Syn:\", totalExonicSyn)\n",
    "print(\"Exonic NonSyn:\", totalExonicNonSyn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>525</th>\n",
       "      <td>3</td>\n",
       "      <td>132450732</td>\n",
       "      <td>132450732</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>3</td>\n",
       "      <td>132456751</td>\n",
       "      <td>132456751</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>604</th>\n",
       "      <td>3</td>\n",
       "      <td>132456763</td>\n",
       "      <td>132456763</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666</th>\n",
       "      <td>3</td>\n",
       "      <td>132461159</td>\n",
       "      <td>132461159</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>743</th>\n",
       "      <td>3</td>\n",
       "      <td>132466338</td>\n",
       "      <td>132466338</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1077</th>\n",
       "      <td>3</td>\n",
       "      <td>132473173</td>\n",
       "      <td>132473173</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1178</th>\n",
       "      <td>3</td>\n",
       "      <td>132477841</td>\n",
       "      <td>132477841</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1183</th>\n",
       "      <td>3</td>\n",
       "      <td>132478139</td>\n",
       "      <td>132478139</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1224</th>\n",
       "      <td>3</td>\n",
       "      <td>132480415</td>\n",
       "      <td>132480415</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1225</th>\n",
       "      <td>3</td>\n",
       "      <td>132480415</td>\n",
       "      <td>132480415</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>3</td>\n",
       "      <td>132484650</td>\n",
       "      <td>132484650</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1329</th>\n",
       "      <td>3</td>\n",
       "      <td>132488428</td>\n",
       "      <td>132488428</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376</th>\n",
       "      <td>3</td>\n",
       "      <td>132492497</td>\n",
       "      <td>132492497</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1377</th>\n",
       "      <td>3</td>\n",
       "      <td>132492604</td>\n",
       "      <td>132492604</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1408</th>\n",
       "      <td>3</td>\n",
       "      <td>132494190</td>\n",
       "      <td>132494190</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1475</th>\n",
       "      <td>3</td>\n",
       "      <td>132499168</td>\n",
       "      <td>132499168</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1485</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1486</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1500</th>\n",
       "      <td>3</td>\n",
       "      <td>132500821</td>\n",
       "      <td>132500821</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1521</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1522</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1586</th>\n",
       "      <td>3</td>\n",
       "      <td>132505378</td>\n",
       "      <td>132505378</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1587</th>\n",
       "      <td>3</td>\n",
       "      <td>132505383</td>\n",
       "      <td>132505383</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1692</th>\n",
       "      <td>3</td>\n",
       "      <td>132511169</td>\n",
       "      <td>132511169</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1713</th>\n",
       "      <td>3</td>\n",
       "      <td>132513022</td>\n",
       "      <td>132513022</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1850</th>\n",
       "      <td>3</td>\n",
       "      <td>132522838</td>\n",
       "      <td>132522838</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1851</th>\n",
       "      <td>3</td>\n",
       "      <td>132522950</td>\n",
       "      <td>132522950</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1858</th>\n",
       "      <td>3</td>\n",
       "      <td>132523636</td>\n",
       "      <td>132523636</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1884</th>\n",
       "      <td>3</td>\n",
       "      <td>132525718</td>\n",
       "      <td>132525718</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>3</td>\n",
       "      <td>132528264</td>\n",
       "      <td>132528264</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1912</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2089</th>\n",
       "      <td>3</td>\n",
       "      <td>132538225</td>\n",
       "      <td>132538225</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Gene.refGene\n",
       "525     3  132450732  132450732      DNAJC13\n",
       "602     3  132456751  132456751      DNAJC13\n",
       "604     3  132456763  132456763      DNAJC13\n",
       "666     3  132461159  132461159      DNAJC13\n",
       "743     3  132466338  132466338      DNAJC13\n",
       "1077    3  132473173  132473173      DNAJC13\n",
       "1138    3  132475009  132475009      DNAJC13\n",
       "1178    3  132477841  132477841      DNAJC13\n",
       "1183    3  132478139  132478139      DNAJC13\n",
       "1224    3  132480415  132480415      DNAJC13\n",
       "1225    3  132480415  132480415      DNAJC13\n",
       "1269    3  132484650  132484650      DNAJC13\n",
       "1329    3  132488428  132488428      DNAJC13\n",
       "1376    3  132492497  132492497      DNAJC13\n",
       "1377    3  132492604  132492604      DNAJC13\n",
       "1408    3  132494190  132494190      DNAJC13\n",
       "1475    3  132499168  132499168      DNAJC13\n",
       "1476    3  132499231  132499231      DNAJC13\n",
       "1485    3  132499777  132499777      DNAJC13\n",
       "1486    3  132499779  132499779      DNAJC13\n",
       "1500    3  132500821  132500821      DNAJC13\n",
       "1521    3  132502295  132502295      DNAJC13\n",
       "1522    3  132502299  132502299      DNAJC13\n",
       "1586    3  132505378  132505378      DNAJC13\n",
       "1587    3  132505383  132505383      DNAJC13\n",
       "1642    3  132507256  132507256      DNAJC13\n",
       "1692    3  132511169  132511169      DNAJC13\n",
       "1713    3  132513022  132513022      DNAJC13\n",
       "1850    3  132522838  132522838      DNAJC13\n",
       "1851    3  132522950  132522950      DNAJC13\n",
       "1858    3  132523636  132523636      DNAJC13\n",
       "1884    3  132525718  132525718      DNAJC13\n",
       "1911    3  132528264  132528264      DNAJC13\n",
       "1912    3  132528316  132528316      DNAJC13\n",
       "2089    3  132538225  132538225      DNAJC13"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "amp_pd_participant_wgs_duplicates.csv\r\n",
      "AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "chr3_EUR.log\r\n",
      "chr3_EUR.pgen\r\n",
      "chr3_EUR.psam\r\n",
      "chr3_EUR.pvar\r\n",
      "chr3.pgen\r\n",
      "chr3.psam\r\n",
      "chr3.pvar\r\n",
      "Demographics.csv\r\n",
      "Enrollment.csv\r\n",
      "EUR_covariate_file.txt\r\n",
      "EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "EUR_DNAJC13.annovar.avinput\r\n",
      "EUR_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "EUR_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "EUR_DNAJC13.log\r\n",
      "EUR_DNAJC13.vcf.gz\r\n",
      "EUR_DNAJC13.vcf.gz.tbi\r\n",
      "EUR.samplestoKeep\r\n",
      "PCA.FILTERED.AMP_PD_EUR.PD.eigenvec\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "toRemove_1stand2ndDegree_Relateds_EUR.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket) (not copying right now)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a5.10LM AVX2 Intel (5 Jan 2024)     www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/chr3_EUR.log.\r\n",
      "Options in effect:\r\n",
      "  --make-bed\r\n",
      "  --max-alleles 2\r\n",
      "  --out 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "  --pfile 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "\r\n",
      "Start time: Sun May  5 15:58:45 2024\r\n",
      "7450 MiB RAM detected, ~5083 available; reserving 3725 MiB for main workspace.\r\n",
      "Using up to 2 compute threads.\r\n",
      "5115 samples (0 females, 0 males, 5115 ambiguous; 5115 founders) loaded from\r\n",
      "2024_BURDEN_AMP_EUR/chr3_EUR.psam.\r\n",
      "5615 out of 6005 variants loaded from 2024_BURDEN_AMP_EUR/chr3_EUR.pvar.\r\n",
      "Note: No phenotype data present.\r\n",
      "5615 variants remaining after main filters.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.fam ... done.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.bim ... done.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR.bed ... 0%\b\bdone.\r\n",
      "End time: Sun May  5 15:58:45 2024\r\n"
     ]
    }
   ],
   "source": [
    "# Convert the files from Plink 2.0 to Plink 1.9 format \n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--make-bed \\\n",
    "--max-alleles 2 \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\n",
      "  --recode vcf-iid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "5615 variants loaded from .bim file.\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\n",
      "--extract range: 5582 variants excluded.\n",
      "--extract range: 33 variants remaining.\n",
      "--keep: 5115 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "33 variants and 5115 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode vcf-iid to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "    # later do based on class and frequency\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--recode vcf-iid \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [2024_BURDEN_AMP_EUR/EUR_DNAJC13.burden.coding_nonsyn]\n",
      "                          --outputRaw\n",
      "       Specify Covariate: --covar [], --covar-name [], --sex\n",
      "       Specify Phenotype: --pheno [2024_BURDEN_AMP_EUR/EUR_covariate_file.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skat\n",
      "                         skato], --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [2024_BURDEN_AMP_EUR/refFlat_HG38_all_chr.txt]\n",
      "                          --gene [DNAJC13], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.vcf.gz\n",
      "    --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.burden.coding_nonsyn\n",
      "    --pheno 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skat,skato\n",
      "    --geneFile 2024_BURDEN_AMP_EUR/refFlat_HG38_all_chr.txt\n",
      "    --gene DNAJC13\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Sun May  5 15:58:50 2024\n",
      "[INFO]\tLoaded [ 5115 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 5115 ] sample phenotypes\n",
      "[INFO]\tLoaded 2821 male, 2294 female and 0 sex-unknown samples from 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\n",
      "[INFO]\tLoaded 2277 cases, 2838 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 5115 ] samples...\n",
      "[INFO]\tSKAT test significance will be evaluated using 10000 permutations at alpha = 0.05 weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 33 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Sun May  5 15:58:52 2024\n",
      "[INFO]\tAnalysis took 2 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "## RVtests with covariates \n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn \\\n",
    "--kernel skat,skato \\\n",
    "--inVcf {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf.gz \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene DNAJC13 \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38_all_chr.txt #\\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX AGE PC1 PC2 PC3 PC4\n",
    "#[WARN]Covariate [ PC5 ] has strong correlation [ r^2 = 1 ] with the response!\n",
    "\n",
    "\n",
    "# --out : Name of output \n",
    "# --burden cmc --kernel skato: tests to run \n",
    "# --inVcf : VCF file \n",
    "# --gene: gene name (if only looking at one or a few)\n",
    "# --geneFile refFlat.txt\n",
    "# --pheno :  covar file\n",
    "# --mpheno : # column that has phenotype information\n",
    "# --pheno-name : column name with phenotype in file\n",
    "# --covar : covar file\n",
    "# --freqUpper : optional, MAF cut-off\n",
    "# --covar-name : covariates, listed by column name, separated by commas (no spaces between commas)\n",
    "## 0=controls; 1=cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t5115\t33\t33\t110685\t0.26713\t10000\t3511\t110685\t1000\t0\t0.284819\r\n"
     ]
    }
   ],
   "source": [
    "## look at results \n",
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t5115\t33\t33\t54487.7\t0.4\t0.406778\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "amp_pd_participant_wgs_duplicates.csv\r\n",
      "AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "chr3_EUR.bed\r\n",
      "chr3_EUR.bim\r\n",
      "chr3_EUR.fam\r\n",
      "chr3_EUR.log\r\n",
      "chr3_EUR.pgen\r\n",
      "chr3_EUR.psam\r\n",
      "chr3_EUR.pvar\r\n",
      "chr3.pgen\r\n",
      "chr3.psam\r\n",
      "chr3.pvar\r\n",
      "Demographics.csv\r\n",
      "Enrollment.csv\r\n",
      "EUR_covariate_file.txt\r\n",
      "EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "EUR_DNAJC13.annovar.avinput\r\n",
      "EUR_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "EUR_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "EUR_DNAJC13.burden.coding_nonsyn.log\r\n",
      "EUR_DNAJC13.burden.coding_nonsyn.Skat.assoc\r\n",
      "EUR_DNAJC13.burden.coding_nonsyn.SkatO.assoc\r\n",
      "EUR_DNAJC13.coding_nonsyn.log\r\n",
      "EUR_DNAJC13.coding_nonsyn.nosex\r\n",
      "EUR_DNAJC13.coding_nonsyn.vcf.gz\r\n",
      "EUR_DNAJC13.coding_nonsyn.vcf.gz.tbi\r\n",
      "EUR_DNAJC13.log\r\n",
      "EUR_DNAJC13.vcf.gz\r\n",
      "EUR_DNAJC13.vcf.gz.tbi\r\n",
      "EUR.samplestoKeep\r\n",
      "PCA.FILTERED.AMP_PD_EUR.PD.eigenvec\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "toRemove_1stand2ndDegree_Relateds_EUR.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.*.assoc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --allow-no-sex\r\n",
      "  --assoc\r\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "  --ci 0.95\r\n",
      "  --covar 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\r\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4\r\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\r\n",
      "5115 phenotype values present after --pheno.\r\n",
      "--extract range: 5582 variants excluded.\r\n",
      "--extract range: 33 variants remaining.\r\n",
      "--keep: 5115 people remaining.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Warning: Ignoring --covar since no commands reference the covariates.\r\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "33 variants and 5115 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2277 are cases and 2838 are controls.\r\n",
      "Writing C/C --assoc report to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.assoc ...\r\n",
      "0%\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4\\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR           SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR           SE          L95          U95 \r\n",
      "   3   rs149480465  132450732    T 0.0002196 0.0003524    A       0.1519       0.6967       0.6231        1.225      0.05648        6.874 \r\n",
      "   3   rs147898644  132456751    T 0.0004392        0    C        2.493       0.1143           NA           NA           NA           NA \r\n",
      "   3             .  132456763    C        0 0.0003524    G        1.605       0.2052            0          inf            0          nan \r\n",
      "   3             .  132461159    C 0.0004392        0    T        2.493       0.1143           NA           NA           NA           NA \r\n",
      "   3   rs200491223  132466338    A        0 0.0003524    G        1.605       0.2052            0          inf            0          nan \r\n",
      "   3             .  132473173    A        0 0.0003524    G        1.605       0.2052            0          inf            0          nan \r\n",
      "   3   rs149121829  132475009    A 0.009003  0.01004    C       0.2876       0.5918       0.8956       0.2058       0.5984         1.34 \r\n",
      "   3   rs201629093  132477841    A 0.0002196 0.0003524    G       0.1519       0.6967       0.6231        1.225      0.05648        6.874 \r\n",
      "   3   rs141952333  132478139    A 0.003074 0.003171    G     0.007628       0.9304       0.9693       0.3569       0.4816        1.951 \r\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>A</td>\n",
       "      <td>0.696700</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.8740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.114300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.205200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.114300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs200491223</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.205200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.205200</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs149121829</td>\n",
       "      <td>A</td>\n",
       "      <td>0.009003</td>\n",
       "      <td>0.010040</td>\n",
       "      <td>C</td>\n",
       "      <td>0.591800</td>\n",
       "      <td>0.8956</td>\n",
       "      <td>0.59840</td>\n",
       "      <td>1.3400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs201629093</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.696700</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.8740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs141952333</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003074</td>\n",
       "      <td>0.003171</td>\n",
       "      <td>G</td>\n",
       "      <td>0.930400</td>\n",
       "      <td>0.9693</td>\n",
       "      <td>0.48160</td>\n",
       "      <td>1.9510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs202127368</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000659</td>\n",
       "      <td>0.001057</td>\n",
       "      <td>C</td>\n",
       "      <td>0.499500</td>\n",
       "      <td>0.6229</td>\n",
       "      <td>0.15570</td>\n",
       "      <td>2.4920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>G</td>\n",
       "      <td>0.876000</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.9300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs201263331</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000878</td>\n",
       "      <td>0.001409</td>\n",
       "      <td>A</td>\n",
       "      <td>0.435400</td>\n",
       "      <td>0.6229</td>\n",
       "      <td>0.18740</td>\n",
       "      <td>2.0700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs372257861</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>A</td>\n",
       "      <td>0.876000</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.9300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs61748101</td>\n",
       "      <td>G</td>\n",
       "      <td>0.005709</td>\n",
       "      <td>0.005814</td>\n",
       "      <td>A</td>\n",
       "      <td>0.944600</td>\n",
       "      <td>0.9819</td>\n",
       "      <td>0.58640</td>\n",
       "      <td>1.6440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs147104839</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000659</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>G</td>\n",
       "      <td>0.219800</td>\n",
       "      <td>3.7410</td>\n",
       "      <td>0.38900</td>\n",
       "      <td>35.9800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs61748102</td>\n",
       "      <td>T</td>\n",
       "      <td>0.014490</td>\n",
       "      <td>0.019910</td>\n",
       "      <td>C</td>\n",
       "      <td>0.037870</td>\n",
       "      <td>0.7240</td>\n",
       "      <td>0.53300</td>\n",
       "      <td>0.9833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs61748103</td>\n",
       "      <td>A</td>\n",
       "      <td>0.012740</td>\n",
       "      <td>0.011800</td>\n",
       "      <td>G</td>\n",
       "      <td>0.669800</td>\n",
       "      <td>1.0800</td>\n",
       "      <td>0.75820</td>\n",
       "      <td>1.5380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs3762672</td>\n",
       "      <td>T</td>\n",
       "      <td>0.451700</td>\n",
       "      <td>0.477600</td>\n",
       "      <td>G</td>\n",
       "      <td>0.008968</td>\n",
       "      <td>0.9010</td>\n",
       "      <td>0.83320</td>\n",
       "      <td>0.9743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>A</td>\n",
       "      <td>0.876000</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.9300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs55825559</td>\n",
       "      <td>T</td>\n",
       "      <td>0.027670</td>\n",
       "      <td>0.033120</td>\n",
       "      <td>C</td>\n",
       "      <td>0.112000</td>\n",
       "      <td>0.8307</td>\n",
       "      <td>0.66060</td>\n",
       "      <td>1.0450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs139620588</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003733</td>\n",
       "      <td>0.003347</td>\n",
       "      <td>G</td>\n",
       "      <td>0.743500</td>\n",
       "      <td>1.1160</td>\n",
       "      <td>0.57920</td>\n",
       "      <td>2.1490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs202044084</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>A</td>\n",
       "      <td>0.876000</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.9300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs141090992</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>C</td>\n",
       "      <td>0.876000</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.9300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs79953286</td>\n",
       "      <td>G</td>\n",
       "      <td>0.068510</td>\n",
       "      <td>0.060610</td>\n",
       "      <td>A</td>\n",
       "      <td>0.104800</td>\n",
       "      <td>1.1400</td>\n",
       "      <td>0.97300</td>\n",
       "      <td>1.3360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs142160751</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001537</td>\n",
       "      <td>0.001586</td>\n",
       "      <td>G</td>\n",
       "      <td>0.950800</td>\n",
       "      <td>0.9694</td>\n",
       "      <td>0.36070</td>\n",
       "      <td>2.6050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>T</td>\n",
       "      <td>0.696700</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.8740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs145242123</td>\n",
       "      <td>T</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.002114</td>\n",
       "      <td>C</td>\n",
       "      <td>0.178200</td>\n",
       "      <td>1.6640</td>\n",
       "      <td>0.78650</td>\n",
       "      <td>3.5210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs147575919</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000659</td>\n",
       "      <td>0.001057</td>\n",
       "      <td>T</td>\n",
       "      <td>0.499500</td>\n",
       "      <td>0.6229</td>\n",
       "      <td>0.15570</td>\n",
       "      <td>2.4920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs10935014</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001757</td>\n",
       "      <td>0.001409</td>\n",
       "      <td>G</td>\n",
       "      <td>0.658700</td>\n",
       "      <td>1.2470</td>\n",
       "      <td>0.46760</td>\n",
       "      <td>3.3250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs138693725</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001318</td>\n",
       "      <td>0.001762</td>\n",
       "      <td>G</td>\n",
       "      <td>0.572000</td>\n",
       "      <td>0.7475</td>\n",
       "      <td>0.27150</td>\n",
       "      <td>2.0580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.073160</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>T</td>\n",
       "      <td>0.667500</td>\n",
       "      <td>1.1390</td>\n",
       "      <td>0.62940</td>\n",
       "      <td>2.0600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs138367039</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>G</td>\n",
       "      <td>0.660300</td>\n",
       "      <td>0.8666</td>\n",
       "      <td>0.45730</td>\n",
       "      <td>1.6420</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP A1       F_A       F_U A2         P      OR      L95      U95\n",
       "0   rs149480465  T  0.000220  0.000352  A  0.696700  0.6231  0.05648   6.8740\n",
       "1   rs147898644  T  0.000439  0.000000  C  0.114300     NaN      NaN      NaN\n",
       "2             .  C  0.000000  0.000352  G  0.205200  0.0000  0.00000      NaN\n",
       "3             .  C  0.000439  0.000000  T  0.114300     NaN      NaN      NaN\n",
       "4   rs200491223  A  0.000000  0.000352  G  0.205200  0.0000  0.00000      NaN\n",
       "5             .  A  0.000000  0.000352  G  0.205200  0.0000  0.00000      NaN\n",
       "6   rs149121829  A  0.009003  0.010040  C  0.591800  0.8956  0.59840   1.3400\n",
       "7   rs201629093  A  0.000220  0.000352  G  0.696700  0.6231  0.05648   6.8740\n",
       "8   rs141952333  A  0.003074  0.003171  G  0.930400  0.9693  0.48160   1.9510\n",
       "9   rs202127368  T  0.000659  0.001057  C  0.499500  0.6229  0.15570   2.4920\n",
       "10            .  A  0.000220  0.000176  G  0.876000  1.2460  0.07794  19.9300\n",
       "11  rs201263331  G  0.000878  0.001409  A  0.435400  0.6229  0.18740   2.0700\n",
       "12  rs372257861  G  0.000220  0.000176  A  0.876000  1.2460  0.07794  19.9300\n",
       "13   rs61748101  G  0.005709  0.005814  A  0.944600  0.9819  0.58640   1.6440\n",
       "14  rs147104839  A  0.000659  0.000176  G  0.219800  3.7410  0.38900  35.9800\n",
       "15   rs61748102  T  0.014490  0.019910  C  0.037870  0.7240  0.53300   0.9833\n",
       "16   rs61748103  A  0.012740  0.011800  G  0.669800  1.0800  0.75820   1.5380\n",
       "17    rs3762672  T  0.451700  0.477600  G  0.008968  0.9010  0.83320   0.9743\n",
       "18            .  G  0.000220  0.000176  A  0.876000  1.2460  0.07794  19.9300\n",
       "19   rs55825559  T  0.027670  0.033120  C  0.112000  0.8307  0.66060   1.0450\n",
       "20  rs139620588  A  0.003733  0.003347  G  0.743500  1.1160  0.57920   2.1490\n",
       "21  rs202044084  C  0.000220  0.000176  A  0.876000  1.2460  0.07794  19.9300\n",
       "22  rs141090992  A  0.000220  0.000176  C  0.876000  1.2460  0.07794  19.9300\n",
       "23   rs79953286  G  0.068510  0.060610  A  0.104800  1.1400  0.97300   1.3360\n",
       "24  rs142160751  C  0.001537  0.001586  G  0.950800  0.9694  0.36070   2.6050\n",
       "25            .  G  0.000220  0.000352  T  0.696700  0.6231  0.05648   6.8740\n",
       "26  rs145242123  T  0.003513  0.002114  C  0.178200  1.6640  0.78650   3.5210\n",
       "27  rs147575919  G  0.000659  0.001057  T  0.499500  0.6229  0.15570   2.4920\n",
       "28   rs10935014  C  0.001757  0.001409  G  0.658700  1.2470  0.46760   3.3250\n",
       "29  rs138693725  T  0.001318  0.001762  G  0.572000  0.7475  0.27150   2.0580\n",
       "30            .  A  0.000000  0.000705  G  0.073160  0.0000  0.00000      NaN\n",
       "31  rs140537885  G  0.004611  0.004052  T  0.667500  1.1390  0.62940   2.0600\n",
       "32  rs138367039  A  0.003513  0.004052  G  0.660300  0.8666  0.45730   1.6420"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc', delim_whitespace=True, usecols=['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR', 'L95', 'U95'])\n",
    "assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --adjust\r\n",
      "  --allow-no-sex\r\n",
      "  --assoc\r\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "  --covar 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\r\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4\r\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\r\n",
      "5115 phenotype values present after --pheno.\r\n",
      "--extract range: 5582 variants excluded.\r\n",
      "--extract range: 33 variants remaining.\r\n",
      "--keep: 5115 people remaining.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Warning: Ignoring --covar since no commands reference the covariates.\r\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "33 variants and 5115 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2277 are cases and 2838 are controls.\r\n",
      "Writing C/C --assoc report to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.assoc ...\r\n",
      "0%\b\bdone.\r\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\r\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b--adjust values (33 variants) written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.assoc.adjusted .\r\n"
     ]
    }
   ],
   "source": [
    "## repeat association study applying multiple testing corrections for the raw p-values (like Bonferroni post-hoc)\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4\\\n",
    "--assoc \\\n",
    "--allow-no-sex \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR           SNP      UNADJ         GC       BONF       HOLM   SIDAK_SS   SIDAK_SD     FDR_BH     FDR_BY\r\n",
      "   3     rs3762672   0.008968   0.008968     0.2959     0.2959     0.2572     0.2572     0.2959          1 \r\n",
      "   3    rs61748102    0.03787    0.03787          1          1     0.7203     0.7093      0.539          1 \r\n",
      "   3             .    0.07316    0.07316          1          1     0.9185     0.9051      0.539          1 \r\n",
      "   3    rs79953286     0.1048     0.1048          1          1     0.9741     0.9639      0.539          1 \r\n",
      "   3    rs55825559      0.112      0.112          1          1     0.9801     0.9681      0.539          1 \r\n",
      "   3   rs147898644     0.1143     0.1143          1          1     0.9818     0.9681      0.539          1 \r\n",
      "   3             .     0.1143     0.1143          1          1     0.9818     0.9681      0.539          1 \r\n",
      "   3   rs145242123     0.1782     0.1782          1          1     0.9985     0.9939     0.6045          1 \r\n",
      "   3             .     0.2052     0.2052          1          1     0.9995     0.9968     0.6045          1 \r\n",
      "   3   rs200491223     0.2052     0.2052          1          1     0.9995     0.9968     0.6045          1 \r\n",
      "   3             .     0.2052     0.2052          1          1     0.9995     0.9968     0.6045          1 \r\n",
      "   3   rs147104839     0.2198     0.2198          1          1     0.9997     0.9968     0.6045          1 \r\n",
      "   3   rs201263331     0.4354     0.4354          1          1          1          1     0.9508          1 \r\n",
      "   3   rs202127368     0.4995     0.4995          1          1          1          1     0.9508          1 \r\n",
      "   3   rs147575919     0.4995     0.4995          1          1          1          1     0.9508          1 \r\n",
      "   3   rs138693725      0.572      0.572          1          1          1          1     0.9508          1 \r\n",
      "   3   rs149121829     0.5918     0.5918          1          1          1          1     0.9508          1 \r\n",
      "   3    rs10935014     0.6587     0.6587          1          1          1          1     0.9508          1 \r\n",
      "   3   rs138367039     0.6603     0.6603          1          1          1          1     0.9508          1 \r\n",
      "   3   rs140537885     0.6675     0.6675          1          1          1          1     0.9508          1 \r\n",
      "   3    rs61748103     0.6698     0.6698          1          1          1          1     0.9508          1 \r\n",
      "   3   rs149480465     0.6967     0.6967          1          1          1          1     0.9508          1 \r\n",
      "   3   rs201629093     0.6967     0.6967          1          1          1          1     0.9508          1 \r\n",
      "   3             .     0.6967     0.6967          1          1          1          1     0.9508          1 \r\n",
      "   3   rs139620588     0.7435     0.7435          1          1          1          1     0.9508          1 \r\n",
      "   3             .      0.876      0.876          1          1          1          1     0.9508          1 \r\n",
      "   3   rs372257861      0.876      0.876          1          1          1          1     0.9508          1 \r\n",
      "   3             .      0.876      0.876          1          1          1          1     0.9508          1 \r\n",
      "   3   rs202044084      0.876      0.876          1          1          1          1     0.9508          1 \r\n",
      "   3   rs141090992      0.876      0.876          1          1          1          1     0.9508          1 \r\n",
      "   3   rs141952333     0.9304     0.9304          1          1          1          1     0.9508          1 \r\n",
      "   3    rs61748101     0.9446     0.9446          1          1          1          1     0.9508          1 \r\n",
      "   3   rs142160751     0.9508     0.9508          1          1          1          1     0.9508          1 \r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs3762672</td>\n",
       "      <td>0.2959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs61748102</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs79953286</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs55825559</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs145242123</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs200491223</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs147104839</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs201263331</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs202127368</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs147575919</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs138693725</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs149121829</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs10935014</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>rs138367039</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs61748103</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs201629093</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs139620588</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs372257861</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs202044084</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs141090992</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>rs141952333</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs61748101</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs142160751</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP    BONF\n",
       "0     rs3762672  0.2959\n",
       "1    rs61748102  1.0000\n",
       "2             .  1.0000\n",
       "3    rs79953286  1.0000\n",
       "4    rs55825559  1.0000\n",
       "5   rs147898644  1.0000\n",
       "6             .  1.0000\n",
       "7   rs145242123  1.0000\n",
       "8             .  1.0000\n",
       "9   rs200491223  1.0000\n",
       "10            .  1.0000\n",
       "11  rs147104839  1.0000\n",
       "12  rs201263331  1.0000\n",
       "13  rs202127368  1.0000\n",
       "14  rs147575919  1.0000\n",
       "15  rs138693725  1.0000\n",
       "16  rs149121829  1.0000\n",
       "17   rs10935014  1.0000\n",
       "18  rs138367039  1.0000\n",
       "19  rs140537885  1.0000\n",
       "20   rs61748103  1.0000\n",
       "21  rs149480465  1.0000\n",
       "22  rs201629093  1.0000\n",
       "23            .  1.0000\n",
       "24  rs139620588  1.0000\n",
       "25            .  1.0000\n",
       "26  rs372257861  1.0000\n",
       "27            .  1.0000\n",
       "28  rs202044084  1.0000\n",
       "29  rs141090992  1.0000\n",
       "30  rs141952333  1.0000\n",
       "31   rs61748101  1.0000\n",
       "32  rs142160751  1.0000"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_adjusted = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted', delim_whitespace=True, usecols=['SNP', 'BONF'])\n",
    "assoc_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --freq\r\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\r\n",
      "--extract range: 5582 variants excluded.\r\n",
      "--extract range: 33 variants remaining.\r\n",
      "--keep: 5115 people remaining.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "--freq: Allele frequencies (founders only) written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.frq .\r\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR           SNP   A1   A2          MAF  NCHROBS\r\n",
      "   3   rs149480465    T    A    0.0002933    10230\r\n",
      "   3   rs147898644    T    C    0.0001955    10230\r\n",
      "   3             .    C    G    0.0001955    10230\r\n",
      "   3             .    C    T    0.0001955    10230\r\n",
      "   3   rs200491223    A    G    0.0001955    10230\r\n",
      "   3             .    A    G    0.0001955    10230\r\n",
      "   3   rs149121829    A    C      0.00958    10230\r\n",
      "   3   rs201629093    A    G    0.0002933    10230\r\n",
      "   3   rs141952333    A    G     0.003128    10230\r\n",
      "   3   rs202127368    T    C    0.0008798    10230\r\n",
      "   3             .    A    G    0.0001955    10230\r\n",
      "   3   rs201263331    G    A     0.001173    10230\r\n",
      "   3   rs372257861    G    A    0.0001955    10230\r\n",
      "   3    rs61748101    G    A     0.005767    10230\r\n",
      "   3   rs147104839    A    G     0.000391    10230\r\n",
      "   3    rs61748102    T    C       0.0175    10230\r\n",
      "   3    rs61748103    A    G      0.01222    10230\r\n",
      "   3     rs3762672    T    G       0.4661    10230\r\n",
      "   3             .    G    A    0.0001955    10230\r\n",
      "   3    rs55825559    T    C      0.03069    10230\r\n",
      "   3   rs139620588    A    G     0.003519    10230\r\n",
      "   3   rs202044084    C    A    0.0001955    10230\r\n",
      "   3   rs141090992    A    C    0.0001955    10230\r\n",
      "   3    rs79953286    G    A      0.06413    10230\r\n",
      "   3   rs142160751    C    G     0.001564    10230\r\n",
      "   3             .    G    T    0.0002933    10230\r\n",
      "   3   rs145242123    T    C     0.002737    10230\r\n",
      "   3   rs147575919    G    T    0.0008798    10230\r\n",
      "   3    rs10935014    C    G     0.001564    10230\r\n",
      "   3   rs138693725    T    G     0.001564    10230\r\n",
      "   3             .    A    G     0.000391    10230\r\n",
      "   3   rs140537885    G    T     0.004301    10230\r\n",
      "   3   rs138367039    A    G     0.003812    10230\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>0.000293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>0.000196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs200491223</td>\n",
       "      <td>0.000196</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP       MAF\n",
       "0  rs149480465  0.000293\n",
       "1  rs147898644  0.000196\n",
       "2            .  0.000196\n",
       "3            .  0.000196\n",
       "4  rs200491223  0.000196"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq', delim_whitespace=True, usecols=['SNP', 'MAF'])\n",
    "freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a5.10LM AVX2 Intel (5 Jan 2024)     www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno.log.\r\n",
      "Options in effect:\r\n",
      "  --make-bed\r\n",
      "  --max-alleles 2\r\n",
      "  --out 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno\r\n",
      "  --pfile 2024_BURDEN_AMP_EUR/chr3_EUR\r\n",
      "  --pheno 2024_BURDEN_AMP_EUR/EUR_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "Start time: Sun May  5 16:00:58 2024\r\n",
      "7450 MiB RAM detected, ~5062 available; reserving 3725 MiB for main workspace.\r\n",
      "Using up to 2 compute threads.\r\n",
      "5115 samples (0 females, 0 males, 5115 ambiguous; 5115 founders) loaded from\r\n",
      "2024_BURDEN_AMP_EUR/chr3_EUR.psam.\r\n",
      "5615 out of 6005 variants loaded from 2024_BURDEN_AMP_EUR/chr3_EUR.pvar.\r\n",
      "1 binary phenotype loaded (2277 cases, 2838 controls).\r\n",
      "5615 variants remaining after main filters.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno.fam ... done.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno.bim ... done.\r\n",
      "Writing 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno.bed ... 0%\b\bdone.\r\n",
      "End time: Sun May  5 16:00:58 2024\r\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--make-bed \\\n",
    "--max-alleles 2 \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}_wPheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --allow-no-sex\r\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno\r\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --freq case-control\r\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\r\n",
      "5115 phenotype values loaded from .fam.\r\n",
      "--extract range: 5582 variants excluded.\r\n",
      "--extract range: 33 variants remaining.\r\n",
      "--keep: 5115 people remaining.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\r\n",
      "written to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.frq.cc .\r\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_wPheno \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--freq case-control \\\n",
    "--allow-no-sex \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR           SNP   A1   A2        MAF_A        MAF_U  NCHROBS_A  NCHROBS_U\r\n",
      "   3   rs149480465    T    A    0.0002196    0.0003524       4554       5676\r\n",
      "   3   rs147898644    T    C    0.0004392            0       4554       5676\r\n",
      "   3             .    C    G            0    0.0003524       4554       5676\r\n",
      "   3             .    C    T    0.0004392            0       4554       5676\r\n",
      "   3   rs200491223    A    G            0    0.0003524       4554       5676\r\n",
      "   3             .    A    G            0    0.0003524       4554       5676\r\n",
      "   3   rs149121829    A    C     0.009003      0.01004       4554       5676\r\n",
      "   3   rs201629093    A    G    0.0002196    0.0003524       4554       5676\r\n",
      "   3   rs141952333    A    G     0.003074     0.003171       4554       5676\r\n",
      "   3   rs202127368    T    C    0.0006588     0.001057       4554       5676\r\n",
      "   3             .    A    G    0.0002196    0.0001762       4554       5676\r\n",
      "   3   rs201263331    G    A    0.0008783     0.001409       4554       5676\r\n",
      "   3   rs372257861    G    A    0.0002196    0.0001762       4554       5676\r\n",
      "   3    rs61748101    G    A     0.005709     0.005814       4554       5676\r\n",
      "   3   rs147104839    A    G    0.0006588    0.0001762       4554       5676\r\n",
      "   3    rs61748102    T    C      0.01449      0.01991       4554       5676\r\n",
      "   3    rs61748103    A    G      0.01274       0.0118       4554       5676\r\n",
      "   3     rs3762672    T    G       0.4517       0.4776       4554       5676\r\n",
      "   3             .    G    A    0.0002196    0.0001762       4554       5676\r\n",
      "   3    rs55825559    T    C      0.02767      0.03312       4554       5676\r\n",
      "   3   rs139620588    A    G     0.003733     0.003347       4554       5676\r\n",
      "   3   rs202044084    C    A    0.0002196    0.0001762       4554       5676\r\n",
      "   3   rs141090992    A    C    0.0002196    0.0001762       4554       5676\r\n",
      "   3    rs79953286    G    A      0.06851      0.06061       4554       5676\r\n",
      "   3   rs142160751    C    G     0.001537     0.001586       4554       5676\r\n",
      "   3             .    G    T    0.0002196    0.0003524       4554       5676\r\n",
      "   3   rs145242123    T    C     0.003513     0.002114       4554       5676\r\n",
      "   3   rs147575919    G    T    0.0006588     0.001057       4554       5676\r\n",
      "   3    rs10935014    C    G     0.001757     0.001409       4554       5676\r\n",
      "   3   rs138693725    T    G     0.001318     0.001762       4554       5676\r\n",
      "   3             .    A    G            0    0.0007047       4554       5676\r\n",
      "   3   rs140537885    G    T     0.004611     0.004052       4554       5676\r\n",
      "   3   rs138367039    A    G     0.003513     0.004052       4554       5676\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs200491223</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP     MAF_A     MAF_U\n",
       "0  rs149480465  0.000220  0.000352\n",
       "1  rs147898644  0.000439  0.000000\n",
       "2            .  0.000000  0.000352\n",
       "3            .  0.000439  0.000000\n",
       "4  rs200491223  0.000000  0.000352"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freq_cc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc', delim_whitespace=True, usecols=['SNP', 'MAF_A', 'MAF_U'])\n",
    "freq_cc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>0.000293</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>0.004301</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>rs138367039</td>\n",
       "      <td>0.003812</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP       MAF     MAF_A     MAF_U\n",
       "0   rs149480465  0.000293  0.000220  0.000352\n",
       "1   rs147898644  0.000196  0.000439  0.000000\n",
       "2             .  0.000196  0.000000  0.000352\n",
       "3             .  0.000196  0.000439  0.000000\n",
       "4             .  0.000196  0.000000  0.000352\n",
       "..          ...       ...       ...       ...\n",
       "70            .  0.000391  0.000220  0.000176\n",
       "71            .  0.000391  0.000220  0.000352\n",
       "72            .  0.000391  0.000000  0.000705\n",
       "73  rs140537885  0.004301  0.004611  0.004052\n",
       "74  rs138367039  0.003812  0.003513  0.004052\n",
       "\n",
       "[75 rows x 4 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_freq = pd.merge(freq, freq_cc, on=\"SNP\", how=\"left\")\n",
    "all_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>0.000293</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>A</td>\n",
       "      <td>0.69670</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.11430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.20520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.11430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.20520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>A</td>\n",
       "      <td>0.87600</td>\n",
       "      <td>1.2460</td>\n",
       "      <td>0.07794</td>\n",
       "      <td>19.930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>T</td>\n",
       "      <td>0.69670</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>G</td>\n",
       "      <td>0.07316</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>0.004301</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>T</td>\n",
       "      <td>0.66750</td>\n",
       "      <td>1.1390</td>\n",
       "      <td>0.62940</td>\n",
       "      <td>2.060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>rs138367039</td>\n",
       "      <td>0.003812</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>G</td>\n",
       "      <td>0.66030</td>\n",
       "      <td>0.8666</td>\n",
       "      <td>0.45730</td>\n",
       "      <td>1.642</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             SNP       MAF     MAF_A     MAF_U A1       F_A       F_U A2  \\\n",
       "0    rs149480465  0.000293  0.000220  0.000352  T  0.000220  0.000352  A   \n",
       "1    rs147898644  0.000196  0.000439  0.000000  T  0.000439  0.000000  C   \n",
       "2              .  0.000196  0.000000  0.000352  C  0.000000  0.000352  G   \n",
       "3              .  0.000196  0.000000  0.000352  C  0.000439  0.000000  T   \n",
       "4              .  0.000196  0.000000  0.000352  A  0.000000  0.000352  G   \n",
       "..           ...       ...       ...       ... ..       ...       ... ..   \n",
       "364            .  0.000391  0.000000  0.000705  G  0.000220  0.000176  A   \n",
       "365            .  0.000391  0.000000  0.000705  G  0.000220  0.000352  T   \n",
       "366            .  0.000391  0.000000  0.000705  A  0.000000  0.000705  G   \n",
       "367  rs140537885  0.004301  0.004611  0.004052  G  0.004611  0.004052  T   \n",
       "368  rs138367039  0.003812  0.003513  0.004052  A  0.003513  0.004052  G   \n",
       "\n",
       "           P      OR      L95     U95  \n",
       "0    0.69670  0.6231  0.05648   6.874  \n",
       "1    0.11430     NaN      NaN     NaN  \n",
       "2    0.20520  0.0000  0.00000     NaN  \n",
       "3    0.11430     NaN      NaN     NaN  \n",
       "4    0.20520  0.0000  0.00000     NaN  \n",
       "..       ...     ...      ...     ...  \n",
       "364  0.87600  1.2460  0.07794  19.930  \n",
       "365  0.69670  0.6231  0.05648   6.874  \n",
       "366  0.07316  0.0000  0.00000     NaN  \n",
       "367  0.66750  1.1390  0.62940   2.060  \n",
       "368  0.66030  0.8666  0.45730   1.642  \n",
       "\n",
       "[369 rows x 12 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_and_freq = pd.merge(all_freq, assoc, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>0.000293</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>A</td>\n",
       "      <td>0.6967</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.874</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2052</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP       MAF     MAF_A     MAF_U A1       F_A       F_U A2  \\\n",
       "0  rs149480465  0.000293  0.000220  0.000352  T  0.000220  0.000352  A   \n",
       "1  rs147898644  0.000196  0.000439  0.000000  T  0.000439  0.000000  C   \n",
       "2            .  0.000196  0.000000  0.000352  C  0.000000  0.000352  G   \n",
       "3            .  0.000196  0.000000  0.000352  C  0.000000  0.000352  G   \n",
       "4            .  0.000196  0.000000  0.000352  C  0.000000  0.000352  G   \n",
       "\n",
       "        P      OR      L95    U95  BONF  \n",
       "0  0.6967  0.6231  0.05648  6.874   1.0  \n",
       "1  0.1143     NaN      NaN    NaN   1.0  \n",
       "2  0.2052  0.0000  0.00000    NaN   1.0  \n",
       "3  0.2052  0.0000  0.00000    NaN   1.0  \n",
       "4  0.2052  0.0000  0.00000    NaN   1.0  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the results\n",
    "assoc_and_freq = pd.merge(assoc_and_freq, assoc_adjusted, on=\"SNP\", how=\"left\")\n",
    "assoc_and_freq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile 2024_BURDEN_AMP_EUR/chr3_EUR_wPheno\r\n",
      "  --extract range 2024_BURDEN_AMP_EUR/EUR_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --keep 2024_BURDEN_AMP_EUR/EUR.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn\r\n",
      "  --recode A\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "5115 people (0 males, 0 females, 5115 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.nosex .\r\n",
      "5115 phenotype values loaded from .fam.\r\n",
      "--extract range: 5582 variants excluded.\r\n",
      "--extract range: 33 variants remaining.\r\n",
      "--keep: 5115 people remaining.\r\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\r\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Before main variant filters, 5115 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "33 variants and 5115 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 2277 are cases and 2838 are controls.\r\n",
      "--recode A to 2024_BURDEN_AMP_EUR/EUR_DNAJC13.coding_nonsyn.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry}_wPheno \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in EUR for DNAJC13: 33\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['rs149480465_T',\n",
       " 'rs147898644_T',\n",
       " '._C',\n",
       " '._C.1',\n",
       " 'rs200491223_A',\n",
       " '._A',\n",
       " 'rs149121829_A',\n",
       " 'rs201629093_A',\n",
       " 'rs141952333_A',\n",
       " 'rs202127368_T',\n",
       " '._A.1',\n",
       " 'rs201263331_G',\n",
       " 'rs372257861_G',\n",
       " 'rs61748101_G',\n",
       " 'rs147104839_A',\n",
       " 'rs61748102_T',\n",
       " 'rs61748103_A',\n",
       " 'rs3762672_T',\n",
       " '._G',\n",
       " 'rs55825559_T',\n",
       " 'rs139620588_A',\n",
       " 'rs202044084_C',\n",
       " 'rs141090992_A',\n",
       " 'rs79953286_G',\n",
       " 'rs142160751_C',\n",
       " '._G.1',\n",
       " 'rs145242123_T',\n",
       " 'rs147575919_G',\n",
       " 'rs10935014_C',\n",
       " 'rs138693725_T',\n",
       " '._A.2',\n",
       " 'rs140537885_G',\n",
       " 'rs138367039_A']"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for DNAJC13: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465_T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs149480465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644_T</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs147898644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>._C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>._C.1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs200491223_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs200491223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>._A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs149121829_A</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>57</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs149121829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs201629093_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs201629093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs141952333_A</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs141952333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs202127368_T</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs202127368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>._A.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs201263331_G</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs201263331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs372257861_G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs372257861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs61748101_G</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs61748101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs147104839_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs147104839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs61748102_T</td>\n",
       "      <td>1</td>\n",
       "      <td>64</td>\n",
       "      <td>2277</td>\n",
       "      <td>2</td>\n",
       "      <td>109</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs61748102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs61748103_A</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>2277</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs61748103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs3762672_T</td>\n",
       "      <td>458</td>\n",
       "      <td>1141</td>\n",
       "      <td>2277</td>\n",
       "      <td>663</td>\n",
       "      <td>1385</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs3762672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>._G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs55825559_T</td>\n",
       "      <td>1</td>\n",
       "      <td>124</td>\n",
       "      <td>2277</td>\n",
       "      <td>5</td>\n",
       "      <td>178</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs55825559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs139620588_A</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs139620588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs202044084_C</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs202044084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs141090992_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs141090992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs79953286_G</td>\n",
       "      <td>3</td>\n",
       "      <td>306</td>\n",
       "      <td>2277</td>\n",
       "      <td>12</td>\n",
       "      <td>320</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs79953286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs142160751_C</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs142160751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>._G.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs145242123_T</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs145242123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs147575919_G</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs147575919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs10935014_C</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs10935014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs138693725_T</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs138693725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>._A.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2838</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs140537885_G</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs140537885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs138367039_A</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>2838</td>\n",
       "      <td>rs138367039</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Variant  Hom Cases  Het Cases  Total Cases  Hom Controls  \\\n",
       "0   rs149480465_T          0          1         2277             0   \n",
       "1   rs147898644_T          0          2         2277             0   \n",
       "2             ._C          0          0         2277             0   \n",
       "3           ._C.1          0          2         2277             0   \n",
       "4   rs200491223_A          0          0         2277             0   \n",
       "5             ._A          0          0         2277             0   \n",
       "6   rs149121829_A          0         41         2277             0   \n",
       "7   rs201629093_A          0          1         2277             0   \n",
       "8   rs141952333_A          0         14         2277             0   \n",
       "9   rs202127368_T          0          3         2277             0   \n",
       "10          ._A.1          0          1         2277             0   \n",
       "11  rs201263331_G          0          4         2277             0   \n",
       "12  rs372257861_G          0          1         2277             0   \n",
       "13   rs61748101_G          0         26         2277             0   \n",
       "14  rs147104839_A          0          3         2277             0   \n",
       "15   rs61748102_T          1         64         2277             2   \n",
       "16   rs61748103_A          0         58         2277             1   \n",
       "17    rs3762672_T        458       1141         2277           663   \n",
       "18            ._G          0          1         2277             0   \n",
       "19   rs55825559_T          1        124         2277             5   \n",
       "20  rs139620588_A          0         17         2277             0   \n",
       "21  rs202044084_C          0          1         2277             0   \n",
       "22  rs141090992_A          0          1         2277             0   \n",
       "23   rs79953286_G          3        306         2277            12   \n",
       "24  rs142160751_C          0          7         2277             0   \n",
       "25          ._G.1          0          1         2277             0   \n",
       "26  rs145242123_T          0         16         2277             0   \n",
       "27  rs147575919_G          0          3         2277             0   \n",
       "28   rs10935014_C          0          8         2277             0   \n",
       "29  rs138693725_T          0          6         2277             0   \n",
       "30          ._A.2          0          0         2277             0   \n",
       "31  rs140537885_G          0         21         2277             0   \n",
       "32  rs138367039_A          0         16         2277             0   \n",
       "\n",
       "    Het Controls  Total Controls          SNP  \n",
       "0              2            2838  rs149480465  \n",
       "1              0            2838  rs147898644  \n",
       "2              2            2838            .  \n",
       "3              0            2838            .  \n",
       "4              2            2838  rs200491223  \n",
       "5              2            2838            .  \n",
       "6             57            2838  rs149121829  \n",
       "7              2            2838  rs201629093  \n",
       "8             18            2838  rs141952333  \n",
       "9              6            2838  rs202127368  \n",
       "10             1            2838            .  \n",
       "11             8            2838  rs201263331  \n",
       "12             1            2838  rs372257861  \n",
       "13            33            2838   rs61748101  \n",
       "14             1            2838  rs147104839  \n",
       "15           109            2838   rs61748102  \n",
       "16            65            2838   rs61748103  \n",
       "17          1385            2838    rs3762672  \n",
       "18             1            2838            .  \n",
       "19           178            2838   rs55825559  \n",
       "20            19            2838  rs139620588  \n",
       "21             1            2838  rs202044084  \n",
       "22             1            2838  rs141090992  \n",
       "23           320            2838   rs79953286  \n",
       "24             9            2838  rs142160751  \n",
       "25             2            2838            .  \n",
       "26            12            2838  rs145242123  \n",
       "27             6            2838  rs147575919  \n",
       "28             8            2838   rs10935014  \n",
       "29            10            2838  rs138693725  \n",
       "30             4            2838            .  \n",
       "31            23            2838  rs140537885  \n",
       "32            23            2838  rs138367039  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "#    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "#    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "#        'Freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "#        'Freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['SNP'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Merge with assoc results \n",
    "full_results = pd.merge(assoc_and_freq, df_results, on=\"SNP\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_full_results = full_results[['SNP', 'A1', 'A2', 'F_A', 'F_U', 'P', 'OR','L95', 'U95', 'BONF', \n",
    "                                   'MAF', 'MAF_A', 'MAF_U', \n",
    "                                   'Hom Cases', 'Het Cases', 'Total Cases', \n",
    "                                   'Hom Controls','Het Controls', 'Total Controls']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(16833, 19)\n",
      "No. of SNPs: 16833\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs149480465</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0.69670</td>\n",
       "      <td>0.6231</td>\n",
       "      <td>0.05648</td>\n",
       "      <td>6.874</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000293</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs147898644</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.11430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0.20520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0.20520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0.20520</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000352</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16828</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.07316</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16829</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.07316</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16830</th>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.07316</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16831</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>0.66750</td>\n",
       "      <td>1.1390</td>\n",
       "      <td>0.62940</td>\n",
       "      <td>2.060</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.004301</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16832</th>\n",
       "      <td>rs138367039</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>0.66030</td>\n",
       "      <td>0.8666</td>\n",
       "      <td>0.45730</td>\n",
       "      <td>1.642</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.003812</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>0.004052</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>2277</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16833 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               SNP A1 A2       F_A       F_U        P      OR      L95    U95  \\\n",
       "0      rs149480465  T  A  0.000220  0.000352  0.69670  0.6231  0.05648  6.874   \n",
       "1      rs147898644  T  C  0.000439  0.000000  0.11430     NaN      NaN    NaN   \n",
       "2                .  C  G  0.000000  0.000352  0.20520  0.0000  0.00000    NaN   \n",
       "3                .  C  G  0.000000  0.000352  0.20520  0.0000  0.00000    NaN   \n",
       "4                .  C  G  0.000000  0.000352  0.20520  0.0000  0.00000    NaN   \n",
       "...            ... .. ..       ...       ...      ...     ...      ...    ...   \n",
       "16828            .  A  G  0.000000  0.000705  0.07316  0.0000  0.00000    NaN   \n",
       "16829            .  A  G  0.000000  0.000705  0.07316  0.0000  0.00000    NaN   \n",
       "16830            .  A  G  0.000000  0.000705  0.07316  0.0000  0.00000    NaN   \n",
       "16831  rs140537885  G  T  0.004611  0.004052  0.66750  1.1390  0.62940  2.060   \n",
       "16832  rs138367039  A  G  0.003513  0.004052  0.66030  0.8666  0.45730  1.642   \n",
       "\n",
       "       BONF       MAF     MAF_A     MAF_U  Hom Cases  Het Cases  Total Cases  \\\n",
       "0       1.0  0.000293  0.000220  0.000352          0          1         2277   \n",
       "1       1.0  0.000196  0.000439  0.000000          0          2         2277   \n",
       "2       1.0  0.000196  0.000000  0.000352          0          0         2277   \n",
       "3       1.0  0.000196  0.000000  0.000352          0          2         2277   \n",
       "4       1.0  0.000196  0.000000  0.000352          0          0         2277   \n",
       "...     ...       ...       ...       ...        ...        ...          ...   \n",
       "16828   1.0  0.000391  0.000000  0.000705          0          1         2277   \n",
       "16829   1.0  0.000391  0.000000  0.000705          0          1         2277   \n",
       "16830   1.0  0.000391  0.000000  0.000705          0          0         2277   \n",
       "16831   1.0  0.004301  0.004611  0.004052          0         21         2277   \n",
       "16832   1.0  0.003812  0.003513  0.004052          0         16         2277   \n",
       "\n",
       "       Hom Controls  Het Controls  Total Controls  \n",
       "0                 0             2            2838  \n",
       "1                 0             0            2838  \n",
       "2                 0             2            2838  \n",
       "3                 0             0            2838  \n",
       "4                 0             2            2838  \n",
       "...             ...           ...             ...  \n",
       "16828             0             1            2838  \n",
       "16829             0             2            2838  \n",
       "16830             0             4            2838  \n",
       "16831             0            23            2838  \n",
       "16832             0            23            2838  \n",
       "\n",
       "[16833 rows x 19 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clean_full_results.shape)\n",
    "print(f'No. of SNPs: {clean_full_results.shape[0]}')\n",
    "clean_full_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs61748102', 'rs3762672']"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_freq = clean_full_results[clean_full_results['P']<0.05]\n",
    "sig_snps = sig_freq['SNP'].tolist()\n",
    "sig_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>P</th>\n",
       "      <th>OR</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>BONF</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9615</th>\n",
       "      <td>rs61748102</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01449</td>\n",
       "      <td>0.01991</td>\n",
       "      <td>0.037870</td>\n",
       "      <td>0.724</td>\n",
       "      <td>0.5330</td>\n",
       "      <td>0.9833</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.0175</td>\n",
       "      <td>0.01449</td>\n",
       "      <td>0.01991</td>\n",
       "      <td>1</td>\n",
       "      <td>64</td>\n",
       "      <td>2277</td>\n",
       "      <td>2</td>\n",
       "      <td>109</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9617</th>\n",
       "      <td>rs3762672</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.45170</td>\n",
       "      <td>0.47760</td>\n",
       "      <td>0.008968</td>\n",
       "      <td>0.901</td>\n",
       "      <td>0.8332</td>\n",
       "      <td>0.9743</td>\n",
       "      <td>0.2959</td>\n",
       "      <td>0.4661</td>\n",
       "      <td>0.45170</td>\n",
       "      <td>0.47760</td>\n",
       "      <td>458</td>\n",
       "      <td>1141</td>\n",
       "      <td>2277</td>\n",
       "      <td>663</td>\n",
       "      <td>1385</td>\n",
       "      <td>2838</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             SNP A1 A2      F_A      F_U         P     OR     L95     U95  \\\n",
       "9615  rs61748102  T  C  0.01449  0.01991  0.037870  0.724  0.5330  0.9833   \n",
       "9617   rs3762672  T  G  0.45170  0.47760  0.008968  0.901  0.8332  0.9743   \n",
       "\n",
       "        BONF     MAF    MAF_A    MAF_U  Hom Cases  Het Cases  Total Cases  \\\n",
       "9615  1.0000  0.0175  0.01449  0.01991          1         64         2277   \n",
       "9617  0.2959  0.4661  0.45170  0.47760        458       1141         2277   \n",
       "\n",
       "      Hom Controls  Het Controls  Total Controls  \n",
       "9615             2           109            2838  \n",
       "9617           663          1385            2838  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Look at significant SNPs, if any \n",
    "sig_df_results = clean_full_results[clean_full_results['SNP'].isin(sig_snps)]\n",
    "sig_df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save files to VM\n",
    "clean_full_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt', sep=\"\\t\", index=False)\n",
    "sig_df_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt' , sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
