{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variants in DNAJC13 and their Association with Parkinson's Disease Across Different Ancestral Backgrounds\n",
    "\n",
    "* **Project**: DNAJC13 Gene Assesment\n",
    "* **Last updated**: September 2024\n",
    "* **Version**: Python 3.9\n",
    "* **Data**: AMP-PD Release 3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "This notebook is focused on analysing association between DNAJC13 variants in Ashkenazi Jewish (AJ) ancestry and Parkinson's disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_6380/3943855956.py:26: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "\n",
    "# Bring some visualization functionality \n",
    "import seaborn as sns\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## AMP-PD v2.5\n",
    "AMP_CLINICAL_RELEASE_PATH = f'{AMP_RELEASE_PATH}/clinical'\n",
    "\n",
    "AMP_WGS_RELEASE_PLINK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'plink')\n",
    "AMP_WGS_RELEASE_GATK_PATH = os.path.join(AMP_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "## Print the information to check we are in the proper release and billing \n",
    "## This will be different for you, the user, depending on the billing project your workspace is on\n",
    "print('Billing and Workspace')\n",
    "print(f'Workspace Name @ `WORKSPACE_NAME`: {WORKSPACE_NAME}')\n",
    "print(f'Billing Project @ `BILLING_PROJECT_ID`: {BILLING_PROJECT_ID}')\n",
    "print(f'Workspace Bucket, where you can upload and download data @ `WORKSPACE_BUCKET`: {WORKSPACE_BUCKET}')\n",
    "print('')\n",
    "\n",
    "print('AMP-PD v2.5')\n",
    "print(f'Path to AMP-PD v2.5 Clinical Data @ `AMP_CLINICAL_RELEASE_PATH`: {AMP_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to AMP-PD v2.5 WGS Data @ `AMP_WGS_RELEASE_PLINK_PATH`: {AMP_WGS_RELEASE_PLINK_PATH}')\n",
    "print('')\n",
    "\n",
    "## GP2 v5.0\n",
    "## Explicitly define release v5.0 path \n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v6.0')\n",
    "print(f'Path to GP2 v6.0 Clinical Data @ `GP2_CLINICAL_RELEASE_PATH`: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v6.0 Metadata @ `GP2_META_RELEASE_PATH`: {GP2_META_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v6.0 Raw Genotype Data @ `GP2_RAW_GENO_PATH`: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v6.0 Imputed Genotype Data @ `GP2_IMPUTED_GENO_PATH`: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry = \"AJ\"\n",
    "WORK_DIR = f'2024_BURDEN_AMP_{ancestry}_GLM'\n",
    "! mkdir {WORK_DIR}\n",
    "\n",
    "AMP_DATA_DIR = f'AMP_DATA_DIR'\n",
    "! mkdir {AMP_DATA_DIR}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/alpha3/plink2_linux_avx2_20220603.zip\n",
    "unzip -o plink2_linux_avx2_20220603.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR:\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "# Install ANNOVAR: Download resources for annotation\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20140902 humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb cytoBand humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ensGene humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar exac03 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp147 humandb/ \n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp30a humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar gnomad211_genome humandb/\n",
    "#perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar ljb26_all humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## chr 3\n",
    "# Commented out because it is already downloaded\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_WGS_RELEASE_PLINK_PATH}/pfiles/chr3.pgen {AMP_DATA_DIR}/chr3.pgen.new')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## clinical data \n",
    "\n",
    "## demographics\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_CLINICAL_RELEASE_PATH}/Demographics.csv {WORK_DIR}')\n",
    "\n",
    "## enrollment\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_CLINICAL_RELEASE_PATH}/Enrollment.csv {WORK_DIR}')\n",
    "\n",
    "## duplicates\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {AMP_RELEASE_PATH}/amp_pd_participant_wgs_duplicates.csv {WORK_DIR}')\n",
    "\n",
    "## related file \n",
    "#shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/release6_{ancestry}.related {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# refFlat\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp -r {WORKSPACE_BUCKET}/notebooks/misc/refFlat_HG38_all_chr.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Predicted ancestries uploaded by Mary\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMPPD_v25_GenoTools_Predictions.txt {WORK_DIR}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Additional files uploaded by Mary\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/PCA.FILTERED.AMP_PD_{ancestry}.PD.eigenvec {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/AMP_v3/toRemove_1stand2ndDegree_Relateds_{ancestry}.txt {WORK_DIR}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check files\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Covariate File"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the disease status\n",
    "enrollment = pd.read_csv(f'{WORK_DIR}/Enrollment.csv')\n",
    "print(enrollment.shape)\n",
    "enrollment.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the demographic data\n",
    "demographics = pd.read_csv(f'{WORK_DIR}/Demographics.csv')\n",
    "print(demographics.shape)\n",
    "demographics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's load the predicted ancestry\n",
    "predictedAncestry= pd.read_csv(f'{WORK_DIR}/AMPPD_v25_GenoTools_Predictions.txt', sep='\\s+' )\n",
    "print(predictedAncestry.shape)\n",
    "predictedAncestry.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get just the needed info from Enrollment\n",
    "enrollment = enrollment[['participant_id', 'study_arm']]\n",
    "enrollment.rename(columns = {'participant_id': 'IID',\n",
    "                      'study_arm':'phenotype'}, inplace = True)\n",
    "print(enrollment.shape)\n",
    "enrollment.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get just the needed info from Demographics\n",
    "demographics = demographics[['participant_id', 'sex', 'age_at_baseline']]\n",
    "demographics.rename(columns = {'participant_id': 'IID',\n",
    "                      'sex':'SEX',\n",
    "                      'age_at_baseline':'AGE'}, inplace = True)\n",
    "print(demographics.shape)\n",
    "demographics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Enrollment with Demographics\n",
    "merged_key = pd.merge(enrollment, demographics, on='IID', how='left')\n",
    "print(merged_key.shape)\n",
    "merged_key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Enrollment/Demographics with Ancestry\n",
    "merged_key2 = pd.merge(merged_key, predictedAncestry, on='IID', how='left')\n",
    "print(merged_key2.shape)\n",
    "merged_key2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_key2['label'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_key2['phenotype'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_key2[merged_key2['label']==ancestry]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Subset to keep ancestry of interest \n",
    "ancestry_key = merged_key2[merged_key2['label']==ancestry].copy()\n",
    "ancestry_key.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert phenotype to binary (0/1)\n",
    "    # PD = 1; control = 0\n",
    "pheno_mapping = {\"PD\": 2, \"Healthy Control\": 1}\n",
    "ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check value counts of pheno\n",
    "ancestry_key['PHENO'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert sex to binary (0/1)\n",
    "    # Female = 1; male = 0\n",
    "sex_mapping = {'Female': 2, 'Male': 1}\n",
    "ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check value counts of SEX\n",
    "ancestry_key['SEX'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Clean up and keep columns we need \n",
    "final_df = ancestry_key[['FID','IID', 'SEX', 'PHENO', 'AGE']].copy()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove duplicate values in IID column\n",
    "final_df = final_df.drop_duplicates(subset=['IID'], keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(final_df.shape)\n",
    "final_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load information about related individuals \n",
    "column_names = ['FID','IID']\n",
    "related_df = pd.read_csv(f'{WORK_DIR}/toRemove_1stand2ndDegree_Relateds_{ancestry}.txt', sep='\\t', header = None, names=column_names)\n",
    "print(related_df.shape)\n",
    "related_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of just one set of related people\n",
    "related_list = list(related_df['IID'])\n",
    "\n",
    "# Check value counts of related and remove only one related individual\n",
    "final_df = final_df[~final_df[\"IID\"].isin(related_list)]\n",
    "\n",
    "# Check size\n",
    "print(final_df.shape)\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Make file of sample IDs to keep \n",
    "samples_toKeep = final_df[['FID','IID']].copy()\n",
    "samples_toKeep.to_csv(f'{WORK_DIR}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! head {WORK_DIR}/{ancestry}.samplestoKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#final_df['FID'] = 0\n",
    "final_df['FATID'] = 0\n",
    "final_df['MATID'] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df2=final_df[['FID','IID','MATID','FATID','SEX','PHENO', 'AGE']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load PCs\n",
    "pcs = pd.read_csv(f'{WORK_DIR}/PCA.FILTERED.AMP_PD_{ancestry}.PD.eigenvec', sep='\\t')\n",
    "print(pcs.shape)\n",
    "pcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Cov and PCs\n",
    "final_df3 = pd.merge(final_df2, pcs, on='IID', how='left')\n",
    "print(final_df3.shape)\n",
    "final_df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df3.drop(['#FID', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10'], axis='columns', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(final_df3.shape)\n",
    "final_df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df3.groupby(['PHENO'])['SEX'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df3.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df3=pd.read_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FID       0\n",
       "IID       0\n",
       "MATID     0\n",
       "FATID     0\n",
       "SEX       0\n",
       "PHENO     0\n",
       "AGE       0\n",
       "PC1      24\n",
       "PC2      24\n",
       "PC3      24\n",
       "PC4      24\n",
       "PC5      24\n",
       "dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df3.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df3.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FID      0\n",
       "IID      0\n",
       "MATID    0\n",
       "FATID    0\n",
       "SEX      0\n",
       "PHENO    0\n",
       "AGE      0\n",
       "PC1      0\n",
       "PC2      0\n",
       "PC3      0\n",
       "PC4      0\n",
       "PC5      0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df3.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!head {WORK_DIR}/{ancestry}_covariate_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Save your covariate file\n",
    "final_df3.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls -lh {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls -lh {AMP_DATA_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## extract ancestry\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--pfile {AMP_DATA_DIR}/chr3 \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--make-pgen \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket) \n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_covariate_file.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_covariate_file.txt')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}.samplestoKeep {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}.samplestoKeep')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation using ANNOVAR\n",
    "\n",
    "* *DNAJC13* from NCBI gene\n",
    "* hg38 (chr3:132417502-132539032)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.log.\n",
      "Options in effect:\n",
      "  --chr 3\n",
      "  --export vcf id-paste=iid\n",
      "  --from-bp 132417502\n",
      "  --mac 2\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13\n",
      "  --pfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\n",
      "  --to-bp 132539032\n",
      "\n",
      "Start time: Sun Aug 18 04:43:25 2024\n",
      "7450 MiB RAM detected, ~5685 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) loaded from\n",
      "2024_BURDEN_AMP_AJ_GLM/chr3_AJ.psam.\n",
      "6005 variants loaded from 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "5369 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "636 variants remaining after main filters.\n",
      "--export vcf to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.vcf ... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Sun Aug 18 04:43:25 2024\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--chr 3 \\\n",
    "--from-bp 132417502  \\\n",
    "--to-bp 132539032 \\\n",
    "--recode vcf id-paste=iid \\\n",
    "--mac 2 \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.vcf.gz > 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.avinput>\n",
      "NOTICE: Finished reading 697 lines from VCF file\n",
      "NOTICE: A total of 636 locus in VCF file passed QC threshold, representing 526 SNPs (341 transitions and 185 transversions) and 289 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 238804 SNP genotypes (154814 transitions and 83990 transversions) and 131206 indels/substitutions for 454 samples\n",
      "WARNING: 63 invalid alternative alleles found in input file\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.refGene -exonsort -nofirstcodondel 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.refGene.variant_function, 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 878 unique variants in 878 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 878 unique variants in 878 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 6\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_DNAJC13.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "-out {WORK_DIR}/{ancestry}_DNAJC13.annovar \\\n",
    "-remove -protocol refGene,clinvar_20140902 \\\n",
    "-operation g,f \\\n",
    "--nopolish \\\n",
    "-nastring . \\\n",
    "-vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo457</th>\n",
       "      <th>Otherinfo458</th>\n",
       "      <th>Otherinfo459</th>\n",
       "      <th>Otherinfo460</th>\n",
       "      <th>Otherinfo461</th>\n",
       "      <th>Otherinfo462</th>\n",
       "      <th>Otherinfo463</th>\n",
       "      <th>Otherinfo464</th>\n",
       "      <th>Otherinfo465</th>\n",
       "      <th>Otherinfo466</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132417864</td>\n",
       "      <td>132417864</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132418296</td>\n",
       "      <td>132418296</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132418415</td>\n",
       "      <td>132418415</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132418488</td>\n",
       "      <td>132418488</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132418506</td>\n",
       "      <td>132418506</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>3</td>\n",
       "      <td>132538575</td>\n",
       "      <td>132538575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*293A&gt;G;NM_015268:c.*293A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>3</td>\n",
       "      <td>132538679</td>\n",
       "      <td>132538679</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*397A&gt;T;NM_015268:c.*397A&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>3</td>\n",
       "      <td>132538787</td>\n",
       "      <td>132538787</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*505T&gt;G;NM_015268:c.*505T&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>3</td>\n",
       "      <td>132538890</td>\n",
       "      <td>132538890</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*608G&gt;A;NM_015268:c.*608G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>877</th>\n",
       "      <td>3</td>\n",
       "      <td>132538903</td>\n",
       "      <td>132538903</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>NM_001329126:c.*621G&gt;A;NM_015268:c.*621G&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>878 rows × 477 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      3  132417864  132417864   G   C     intronic      DNAJC13   \n",
       "1      3  132418296  132418296   C   T     intronic      DNAJC13   \n",
       "2      3  132418415  132418415   A   G     intronic      DNAJC13   \n",
       "3      3  132418488  132418488   G   T     intronic      DNAJC13   \n",
       "4      3  132418506  132418506   T   G     intronic      DNAJC13   \n",
       "..   ...        ...        ...  ..  ..          ...          ...   \n",
       "873    3  132538575  132538575   A   G         UTR3      DNAJC13   \n",
       "874    3  132538679  132538679   A   T         UTR3      DNAJC13   \n",
       "875    3  132538787  132538787   T   G         UTR3      DNAJC13   \n",
       "876    3  132538890  132538890   G   A         UTR3      DNAJC13   \n",
       "877    3  132538903  132538903   G   A         UTR3      DNAJC13   \n",
       "\n",
       "                             GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                             .                  .   \n",
       "1                                             .                  .   \n",
       "2                                             .                  .   \n",
       "3                                             .                  .   \n",
       "4                                             .                  .   \n",
       "..                                          ...                ...   \n",
       "873  NM_001329126:c.*293A>G;NM_015268:c.*293A>G                  .   \n",
       "874  NM_001329126:c.*397A>T;NM_015268:c.*397A>T                  .   \n",
       "875  NM_001329126:c.*505T>G;NM_015268:c.*505T>G                  .   \n",
       "876  NM_001329126:c.*608G>A;NM_015268:c.*608G>A                  .   \n",
       "877  NM_001329126:c.*621G>A;NM_015268:c.*621G>A                  .   \n",
       "\n",
       "    AAChange.refGene  ... Otherinfo457  Otherinfo458  Otherinfo459  \\\n",
       "0                  .  ...          0/0           0/0           0/0   \n",
       "1                  .  ...          0/0           0/0           0/0   \n",
       "2                  .  ...          0/0           1/1           0/0   \n",
       "3                  .  ...          0/0           0/0           0/0   \n",
       "4                  .  ...          0/0           0/0           0/0   \n",
       "..               ...  ...          ...           ...           ...   \n",
       "873                .  ...          0/0           1/1           0/0   \n",
       "874                .  ...          0/0           0/0           0/0   \n",
       "875                .  ...          0/0           0/0           0/0   \n",
       "876                .  ...          0/0           0/0           0/0   \n",
       "877                .  ...          1/1           0/0           0/1   \n",
       "\n",
       "     Otherinfo460  Otherinfo461  Otherinfo462 Otherinfo463 Otherinfo464  \\\n",
       "0             0/0           0/0           0/0          0/0          0/0   \n",
       "1             0/0           0/0           0/0          0/0          0/0   \n",
       "2             0/0           0/0           0/0          0/0          1/1   \n",
       "3             0/0           0/0           0/0          0/0          0/0   \n",
       "4             0/0           0/0           0/0          0/0          0/0   \n",
       "..            ...           ...           ...          ...          ...   \n",
       "873           0/0           0/0           0/0          0/0          1/1   \n",
       "874           0/0           0/0           0/0          0/0          0/0   \n",
       "875           0/0           0/0           0/0          0/0          0/0   \n",
       "876           0/0           0/0           0/0          0/0          0/0   \n",
       "877           0/0           0/1           0/0          0/1          0/0   \n",
       "\n",
       "    Otherinfo465  Otherinfo466  \n",
       "0            0/0           0/0  \n",
       "1            0/0           0/0  \n",
       "2            0/0           0/0  \n",
       "3            0/0           0/0  \n",
       "4            0/0           0/0  \n",
       "..           ...           ...  \n",
       "873          0/0           0/0  \n",
       "874          0/0           0/0  \n",
       "875          0/0           0/0  \n",
       "876          0/0           0/0  \n",
       "877          0/1           0/0  \n",
       "\n",
       "[878 rows x 477 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter intronic\n",
    "intronic = gene[(gene['Func.refGene'] == 'intronic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR3\n",
    "utr3 = gene[(gene['Func.refGene'] == 'UTR3')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter UTR5\n",
    "utr5 = gene[(gene['Func.refGene'] == 'UTR5')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter exonic and synonymous variants\n",
    "coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>Otherinfo457</th>\n",
       "      <th>Otherinfo458</th>\n",
       "      <th>Otherinfo459</th>\n",
       "      <th>Otherinfo460</th>\n",
       "      <th>Otherinfo461</th>\n",
       "      <th>Otherinfo462</th>\n",
       "      <th>Otherinfo463</th>\n",
       "      <th>Otherinfo464</th>\n",
       "      <th>Otherinfo465</th>\n",
       "      <th>Otherinfo466</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>132434618</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon2:c.A68G:p.K23R,DNAJC...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>132447465</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_001329126:exon4:c.G289A:p.A97T,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>790</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>DNAJC13</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...</td>\n",
       "      <td>...</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 477 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "121    3  132434618  132434618   A   G       exonic      DNAJC13   \n",
       "197    3  132447465  132447465   G   A       exonic      DNAJC13   \n",
       "444    3  132475009  132475009   C   A       exonic      DNAJC13   \n",
       "595    3  132499231  132499231   C   T       exonic      DNAJC13   \n",
       "600    3  132499777  132499777   G   A       exonic      DNAJC13   \n",
       "601    3  132499779  132499779   G   T       exonic      DNAJC13   \n",
       "614    3  132502295  132502295   C   T       exonic      DNAJC13   \n",
       "615    3  132502299  132502299   G   A       exonic      DNAJC13   \n",
       "684    3  132507256  132507256   A   G       exonic      DNAJC13   \n",
       "790    3  132528316  132528316   T   G       exonic      DNAJC13   \n",
       "\n",
       "    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "121                  .  nonsynonymous SNV   \n",
       "197                  .  nonsynonymous SNV   \n",
       "444                  .  nonsynonymous SNV   \n",
       "595                  .  nonsynonymous SNV   \n",
       "600                  .  nonsynonymous SNV   \n",
       "601                  .  nonsynonymous SNV   \n",
       "614                  .  nonsynonymous SNV   \n",
       "615                  .  nonsynonymous SNV   \n",
       "684                  .  nonsynonymous SNV   \n",
       "790                  .  nonsynonymous SNV   \n",
       "\n",
       "                                      AAChange.refGene  ... Otherinfo457  \\\n",
       "121  DNAJC13:NM_001329126:exon2:c.A68G:p.K23R,DNAJC...  ...          0/0   \n",
       "197  DNAJC13:NM_001329126:exon4:c.G289A:p.A97T,DNAJ...  ...          0/0   \n",
       "444  DNAJC13:NM_015268:exon22:c.C2369A:p.S790Y,DNAJ...  ...          0/0   \n",
       "595  DNAJC13:NM_015268:exon37:c.C4262T:p.A1421V,DNA...  ...          0/0   \n",
       "600  DNAJC13:NM_015268:exon38:c.G4385A:p.R1462H,DNA...  ...          0/0   \n",
       "601  DNAJC13:NM_015268:exon38:c.G4387T:p.A1463S,DNA...  ...          1/1   \n",
       "614  DNAJC13:NM_015268:exon40:c.C4543T:p.P1515S,DNA...  ...          0/0   \n",
       "615  DNAJC13:NM_015268:exon40:c.G4547A:p.R1516H,DNA...  ...          0/0   \n",
       "684  DNAJC13:NM_015268:exon43:c.A5018G:p.Y1673C,DNA...  ...          0/0   \n",
       "790  DNAJC13:NM_015268:exon54:c.T6509G:p.L2170W,DNA...  ...          0/0   \n",
       "\n",
       "     Otherinfo458  Otherinfo459  Otherinfo460  Otherinfo461  Otherinfo462  \\\n",
       "121           0/0           0/0           0/0           0/0           0/0   \n",
       "197           0/0           0/0           0/0           0/0           0/0   \n",
       "444           0/0           0/0           0/0           0/0           0/0   \n",
       "595           0/0           0/0           0/0           0/0           0/0   \n",
       "600           0/0           0/0           0/0           0/0           0/0   \n",
       "601           1/1           0/1           0/0           0/1           0/0   \n",
       "614           0/0           0/0           0/0           0/0           0/0   \n",
       "615           0/0           0/0           0/0           0/0           0/0   \n",
       "684           0/1           0/0           0/0           0/0           0/0   \n",
       "790           0/0           0/0           0/0           0/0           0/0   \n",
       "\n",
       "    Otherinfo463 Otherinfo464 Otherinfo465  Otherinfo466  \n",
       "121          0/0          0/0          0/0           0/0  \n",
       "197          0/0          0/0          0/0           0/0  \n",
       "444          0/0          0/0          0/0           0/0  \n",
       "595          0/0          0/0          0/0           0/0  \n",
       "600          0/0          0/0          0/0           0/0  \n",
       "601          0/1          1/1          0/1           0/0  \n",
       "614          0/1          0/0          0/0           0/0  \n",
       "615          0/0          0/0          0/0           0/0  \n",
       "684          0/0          1/1          0/0           0/0  \n",
       "790          0/0          0/1          0/0           0/0  \n",
       "\n",
       "[10 rows x 477 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic and non-synonymous variants\n",
    "coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "coding_nonsynonymous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "totalVariants = len(gene.axes[0])\n",
    "totalIntronic = len(intronic.axes[0])\n",
    "totalUTR3 = len(utr3.axes[0])\n",
    "totalUTR5 = len(utr5.axes[0])\n",
    "totalExonicSyn = len(coding_synonymous.axes[0])\n",
    "totalExonicNonSyn = len(coding_nonsynonymous.axes[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Variants:  878\n",
      "Intronic: 856\n",
      "UTR3: 6\n",
      "UTR5: 0\n",
      "Exonic Syn: 6\n",
      "Exonic NonSyn: 10\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Variants: \", totalVariants)\n",
    "print(\"Intronic:\", totalIntronic)\n",
    "print(\"UTR3:\", totalUTR3)\n",
    "print(\"UTR5:\", totalUTR5)\n",
    "print(\"Exonic Syn:\", totalExonicSyn)\n",
    "print(\"Exonic NonSyn:\", totalExonicNonSyn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>132434618</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>132447465</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>132475009</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>132499231</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>132499777</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>601</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>132499779</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>132502295</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>132502299</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>132507256</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>790</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>132528316</td>\n",
       "      <td>DNAJC13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Gene.refGene\n",
       "121    3  132434618  132434618      DNAJC13\n",
       "197    3  132447465  132447465      DNAJC13\n",
       "444    3  132475009  132475009      DNAJC13\n",
       "595    3  132499231  132499231      DNAJC13\n",
       "600    3  132499777  132499777      DNAJC13\n",
       "601    3  132499779  132499779      DNAJC13\n",
       "614    3  132502295  132502295      DNAJC13\n",
       "615    3  132502299  132502299      DNAJC13\n",
       "684    3  132507256  132507256      DNAJC13\n",
       "790    3  132528316  132528316      DNAJC13"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save in PLINK format \n",
    "variants_toKeep = coding_nonsynonymous[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep', sep=\"\\t\", index=False, header=False)\n",
    "variants_toKeep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_covariate_file.txt\r\n",
      "AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "AJ_DNAJC13.annovar.avinput\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "AJ_DNAJC13.log\r\n",
      "AJ_DNAJC13.vcf.gz\r\n",
      "AJ_DNAJC13.vcf.gz.tbi\r\n",
      "AJ.samplestoKeep\r\n",
      "amp_pd_participant_wgs_duplicates.csv\r\n",
      "AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "chr3_AJ.log\r\n",
      "chr3_AJ.pgen\r\n",
      "chr3_AJ.psam\r\n",
      "chr3_AJ.pvar\r\n",
      "Demographics.csv\r\n",
      "Enrollment.csv\r\n",
      "PCA.FILTERED.AMP_PD_AJ.PD.eigenvec\r\n",
      "plink2\r\n",
      "plink2_linux_amd_avx2_20240806.zip\r\n",
      "plink2_linux_avx2_20240625.zip\r\n",
      "plink2_linux_avx2_20240806.zip\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "toRemove_1stand2ndDegree_Relateds_AJ.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket) (not copying right now)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\n",
      "  --pfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\n",
      "\n",
      "Start time: Sun Aug 18 04:46:15 2024\n",
      "7450 MiB RAM detected, ~6043 available; reserving 3725 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) loaded from\n",
      "2024_BURDEN_AMP_AJ_GLM/chr3_AJ.psam.\n",
      "5615 out of 6005 variants loaded from 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.pvar.\n",
      "Note: No phenotype data present.\n",
      "5615 variants remaining after main filters.\n",
      "Writing 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.fam ... done.\n",
      "Writing 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.bim ... done.\n",
      "Writing 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.bed ... done.\n",
      "End time: Sun Aug 18 04:46:15 2024\n"
     ]
    }
   ],
   "source": [
    "# Convert the files from Plink 2.0 to Plink 1.9 format \n",
    "\n",
    "! /home/jupyter/tools/plink2 --pfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--make-bed \\\n",
    "--max-alleles 2 \\\n",
    "--out {WORK_DIR}/chr3_{ancestry}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.log.\n",
      "Options in effect:\n",
      "  --bfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\n",
      "  --extract range 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\n",
      "  --keep 2024_BURDEN_AMP_AJ_GLM/AJ.samplestoKeep\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn\n",
      "  --recode vcf-iid\n",
      "\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\n",
      "5615 variants loaded from .bim file.\n",
      "454 people (0 males, 0 females, 454 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.nosex .\n",
      "--extract range: 5605 variants excluded.\n",
      "--extract range: 10 variants remaining.\n",
      "--keep: 454 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 454 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "10 variants and 454 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode vcf-iid to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.vcf ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "    # later do based on class and frequency\n",
    "\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--recode vcf-iid \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "! bgzip -f {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf\n",
    "! tabix -f -p vcf {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.burden.coding_nonsyn]\n",
      "                          --outputRaw\n",
      "       Specify Covariate: --covar [], --covar-name [], --sex\n",
      "       Specify Phenotype: --pheno [2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skat\n",
      "                         skato], --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [2024_BURDEN_AMP_AJ_GLM/refFlat_HG38_all_chr.txt]\n",
      "                          --gene [DNAJC13], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.vcf.gz\n",
      "    --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.burden.coding_nonsyn\n",
      "    --pheno 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skat,skato\n",
      "    --geneFile 2024_BURDEN_AMP_AJ_GLM/refFlat_HG38_all_chr.txt\n",
      "    --gene DNAJC13\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Sun Aug 18 04:48:34 2024\n",
      "[INFO]\tLoaded [ 454 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 454 ] sample phenotypes\n",
      "[INFO]\tLoaded 260 male, 194 female and 0 sex-unknown samples from 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\n",
      "[INFO]\tLoaded 194 cases, 260 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 454 ] samples...\n",
      "[INFO]\tSKAT test significance will be evaluated using 10000 permutations at alpha = 0.05 weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 10 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Sun Aug 18 04:48:35 2024\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "## RVtests with covariates \n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn \\\n",
    "--kernel skat,skato \\\n",
    "--inVcf {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.vcf.gz \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene DNAJC13 \\\n",
    "--geneFile {WORK_DIR}/refFlat_HG38_all_chr.txt #\\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX AGE PC1 PC2 PC3 PC4 PC5\n",
    "#[WARN]Covariate [ PC5 ] has strong correlation [ r^2 = 1 ] with the response!\n",
    "\n",
    "\n",
    "# --out : Name of output \n",
    "# --burden cmc --kernel skato: tests to run \n",
    "# --inVcf : VCF file \n",
    "# --gene: gene name (if only looking at one or a few)\n",
    "# --geneFile refFlat.txt\n",
    "# --pheno :  covar file\n",
    "# --mpheno : # column that has phenotype information\n",
    "# --pheno-name : column name with phenotype in file\n",
    "# --covar : covar file\n",
    "# --freqUpper : optional, MAF cut-off\n",
    "# --covar-name : covariates, listed by column name, separated by commas (no spaces between commas)\n",
    "## 0=controls; 1=cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t454\t10\t10\t1531.19\t0.958643\t10000\t1031\t1531.19\t1000\t0\t0.969932\r\n"
     ]
    }
   ],
   "source": [
    "## look at results \n",
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\r\n",
      "DNAJC13\t3:132417501-132539032,3:132417501-132539032\t454\t10\t10\t199.482\t1\t0.924805\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AJ_covariate_file.txt\r\n",
      "AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "AJ_DNAJC13.annovar.avinput\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.txt\r\n",
      "AJ_DNAJC13.annovar.hg38_multianno.vcf\r\n",
      "AJ_DNAJC13.burden.coding_nonsyn.log\r\n",
      "AJ_DNAJC13.burden.coding_nonsyn.Skat.assoc\r\n",
      "AJ_DNAJC13.burden.coding_nonsyn.SkatO.assoc\r\n",
      "AJ_DNAJC13.coding_nonsyn.log\r\n",
      "AJ_DNAJC13.coding_nonsyn.nosex\r\n",
      "AJ_DNAJC13.coding_nonsyn.vcf.gz\r\n",
      "AJ_DNAJC13.coding_nonsyn.vcf.gz.tbi\r\n",
      "AJ_DNAJC13.log\r\n",
      "AJ_DNAJC13.vcf.gz\r\n",
      "AJ_DNAJC13.vcf.gz.tbi\r\n",
      "AJ.samplestoKeep\r\n",
      "amp_pd_participant_wgs_duplicates.csv\r\n",
      "AMPPD_v25_GenoTools_Predictions.txt\r\n",
      "AMPPD_v3_COV_wPHENOS_wGENOTOOLS.csv\r\n",
      "chr3_AJ.bed\r\n",
      "chr3_AJ.bim\r\n",
      "chr3_AJ.fam\r\n",
      "chr3_AJ.log\r\n",
      "chr3_AJ.pgen\r\n",
      "chr3_AJ.psam\r\n",
      "chr3_AJ.pvar\r\n",
      "Demographics.csv\r\n",
      "Enrollment.csv\r\n",
      "PCA.FILTERED.AMP_PD_AJ.PD.eigenvec\r\n",
      "plink2\r\n",
      "plink2_linux_amd_avx2_20240806.zip\r\n",
      "plink2_linux_avx2_20240625.zip\r\n",
      "plink2_linux_avx2_20240806.zip\r\n",
      "refFlat_HG38_all_chr.txt\r\n",
      "toRemove_1stand2ndDegree_Relateds_AJ.txt\r\n"
     ]
    }
   ],
   "source": [
    "## check to make sure file was created and saved\n",
    "! ls {WORK_DIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.burden.coding_nonsyn.*.assoc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --allow-no-sex\r\n",
      "  --assoc\r\n",
      "  --bfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\r\n",
      "  --extract range 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --keep 2024_BURDEN_AMP_AJ_GLM/AJ.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "454 people (0 males, 0 females, 454 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.nosex .\r\n",
      "430 phenotype values present after --pheno.\r\n",
      "--extract range: 5605 variants excluded.\r\n",
      "--extract range: 10 variants remaining.\r\n",
      "--keep: 454 people remaining.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Before main variant filters, 454 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "10 variants and 454 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 186 are cases and 244 are controls.  (24 phenotypes\r\n",
      "are missing.)\r\n",
      "Writing C/C --assoc report to\r\n",
      "2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.assoc ...\r\n",
      "0%\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "## Run basic association analysis just to get allele freq\n",
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--allow-no-sex \\\n",
    "--assoc \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR           SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR \r\n",
      "   3   rs374320101  132434618    G 0.002688 0.002049    A      0.03715       0.8472        1.313 \r\n",
      "   3   rs201731452  132447465    A 0.002688 0.004098    G       0.1208       0.7282        0.655 \r\n",
      "   3   rs149121829  132475009    A        0 0.004098    C        1.528       0.2164            0 \r\n",
      "   3    rs61748102  132499231    T  0.01075  0.01025    C     0.005235       0.9423         1.05 \r\n",
      "   3    rs61748103  132499777    A  0.01075 0.008197    G       0.1496       0.6989        1.315 \r\n",
      "   3     rs3762672  132499779    T   0.4409   0.5061    G        3.606      0.05757       0.7693 \r\n",
      "   3    rs55825559  132502295    T  0.02957  0.02254    C       0.4184       0.5177        1.321 \r\n",
      "   3   rs139620588  132502299    A 0.005376 0.004098    G      0.07447       0.7849        1.314 \r\n",
      "   3    rs79953286  132507256    G  0.08602   0.1004    A       0.5122       0.4742       0.8432 \r\n"
     ]
    }
   ],
   "source": [
    "!head {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs374320101</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.002049</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs201731452</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs149121829</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs61748102</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.010250</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs61748103</td>\n",
       "      <td>A</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP A1       F_A       F_U A2\n",
       "0  rs374320101  G  0.002688  0.002049  A\n",
       "1  rs201731452  A  0.002688  0.004098  G\n",
       "2  rs149121829  A  0.000000  0.004098  C\n",
       "3   rs61748102  T  0.010750  0.010250  C\n",
       "4   rs61748103  A  0.010750  0.008197  G"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc', delim_whitespace=True, usecols=['SNP', 'A1', 'A2', 'F_A', 'F_U'])\n",
    "assoc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "assoc.rename(columns = {'SNP': 'ID'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\r\n",
      "  --ci 0.95\r\n",
      "  --covar 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\r\n",
      "  --covar-variance-standardize\r\n",
      "  --extract range 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --glm\r\n",
      "  --keep 2024_BURDEN_AMP_AJ_GLM/AJ.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "Start time: Sun Aug 18 05:36:36 2024\r\n",
      "7450 MiB RAM detected, ~6066 available; reserving 3725 MiB for main workspace.\r\n",
      "Using up to 2 compute threads.\r\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) loaded from\r\n",
      "2024_BURDEN_AMP_AJ_GLM/chr3_AJ.fam.\r\n",
      "5615 variants loaded from 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.bim.\r\n",
      "1 binary phenotype loaded (186 cases, 244 controls).\r\n",
      "--extract bed1: 5605 variants excluded.\r\n",
      "--keep: 454 samples remaining.\r\n",
      "7 covariates loaded from 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt.\r\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) remaining after\r\n",
      "main filters.\r\n",
      "186 cases and 244 controls remaining after main filters.\r\n",
      "--covar-variance-standardize: 7 covariates transformed.\r\n",
      "Calculating allele frequencies... 0%\b\b0%\b\bdone.\r\n",
      "10 variants remaining after main filters.\r\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%\b\b0%\b\bdone.\r\n",
      "Results written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid .\r\n",
      "End time: Sun Aug 18 05:36:36 2024\r\n"
     ]
    }
   ],
   "source": [
    "## extract variants\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--glm \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --ci 0.95"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tPOS\tID\tREF\tALT\tPROVISIONAL_REF?\tA1\tOMITTED\tA1_FREQ\tFIRTH?\tTEST\tOBS_CT\tOR\tLOG(OR)_SE\tL95\tU95\tZ_STAT\tP\tERRCODE\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tADD\t430\t1.20208\t1.81173\t0.0344972\t41.8871\t0.101588\t0.919084\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tSEX\t430\t0.68523\t0.129709\t0.531407\t0.883579\t-2.91421\t0.0035659\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tAGE\t430\t0.360817\t0.147617\t0.270169\t0.481879\t-6.90563\t4.99816e-12\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tPC1\t430\t1.76299\t0.144115\t1.32916\t2.33841\t3.93443\t8.33943e-05\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tPC2\t430\t3.25313\t0.152445\t2.4129\t4.38594\t7.73798\t1.01006e-14\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tPC3\t430\t0.833073\t0.139121\t0.634254\t1.09422\t-1.31277\t0.18926\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tPC4\t430\t0.827548\t0.146617\t0.620859\t1.10305\t-1.29104\t0.19669\t.\r\n",
      "3\t132434618\trs374320101\tA\tG\tY\tG\tA\t0.00232558\tN\tPC5\t430\t1.17043\t0.134019\t0.900055\t1.52203\t1.17426\t0.240292\t.\r\n",
      "3\t132447465\trs201731452\tG\tA\tY\tA\tG\t0.00348837\tN\tADD\t430\t0.965621\t1.52228\t0.0488697\t19.0798\t-0.0229816\t0.981665\t.\r\n"
     ]
    }
   ],
   "source": [
    "! head {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>OR</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>rs374320101</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>1.202080</td>\n",
       "      <td>1.811730</td>\n",
       "      <td>0.034497</td>\n",
       "      <td>41.887100</td>\n",
       "      <td>0.101588</td>\n",
       "      <td>0.919084</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>rs201731452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003488</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.965621</td>\n",
       "      <td>1.522280</td>\n",
       "      <td>0.048870</td>\n",
       "      <td>19.079800</td>\n",
       "      <td>-0.022982</td>\n",
       "      <td>0.981665</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>rs149121829</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>Y</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>2.344900</td>\n",
       "      <td>1.571910</td>\n",
       "      <td>0.107675</td>\n",
       "      <td>51.066300</td>\n",
       "      <td>0.542170</td>\n",
       "      <td>0.587701</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>rs61748102</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.010465</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.473545</td>\n",
       "      <td>0.859488</td>\n",
       "      <td>0.087854</td>\n",
       "      <td>2.552460</td>\n",
       "      <td>-0.869713</td>\n",
       "      <td>0.384457</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>rs61748103</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.009302</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.972473</td>\n",
       "      <td>0.998177</td>\n",
       "      <td>0.137476</td>\n",
       "      <td>6.879030</td>\n",
       "      <td>-0.027964</td>\n",
       "      <td>0.977691</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>3</td>\n",
       "      <td>132499779</td>\n",
       "      <td>rs3762672</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.477907</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.791657</td>\n",
       "      <td>0.179982</td>\n",
       "      <td>0.556334</td>\n",
       "      <td>1.126520</td>\n",
       "      <td>-1.298050</td>\n",
       "      <td>0.194269</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>3</td>\n",
       "      <td>132502295</td>\n",
       "      <td>rs55825559</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.025581</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>1.387290</td>\n",
       "      <td>0.565699</td>\n",
       "      <td>0.457764</td>\n",
       "      <td>4.204270</td>\n",
       "      <td>0.578664</td>\n",
       "      <td>0.562816</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>3</td>\n",
       "      <td>132502299</td>\n",
       "      <td>rs139620588</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004651</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>1.699510</td>\n",
       "      <td>1.067460</td>\n",
       "      <td>0.209750</td>\n",
       "      <td>13.770400</td>\n",
       "      <td>0.496826</td>\n",
       "      <td>0.619311</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>3</td>\n",
       "      <td>132507256</td>\n",
       "      <td>rs79953286</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.094186</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.752730</td>\n",
       "      <td>0.318478</td>\n",
       "      <td>0.403228</td>\n",
       "      <td>1.405170</td>\n",
       "      <td>-0.891894</td>\n",
       "      <td>0.372450</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>3</td>\n",
       "      <td>132528316</td>\n",
       "      <td>rs140537885</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.005814</td>\n",
       "      <td>N</td>\n",
       "      <td>ADD</td>\n",
       "      <td>430</td>\n",
       "      <td>0.067229</td>\n",
       "      <td>1.232240</td>\n",
       "      <td>0.006007</td>\n",
       "      <td>0.752396</td>\n",
       "      <td>-2.190840</td>\n",
       "      <td>0.028464</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM        POS           ID REF ALT PROVISIONAL_REF? A1 OMITTED  \\\n",
       "0        3  132434618  rs374320101   A   G                Y  G       A   \n",
       "8        3  132447465  rs201731452   G   A                Y  A       G   \n",
       "16       3  132475009  rs149121829   C   A                Y  A       C   \n",
       "24       3  132499231   rs61748102   C   T                Y  T       C   \n",
       "32       3  132499777   rs61748103   G   A                Y  A       G   \n",
       "40       3  132499779    rs3762672   G   T                Y  T       G   \n",
       "48       3  132502295   rs55825559   C   T                Y  T       C   \n",
       "56       3  132502299  rs139620588   G   A                Y  A       G   \n",
       "64       3  132507256   rs79953286   A   G                Y  G       A   \n",
       "72       3  132528316  rs140537885   T   G                Y  G       T   \n",
       "\n",
       "     A1_FREQ FIRTH? TEST  OBS_CT        OR  LOG(OR)_SE       L95        U95  \\\n",
       "0   0.002326      N  ADD     430  1.202080    1.811730  0.034497  41.887100   \n",
       "8   0.003488      N  ADD     430  0.965621    1.522280  0.048870  19.079800   \n",
       "16  0.002326      Y  ADD     430  2.344900    1.571910  0.107675  51.066300   \n",
       "24  0.010465      N  ADD     430  0.473545    0.859488  0.087854   2.552460   \n",
       "32  0.009302      N  ADD     430  0.972473    0.998177  0.137476   6.879030   \n",
       "40  0.477907      N  ADD     430  0.791657    0.179982  0.556334   1.126520   \n",
       "48  0.025581      N  ADD     430  1.387290    0.565699  0.457764   4.204270   \n",
       "56  0.004651      N  ADD     430  1.699510    1.067460  0.209750  13.770400   \n",
       "64  0.094186      N  ADD     430  0.752730    0.318478  0.403228   1.405170   \n",
       "72  0.005814      N  ADD     430  0.067229    1.232240  0.006007   0.752396   \n",
       "\n",
       "      Z_STAT         P ERRCODE  \n",
       "0   0.101588  0.919084       .  \n",
       "8  -0.022982  0.981665       .  \n",
       "16  0.542170  0.587701       .  \n",
       "24 -0.869713  0.384457       .  \n",
       "32 -0.027964  0.977691       .  \n",
       "40 -1.298050  0.194269       .  \n",
       "48  0.578664  0.562816       .  \n",
       "56  0.496826  0.619311       .  \n",
       "64 -0.891894  0.372450       .  \n",
       "72 -2.190840  0.028464       .  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_glm = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid', delim_whitespace=True)\n",
    "assoc_add = assoc_glm[assoc_glm['TEST']==\"ADD\"]\n",
    "assoc_add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1_x</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>...</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "      <th>A1_y</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>rs374320101</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.811730</td>\n",
       "      <td>0.034497</td>\n",
       "      <td>41.88710</td>\n",
       "      <td>0.101588</td>\n",
       "      <td>0.919084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.002049</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>rs201731452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003488</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.522280</td>\n",
       "      <td>0.048870</td>\n",
       "      <td>19.07980</td>\n",
       "      <td>-0.022982</td>\n",
       "      <td>0.981665</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>rs149121829</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>Y</td>\n",
       "      <td>...</td>\n",
       "      <td>1.571910</td>\n",
       "      <td>0.107675</td>\n",
       "      <td>51.06630</td>\n",
       "      <td>0.542170</td>\n",
       "      <td>0.587701</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>rs61748102</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.010465</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.859488</td>\n",
       "      <td>0.087854</td>\n",
       "      <td>2.55246</td>\n",
       "      <td>-0.869713</td>\n",
       "      <td>0.384457</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.010250</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>rs61748103</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.009302</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.998177</td>\n",
       "      <td>0.137476</td>\n",
       "      <td>6.87903</td>\n",
       "      <td>-0.027964</td>\n",
       "      <td>0.977691</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS           ID REF ALT PROVISIONAL_REF? A1_x OMITTED  \\\n",
       "0       3  132434618  rs374320101   A   G                Y    G       A   \n",
       "1       3  132447465  rs201731452   G   A                Y    A       G   \n",
       "2       3  132475009  rs149121829   C   A                Y    A       C   \n",
       "3       3  132499231   rs61748102   C   T                Y    T       C   \n",
       "4       3  132499777   rs61748103   G   A                Y    A       G   \n",
       "\n",
       "    A1_FREQ FIRTH?  ... LOG(OR)_SE       L95       U95    Z_STAT         P  \\\n",
       "0  0.002326      N  ...   1.811730  0.034497  41.88710  0.101588  0.919084   \n",
       "1  0.003488      N  ...   1.522280  0.048870  19.07980 -0.022982  0.981665   \n",
       "2  0.002326      Y  ...   1.571910  0.107675  51.06630  0.542170  0.587701   \n",
       "3  0.010465      N  ...   0.859488  0.087854   2.55246 -0.869713  0.384457   \n",
       "4  0.009302      N  ...   0.998177  0.137476   6.87903 -0.027964  0.977691   \n",
       "\n",
       "   ERRCODE  A1_y       F_A       F_U A2  \n",
       "0        .     G  0.002688  0.002049  A  \n",
       "1        .     A  0.002688  0.004098  G  \n",
       "2        .     A  0.000000  0.004098  C  \n",
       "3        .     T  0.010750  0.010250  C  \n",
       "4        .     A  0.010750  0.008197  G  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Merge with allele freq from basic assoc analysis\n",
    "assoc_add = pd.merge(assoc_add, assoc, on=\"ID\", how=\"left\")\n",
    "assoc_add.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "assoc_add.drop(columns=['A1_y'])\n",
    "assoc_add.rename(columns = {'A1_x': 'A1'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM AVX2 AMD (4 Jul 2024)          www.cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --adjust\r\n",
      "  --bfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\r\n",
      "  --ci 0.95\r\n",
      "  --covar 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\r\n",
      "  --covar-variance-standardize\r\n",
      "  --extract range 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --glm\r\n",
      "  --keep 2024_BURDEN_AMP_AJ_GLM/AJ.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "\r\n",
      "Start time: Sun Aug 18 05:36:54 2024\r\n",
      "7450 MiB RAM detected, ~6064 available; reserving 3725 MiB for main workspace.\r\n",
      "Using up to 2 compute threads.\r\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) loaded from\r\n",
      "2024_BURDEN_AMP_AJ_GLM/chr3_AJ.fam.\r\n",
      "5615 variants loaded from 2024_BURDEN_AMP_AJ_GLM/chr3_AJ.bim.\r\n",
      "1 binary phenotype loaded (186 cases, 244 controls).\r\n",
      "--extract bed1: 5605 variants excluded.\r\n",
      "--keep: 454 samples remaining.\r\n",
      "7 covariates loaded from 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt.\r\n",
      "454 samples (0 females, 0 males, 454 ambiguous; 454 founders) remaining after\r\n",
      "main filters.\r\n",
      "186 cases and 244 controls remaining after main filters.\r\n",
      "--covar-variance-standardize: 7 covariates transformed.\r\n",
      "Calculating allele frequencies... 0%\b\b0%\b\bdone.\r\n",
      "10 variants remaining after main filters.\r\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO': 0%\b\b0%\b\bdone.\r\n",
      "Results written to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid .\r\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 0.689474.\r\n",
      "(Treating lambda as 1 in GC-corrected p-value calculation.)\r\n",
      "--adjust values (10 tests) written to\r\n",
      "2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid.adjusted\r\n",
      ".\r\n",
      "End time: Sun Aug 18 05:36:54 2024\r\n"
     ]
    }
   ],
   "source": [
    "## repeat association study applying multiple testing corrections for the raw p-values (like Bonferroni post-hoc)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--glm \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn --ci 0.95 --adjust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#CHROM\tID\tA1\tUNADJ\tGC\tBONF\tHOLM\tSIDAK_SS\tSIDAK_SD\tFDR_BH\tFDR_BY\r\n",
      "3\trs140537885\tG\t0.0284636\t0.0284636\t0.284636\t0.284636\t0.250812\t0.250812\t0.284636\t0.833689\r\n",
      "3\trs3762672\tT\t0.194269\t0.194269\t1\t1\t0.88468\t0.856876\t0.884731\t1\r\n",
      "3\trs79953286\tG\t0.37245\t0.37245\t1\t1\t0.990527\t0.975946\t0.884731\t1\r\n",
      "3\trs61748102\tT\t0.384457\t0.384457\t1\t1\t0.992191\t0.975946\t0.884731\t1\r\n",
      "3\trs55825559\tT\t0.562816\t0.562816\t1\t1\t0.999745\t0.993018\t0.884731\t1\r\n",
      "3\trs149121829\tA\t0.587701\t0.587701\t1\t1\t0.999858\t0.993018\t0.884731\t1\r\n",
      "3\trs139620588\tA\t0.619311\t0.619311\t1\t1\t0.999936\t0.993018\t0.884731\t1\r\n",
      "3\trs374320101\tG\t0.919084\t0.919084\t1\t1\t1\t0.99947\t0.981665\t1\r\n",
      "3\trs61748103\tA\t0.977691\t0.977691\t1\t1\t1\t0.999502\t0.981665\t1\r\n",
      "3\trs201731452\tA\t0.981665\t0.981665\t1\t1\t1\t0.999502\t0.981665\t1\r\n"
     ]
    }
   ],
   "source": [
    "! cat {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs140537885</td>\n",
       "      <td>0.284636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs3762672</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs79953286</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs61748102</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs55825559</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs149121829</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs139620588</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs374320101</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs61748103</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs201731452</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID      BONF\n",
       "0  rs140537885  0.284636\n",
       "1    rs3762672  1.000000\n",
       "2   rs79953286  1.000000\n",
       "3   rs61748102  1.000000\n",
       "4   rs55825559  1.000000\n",
       "5  rs149121829  1.000000\n",
       "6  rs139620588  1.000000\n",
       "7  rs374320101  1.000000\n",
       "8   rs61748103  1.000000\n",
       "9  rs201731452  1.000000"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assoc_adjusted = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.PHENO.glm.logistic.hybrid.adjusted', delim_whitespace=True, usecols=['ID', 'BONF'])\n",
    "assoc_adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PROVISIONAL_REF?</th>\n",
       "      <th>A1</th>\n",
       "      <th>OMITTED</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>FIRTH?</th>\n",
       "      <th>...</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "      <th>A1_y</th>\n",
       "      <th>F_A</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>132434618</td>\n",
       "      <td>rs374320101</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Y</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034497</td>\n",
       "      <td>41.88710</td>\n",
       "      <td>0.101588</td>\n",
       "      <td>0.919084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.002049</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>132447465</td>\n",
       "      <td>rs201731452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.003488</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.048870</td>\n",
       "      <td>19.07980</td>\n",
       "      <td>-0.022982</td>\n",
       "      <td>0.981665</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>132475009</td>\n",
       "      <td>rs149121829</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002326</td>\n",
       "      <td>Y</td>\n",
       "      <td>...</td>\n",
       "      <td>0.107675</td>\n",
       "      <td>51.06630</td>\n",
       "      <td>0.542170</td>\n",
       "      <td>0.587701</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>132499231</td>\n",
       "      <td>rs61748102</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Y</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.010465</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.087854</td>\n",
       "      <td>2.55246</td>\n",
       "      <td>-0.869713</td>\n",
       "      <td>0.384457</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.010250</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>132499777</td>\n",
       "      <td>rs61748103</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Y</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.009302</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137476</td>\n",
       "      <td>6.87903</td>\n",
       "      <td>-0.027964</td>\n",
       "      <td>0.977691</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>0.010750</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS           ID REF ALT PROVISIONAL_REF? A1 OMITTED  \\\n",
       "0       3  132434618  rs374320101   A   G                Y  G       A   \n",
       "1       3  132447465  rs201731452   G   A                Y  A       G   \n",
       "2       3  132475009  rs149121829   C   A                Y  A       C   \n",
       "3       3  132499231   rs61748102   C   T                Y  T       C   \n",
       "4       3  132499777   rs61748103   G   A                Y  A       G   \n",
       "\n",
       "    A1_FREQ FIRTH?  ...       L95       U95    Z_STAT         P  ERRCODE  \\\n",
       "0  0.002326      N  ...  0.034497  41.88710  0.101588  0.919084        .   \n",
       "1  0.003488      N  ...  0.048870  19.07980 -0.022982  0.981665        .   \n",
       "2  0.002326      Y  ...  0.107675  51.06630  0.542170  0.587701        .   \n",
       "3  0.010465      N  ...  0.087854   2.55246 -0.869713  0.384457        .   \n",
       "4  0.009302      N  ...  0.137476   6.87903 -0.027964  0.977691        .   \n",
       "\n",
       "   A1_y       F_A       F_U A2 BONF  \n",
       "0     G  0.002688  0.002049  A  1.0  \n",
       "1     A  0.002688  0.004098  G  1.0  \n",
       "2     A  0.000000  0.004098  C  1.0  \n",
       "3     T  0.010750  0.010250  C  1.0  \n",
       "4     A  0.010750  0.008197  G  1.0  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the results\n",
    "assoc_add = pd.merge(assoc_add, assoc_adjusted, on=\"ID\", how=\"left\")\n",
    "assoc_add.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\r\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.log.\r\n",
      "Options in effect:\r\n",
      "  --bfile 2024_BURDEN_AMP_AJ_GLM/chr3_AJ\r\n",
      "  --extract range 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.all_coding_nonsyn.variantstoKeep\r\n",
      "  --keep 2024_BURDEN_AMP_AJ_GLM/AJ.samplestoKeep\r\n",
      "  --out 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn\r\n",
      "  --pheno 2024_BURDEN_AMP_AJ_GLM/AJ_covariate_file.txt\r\n",
      "  --pheno-name PHENO\r\n",
      "  --recode A\r\n",
      "\r\n",
      "7450 MB RAM detected; reserving 3725 MB for main workspace.\r\n",
      "5615 variants loaded from .bim file.\r\n",
      "454 people (0 males, 0 females, 454 ambiguous) loaded from .fam.\r\n",
      "Ambiguous sex IDs written to\r\n",
      "2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.nosex .\r\n",
      "430 phenotype values present after --pheno.\r\n",
      "--extract range: 5605 variants excluded.\r\n",
      "--extract range: 10 variants remaining.\r\n",
      "--keep: 454 people remaining.\r\n",
      "Warning: Ignoring phenotypes of missing-sex samples.  If you don't want those\r\n",
      "phenotypes to be ignored, use the --allow-no-sex flag.\r\n",
      "Using 1 thread (no multithreaded calculations invoked).\r\n",
      "Before main variant filters, 454 founders and 0 nonfounders present.\r\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\r\n",
      "10 variants and 454 people pass filters and QC.\r\n",
      "Among remaining phenotypes, 186 are cases and 244 are controls.  (24 phenotypes\r\n",
      "are missing.)\r\n",
      "--recode A to 2024_BURDEN_AMP_AJ_GLM/AJ_DNAJC13.coding_nonsyn.raw ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\r\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/plink \\\n",
    "--bfile {WORK_DIR}/chr3_{ancestry} \\\n",
    "--extract range {WORK_DIR}/{ancestry}_DNAJC13.all_coding_nonsyn.variantstoKeep \\\n",
    "--pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "--recode A \\\n",
    "--out {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw', delim_whitespace=True)\n",
    "recode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of variants in AJ for DNAJC13: 10\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['rs374320101_G',\n",
       " 'rs201731452_A',\n",
       " 'rs149121829_A',\n",
       " 'rs61748102_T',\n",
       " 'rs61748103_A',\n",
       " 'rs3762672_T',\n",
       " 'rs55825559_T',\n",
       " 'rs139620588_A',\n",
       " 'rs79953286_G',\n",
       " 'rs140537885_G']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list from the column names\n",
    "column_names = recode.columns.tolist()\n",
    "\n",
    "# Drop the first 6 columns to keep the variants \n",
    "variants = column_names[6:]\n",
    "\n",
    "print(f'Number of variants in {ancestry} for DNAJC13: {len(variants)}')\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier freq in Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>Carrier freq in Controls</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs374320101_G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>186</td>\n",
       "      <td>0.005376</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>244</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>rs374320101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs201731452_A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>186</td>\n",
       "      <td>0.005376</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>244</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>rs201731452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs149121829_A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>186</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>244</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>rs149121829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs61748102_T</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>186</td>\n",
       "      <td>0.021505</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>244</td>\n",
       "      <td>0.020492</td>\n",
       "      <td>rs61748102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs61748103_A</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>186</td>\n",
       "      <td>0.021505</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>244</td>\n",
       "      <td>0.016393</td>\n",
       "      <td>rs61748103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs3762672_T</td>\n",
       "      <td>35</td>\n",
       "      <td>94</td>\n",
       "      <td>186</td>\n",
       "      <td>0.693548</td>\n",
       "      <td>65</td>\n",
       "      <td>117</td>\n",
       "      <td>244</td>\n",
       "      <td>0.745902</td>\n",
       "      <td>rs3762672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs55825559_T</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>186</td>\n",
       "      <td>0.059140</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>244</td>\n",
       "      <td>0.040984</td>\n",
       "      <td>rs55825559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs139620588_A</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>186</td>\n",
       "      <td>0.010753</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>244</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>rs139620588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs79953286_G</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>186</td>\n",
       "      <td>0.161290</td>\n",
       "      <td>4</td>\n",
       "      <td>41</td>\n",
       "      <td>244</td>\n",
       "      <td>0.184426</td>\n",
       "      <td>rs79953286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs140537885_G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>186</td>\n",
       "      <td>0.005376</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>244</td>\n",
       "      <td>0.016393</td>\n",
       "      <td>rs140537885</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Variant  Hom Cases  Het Cases  Total Cases  Carrier freq in Cases  \\\n",
       "0  rs374320101_G          0          1          186               0.005376   \n",
       "1  rs201731452_A          0          1          186               0.005376   \n",
       "2  rs149121829_A          0          0          186               0.000000   \n",
       "3   rs61748102_T          0          4          186               0.021505   \n",
       "4   rs61748103_A          0          4          186               0.021505   \n",
       "5    rs3762672_T         35         94          186               0.693548   \n",
       "6   rs55825559_T          0         11          186               0.059140   \n",
       "7  rs139620588_A          0          2          186               0.010753   \n",
       "8   rs79953286_G          2         28          186               0.161290   \n",
       "9  rs140537885_G          0          1          186               0.005376   \n",
       "\n",
       "   Hom Controls  Het Controls  Total Controls  Carrier freq in Controls  \\\n",
       "0             0             1             244                  0.004098   \n",
       "1             0             2             244                  0.008197   \n",
       "2             0             2             244                  0.008197   \n",
       "3             0             5             244                  0.020492   \n",
       "4             0             4             244                  0.016393   \n",
       "5            65           117             244                  0.745902   \n",
       "6             1             9             244                  0.040984   \n",
       "7             0             2             244                  0.008197   \n",
       "8             4            41             244                  0.184426   \n",
       "9             0             4             244                  0.016393   \n",
       "\n",
       "            ID  \n",
       "0  rs374320101  \n",
       "1  rs201731452  \n",
       "2  rs149121829  \n",
       "3   rs61748102  \n",
       "4   rs61748103  \n",
       "5    rs3762672  \n",
       "6   rs55825559  \n",
       "7  rs139620588  \n",
       "8   rs79953286  \n",
       "9  rs140537885  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pre-filter the dataset\n",
    "cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "results = []\n",
    "\n",
    "for variant in variants:\n",
    "    # For cases\n",
    "    hom_cases = cases_data[cases_data[variant] == 2].shape[0]\n",
    "    het_cases = cases_data[cases_data[variant] == 1].shape[0]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    freq_cases = (hom_cases + het_cases) / total_cases\n",
    "\n",
    "    # For controls\n",
    "    hom_controls = controls_data[controls_data[variant] == 2].shape[0]\n",
    "    het_controls = controls_data[controls_data[variant] == 1].shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    freq_controls = (hom_controls + het_controls) / total_controls\n",
    "\n",
    "    results.append({\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier freq in Cases': freq_cases,\n",
    "        'Hom Controls': hom_controls,\n",
    "        'Het Controls': het_controls,\n",
    "        'Total Controls': total_controls,\n",
    "        'Carrier freq in Controls': freq_controls\n",
    "    })\n",
    "\n",
    "# Return\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Merge case/control freq with the glm assoc analysis file\n",
    "full_results = pd.merge(df_results, assoc_add,  on=\"ID\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_full_results = full_results[['Variant', 'Hom Cases', 'Het Cases', 'Total Cases',\n",
    "                                   'Hom Controls','Het Controls', 'Total Controls',\n",
    "                                   'ID','A1_FREQ','A1','F_A', 'F_U', 'A2','OBS_CT','L95', 'OR','U95','LOG(OR)_SE','Z_STAT','P', 'BONF']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs140537885']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list of significant SNPs, if any\n",
    "sig_freq = clean_full_results[clean_full_results['P']<0.05]\n",
    "sig_snps = sig_freq['ID'].tolist()\n",
    "sig_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Hom Controls</th>\n",
       "      <th>Het Controls</th>\n",
       "      <th>Total Controls</th>\n",
       "      <th>ID</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>A1</th>\n",
       "      <th>...</th>\n",
       "      <th>F_U</th>\n",
       "      <th>A2</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>L95</th>\n",
       "      <th>OR</th>\n",
       "      <th>U95</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>BONF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs140537885_G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>186</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>244</td>\n",
       "      <td>rs140537885</td>\n",
       "      <td>0.005814</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008197</td>\n",
       "      <td>T</td>\n",
       "      <td>430</td>\n",
       "      <td>0.006007</td>\n",
       "      <td>0.067229</td>\n",
       "      <td>0.752396</td>\n",
       "      <td>1.23224</td>\n",
       "      <td>-2.19084</td>\n",
       "      <td>0.028464</td>\n",
       "      <td>0.284636</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Variant  Hom Cases  Het Cases  Total Cases  Hom Controls  \\\n",
       "9  rs140537885_G          0          1          186             0   \n",
       "\n",
       "   Het Controls  Total Controls           ID   A1_FREQ A1  ...       F_U  A2  \\\n",
       "9             4             244  rs140537885  0.005814  G  ...  0.008197   T   \n",
       "\n",
       "  OBS_CT       L95        OR       U95  LOG(OR)_SE   Z_STAT         P  \\\n",
       "9    430  0.006007  0.067229  0.752396     1.23224 -2.19084  0.028464   \n",
       "\n",
       "       BONF  \n",
       "9  0.284636  \n",
       "\n",
       "[1 rows x 21 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter SNPs found significant on basic analysis, if any, for comparison purposes \n",
    "sig_df_results = clean_full_results[clean_full_results['ID'].isin(sig_snps)]\n",
    "sig_df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9    0.002688\n",
       "Name: F_A, dtype: float64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_df_results[\"F_A\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save files to VM\n",
    "clean_full_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt', sep=\"\\t\", index=False)\n",
    "sig_df_results.to_csv(f'{WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt' , sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to workspace bucket (move from VM to workspace bucket)\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/{ancestry}_DNAJC13.annovar.hg38_multianno.txt')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.fullVariantInformation.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.SignificantVariantInformation.txt {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.raw {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq.cc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.frq {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry}_DNAJC13.coding_nonsyn.assoc.adjusted {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Check workspace bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} ls {WORKSPACE_BUCKET}/2024_BURDEN_AMP_{ancestry}/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
